// @adyen/api-library/lib/src/services/sessionAuthentication/sessionAuthenticationApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionAuthenticationApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/sessionAuthentication/objectSerializer");
/**
 * API handler for SessionAuthenticationApi
 */
class SessionAuthenticationApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://test.adyen.com/authe/api/v1";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create a session token
    * @param authenticationSessionRequest {@link AuthenticationSessionRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link AuthenticationSessionResponse }
    */
    async createAuthenticationSession(authenticationSessionRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/sessions`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(authenticationSessionRequest, "AuthenticationSessionRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "AuthenticationSessionResponse");
    }
}
exports.SessionAuthenticationApi = SessionAuthenticationApi;
