// @adyen/api-library/lib/src/services/balanceControl/balanceControlApi.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceControlApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/balanceControl/objectSerializer");
/**
 * API handler for BalanceControlApi
 */
class BalanceControlApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/BalanceControl/v1";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Start a balance transfer
    * @param balanceTransferRequest {@link BalanceTransferRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link BalanceTransferResponse }
    *
    * @deprecated since Adyen Balance Control API v1
    */
    async balanceTransfer(balanceTransferRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/balanceTransfer`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(balanceTransferRequest, "BalanceTransferRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "BalanceTransferResponse");
    }
}
exports.BalanceControlApi = BalanceControlApi;
