// @adyen/api-library/lib/src/typings/legalEntityManagement/businessLineInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessLineInfo = void 0;
class BusinessLineInfo {
    static getAttributeTypeMap() {
        return BusinessLineInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BusinessLineInfo = BusinessLineInfo;
BusinessLineInfo.discriminator = undefined;
BusinessLineInfo.mapping = undefined;
BusinessLineInfo.attributeTypeMap = [
    {
        "name": "capability",
        "baseName": "capability",
        "type": "BusinessLineInfo.CapabilityEnum",
        "format": ""
    },
    {
        "name": "industryCode",
        "baseName": "industryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string",
        "format": ""
    },
    {
        "name": "salesChannels",
        "baseName": "salesChannels",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "service",
        "baseName": "service",
        "type": "BusinessLineInfo.ServiceEnum",
        "format": ""
    },
    {
        "name": "sourceOfFunds",
        "baseName": "sourceOfFunds",
        "type": "SourceOfFunds | null",
        "format": ""
    },
    {
        "name": "webData",
        "baseName": "webData",
        "type": "Array<WebData>",
        "format": ""
    },
    {
        "name": "webDataExemption",
        "baseName": "webDataExemption",
        "type": "WebDataExemption | null",
        "format": ""
    }
];
(function (BusinessLineInfo) {
    let CapabilityEnum;
    (function (CapabilityEnum) {
        CapabilityEnum["ReceivePayments"] = "receivePayments";
        CapabilityEnum["ReceiveFromPlatformPayments"] = "receiveFromPlatformPayments";
        CapabilityEnum["IssueBankAccount"] = "issueBankAccount";
    })(CapabilityEnum = BusinessLineInfo.CapabilityEnum || (BusinessLineInfo.CapabilityEnum = {}));
    let ServiceEnum;
    (function (ServiceEnum) {
        ServiceEnum["PaymentProcessing"] = "paymentProcessing";
        ServiceEnum["Banking"] = "banking";
    })(ServiceEnum = BusinessLineInfo.ServiceEnum || (BusinessLineInfo.ServiceEnum = {}));
})(BusinessLineInfo = exports.BusinessLineInfo || (exports.BusinessLineInfo = {}));
