// @adyen/api-library/lib/src/typings/legalEntityManagement/calculatePciStatusRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CalculatePciStatusRequest = void 0;
class CalculatePciStatusRequest {
    static getAttributeTypeMap() {
        return CalculatePciStatusRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CalculatePciStatusRequest = CalculatePciStatusRequest;
CalculatePciStatusRequest.discriminator = undefined;
CalculatePciStatusRequest.mapping = undefined;
CalculatePciStatusRequest.attributeTypeMap = [
    {
        "name": "additionalSalesChannels",
        "baseName": "additionalSalesChannels",
        "type": "CalculatePciStatusRequest.AdditionalSalesChannelsEnum",
        "format": ""
    }
];
(function (CalculatePciStatusRequest) {
    let AdditionalSalesChannelsEnum;
    (function (AdditionalSalesChannelsEnum) {
        AdditionalSalesChannelsEnum["ECommerce"] = "eCommerce";
        AdditionalSalesChannelsEnum["EcomMoto"] = "ecomMoto";
        AdditionalSalesChannelsEnum["Pos"] = "pos";
        AdditionalSalesChannelsEnum["PosMoto"] = "posMoto";
    })(AdditionalSalesChannelsEnum = CalculatePciStatusRequest.AdditionalSalesChannelsEnum || (CalculatePciStatusRequest.AdditionalSalesChannelsEnum = {}));
})(CalculatePciStatusRequest = exports.CalculatePciStatusRequest || (exports.CalculatePciStatusRequest = {}));
