// @adyen/api-library/lib/src/typings/legalEntityManagement/getTermsOfServiceDocumentResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetTermsOfServiceDocumentResponse = void 0;
class GetTermsOfServiceDocumentResponse {
    static getAttributeTypeMap() {
        return GetTermsOfServiceDocumentResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.GetTermsOfServiceDocumentResponse = GetTermsOfServiceDocumentResponse;
GetTermsOfServiceDocumentResponse.discriminator = undefined;
GetTermsOfServiceDocumentResponse.mapping = undefined;
GetTermsOfServiceDocumentResponse.attributeTypeMap = [
    {
        "name": "document",
        "baseName": "document",
        "type": "string",
        "format": "byte"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "language",
        "baseName": "language",
        "type": "string",
        "format": ""
    },
    {
        "name": "termsOfServiceDocumentFormat",
        "baseName": "termsOfServiceDocumentFormat",
        "type": "string",
        "format": ""
    },
    {
        "name": "termsOfServiceDocumentId",
        "baseName": "termsOfServiceDocumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GetTermsOfServiceDocumentResponse.TypeEnum",
        "format": ""
    }
];
(function (GetTermsOfServiceDocumentResponse) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdyenAccount"] = "adyenAccount";
        TypeEnum["AdyenCapital"] = "adyenCapital";
        TypeEnum["AdyenCard"] = "adyenCard";
        TypeEnum["AdyenChargeCard"] = "adyenChargeCard";
        TypeEnum["AdyenForPlatformsAdvanced"] = "adyenForPlatformsAdvanced";
        TypeEnum["AdyenForPlatformsManage"] = "adyenForPlatformsManage";
        TypeEnum["AdyenFranchisee"] = "adyenFranchisee";
        TypeEnum["AdyenIssuing"] = "adyenIssuing";
        TypeEnum["AdyenPccr"] = "adyenPccr";
        TypeEnum["KycOnInvite"] = "kycOnInvite";
    })(TypeEnum = GetTermsOfServiceDocumentResponse.TypeEnum || (GetTermsOfServiceDocumentResponse.TypeEnum = {}));
})(GetTermsOfServiceDocumentResponse = exports.GetTermsOfServiceDocumentResponse || (exports.GetTermsOfServiceDocumentResponse = {}));
