// @adyen/api-library/lib/src/typings/legalEntityManagement/identificationData.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentificationData = void 0;
class IdentificationData {
    static getAttributeTypeMap() {
        return IdentificationData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.IdentificationData = IdentificationData;
IdentificationData.discriminator = undefined;
IdentificationData.mapping = undefined;
IdentificationData.attributeTypeMap = [
    {
        "name": "cardNumber",
        "baseName": "cardNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryDate",
        "baseName": "expiryDate",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuerState",
        "baseName": "issuerState",
        "type": "string",
        "format": ""
    },
    {
        "name": "nationalIdExempt",
        "baseName": "nationalIdExempt",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "IdentificationData.TypeEnum",
        "format": ""
    }
];
(function (IdentificationData) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["NationalIdNumber"] = "nationalIdNumber";
        TypeEnum["Passport"] = "passport";
        TypeEnum["DriversLicense"] = "driversLicense";
        TypeEnum["IdentityCard"] = "identityCard";
    })(TypeEnum = IdentificationData.TypeEnum || (IdentificationData.TypeEnum = {}));
})(IdentificationData = exports.IdentificationData || (exports.IdentificationData = {}));
