// @adyen/api-library/lib/src/typings/legalEntityManagement/legalEntityInfoRequiredType.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LegalEntityInfoRequiredType = void 0;
class LegalEntityInfoRequiredType {
    static getAttributeTypeMap() {
        return LegalEntityInfoRequiredType.attributeTypeMap;
    }
    constructor() {
    }
}
exports.LegalEntityInfoRequiredType = LegalEntityInfoRequiredType;
LegalEntityInfoRequiredType.discriminator = undefined;
LegalEntityInfoRequiredType.mapping = undefined;
LegalEntityInfoRequiredType.attributeTypeMap = [
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: LegalEntityCapability; }",
        "format": ""
    },
    {
        "name": "entityAssociations",
        "baseName": "entityAssociations",
        "type": "Array<LegalEntityAssociation>",
        "format": ""
    },
    {
        "name": "individual",
        "baseName": "individual",
        "type": "Individual | null",
        "format": ""
    },
    {
        "name": "organization",
        "baseName": "organization",
        "type": "Organization | null",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "soleProprietorship",
        "baseName": "soleProprietorship",
        "type": "SoleProprietorship | null",
        "format": ""
    },
    {
        "name": "trust",
        "baseName": "trust",
        "type": "Trust | null",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "LegalEntityInfoRequiredType.TypeEnum",
        "format": ""
    },
    {
        "name": "unincorporatedPartnership",
        "baseName": "unincorporatedPartnership",
        "type": "UnincorporatedPartnership | null",
        "format": ""
    },
    {
        "name": "verificationPlan",
        "baseName": "verificationPlan",
        "type": "string",
        "format": ""
    }
];
(function (LegalEntityInfoRequiredType) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Individual"] = "individual";
        TypeEnum["Organization"] = "organization";
        TypeEnum["SoleProprietorship"] = "soleProprietorship";
        TypeEnum["Trust"] = "trust";
        TypeEnum["UnincorporatedPartnership"] = "unincorporatedPartnership";
    })(TypeEnum = LegalEntityInfoRequiredType.TypeEnum || (LegalEntityInfoRequiredType.TypeEnum = {}));
})(LegalEntityInfoRequiredType = exports.LegalEntityInfoRequiredType || (exports.LegalEntityInfoRequiredType = {}));
