// @adyen/api-library/lib/src/typings/legalEntityManagement/soleProprietorship.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SoleProprietorship = void 0;
class SoleProprietorship {
    static getAttributeTypeMap() {
        return SoleProprietorship.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SoleProprietorship = SoleProprietorship;
SoleProprietorship.discriminator = undefined;
SoleProprietorship.mapping = undefined;
SoleProprietorship.attributeTypeMap = [
    {
        "name": "countryOfGoverningLaw",
        "baseName": "countryOfGoverningLaw",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfIncorporation",
        "baseName": "dateOfIncorporation",
        "type": "string",
        "format": ""
    },
    {
        "name": "doingBusinessAs",
        "baseName": "doingBusinessAs",
        "type": "string",
        "format": ""
    },
    {
        "name": "financialReports",
        "baseName": "financialReports",
        "type": "Array<FinancialReport>",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "principalPlaceOfBusiness",
        "baseName": "principalPlaceOfBusiness",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "registeredAddress",
        "baseName": "registeredAddress",
        "type": "Address",
        "format": ""
    },
    {
        "name": "registrationNumber",
        "baseName": "registrationNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "taxAbsent",
        "baseName": "taxAbsent",
        "type": "boolean | null",
        "format": ""
    },
    {
        "name": "taxInformation",
        "baseName": "taxInformation",
        "type": "Array<TaxInformation>",
        "format": ""
    },
    {
        "name": "vatAbsenceReason",
        "baseName": "vatAbsenceReason",
        "type": "SoleProprietorship.VatAbsenceReasonEnum",
        "format": ""
    },
    {
        "name": "vatNumber",
        "baseName": "vatNumber",
        "type": "string",
        "format": ""
    }
];
(function (SoleProprietorship) {
    let VatAbsenceReasonEnum;
    (function (VatAbsenceReasonEnum) {
        VatAbsenceReasonEnum["IndustryExemption"] = "industryExemption";
        VatAbsenceReasonEnum["BelowTaxThreshold"] = "belowTaxThreshold";
    })(VatAbsenceReasonEnum = SoleProprietorship.VatAbsenceReasonEnum || (SoleProprietorship.VatAbsenceReasonEnum = {}));
})(SoleProprietorship = exports.SoleProprietorship || (exports.SoleProprietorship = {}));
