// @adyen/api-library/lib/src/typings/legalEntityManagement/transferInstrument.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInstrument = void 0;
class TransferInstrument {
    static getAttributeTypeMap() {
        return TransferInstrument.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TransferInstrument = TransferInstrument;
TransferInstrument.discriminator = undefined;
TransferInstrument.mapping = undefined;
TransferInstrument.attributeTypeMap = [
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "BankAccountInfo",
        "format": ""
    },
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: SupportingEntityCapability; }",
        "format": ""
    },
    {
        "name": "documentDetails",
        "baseName": "documentDetails",
        "type": "Array<DocumentReference>",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string",
        "format": ""
    },
    {
        "name": "problems",
        "baseName": "problems",
        "type": "Array<CapabilityProblem>",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferInstrument.TypeEnum",
        "format": ""
    }
];
(function (TransferInstrument) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankAccount"] = "bankAccount";
        TypeEnum["RecurringDetail"] = "recurringDetail";
    })(TypeEnum = TransferInstrument.TypeEnum || (TransferInstrument.TypeEnum = {}));
})(TransferInstrument = exports.TransferInstrument || (exports.TransferInstrument = {}));
