// @adyen/api-library/lib/src/typings/legalEntityManagement/webDataExemption.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebDataExemption = void 0;
class WebDataExemption {
    static getAttributeTypeMap() {
        return WebDataExemption.attributeTypeMap;
    }
    constructor() {
    }
}
exports.WebDataExemption = WebDataExemption;
WebDataExemption.discriminator = undefined;
WebDataExemption.mapping = undefined;
WebDataExemption.attributeTypeMap = [
    {
        "name": "reason",
        "baseName": "reason",
        "type": "WebDataExemption.ReasonEnum",
        "format": ""
    }
];
(function (WebDataExemption) {
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["NoOnlinePresence"] = "noOnlinePresence";
        ReasonEnum["NotCollectedDuringOnboarding"] = "notCollectedDuringOnboarding";
    })(ReasonEnum = WebDataExemption.ReasonEnum || (WebDataExemption.ReasonEnum = {}));
})(WebDataExemption = exports.WebDataExemption || (exports.WebDataExemption = {}));
