// @adyen/api-library/lib/src/services/payment/paymentsApi.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentsApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/payment/objectSerializer");
/**
 * API handler for PaymentsApi
 */
class PaymentsApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/Payment/v68";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create an authorisation
    * @param paymentRequest {@link PaymentRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link PaymentResult }
    */
    async authorise(paymentRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/authorise`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(paymentRequest, "PaymentRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "PaymentResult");
    }
    /**
    * @summary Complete a 3DS authorisation
    * @param paymentRequest3d {@link PaymentRequest3d }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link PaymentResult }
    */
    async authorise3d(paymentRequest3d, requestOptions) {
        const endpoint = `${this.baseUrl}/authorise3d`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(paymentRequest3d, "PaymentRequest3d");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "PaymentResult");
    }
    /**
    * @summary Complete a 3DS2 authorisation
    * @param paymentRequest3ds2 {@link PaymentRequest3ds2 }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link PaymentResult }
    */
    async authorise3ds2(paymentRequest3ds2, requestOptions) {
        const endpoint = `${this.baseUrl}/authorise3ds2`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(paymentRequest3ds2, "PaymentRequest3ds2");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "PaymentResult");
    }
    /**
    * @summary Get the 3DS authentication result
    * @param authenticationResultRequest {@link AuthenticationResultRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link AuthenticationResultResponse }
    */
    async getAuthenticationResult(authenticationResultRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/getAuthenticationResult`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(authenticationResultRequest, "AuthenticationResultRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "AuthenticationResultResponse");
    }
    /**
    * @summary Get the 3DS2 authentication result
    * @param threeDS2ResultRequest {@link ThreeDS2ResultRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link ThreeDS2ResultResponse }
    */
    async retrieve3ds2Result(threeDS2ResultRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/retrieve3ds2Result`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(threeDS2ResultRequest, "ThreeDS2ResultRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "ThreeDS2ResultResponse");
    }
}
exports.PaymentsApi = PaymentsApi;
