// @adyen/api-library/lib/src/typings/configurationWebhooks/accountHolder.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolder = void 0;
class AccountHolder {
    static getAttributeTypeMap() {
        return AccountHolder.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AccountHolder = AccountHolder;
AccountHolder.discriminator = undefined;
AccountHolder.mapping = undefined;
AccountHolder.attributeTypeMap = [
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string",
        "format": ""
    },
    {
        "name": "capabilities",
        "baseName": "capabilities",
        "type": "{ [key: string]: AccountHolderCapability; }",
        "format": ""
    },
    {
        "name": "contactDetails",
        "baseName": "contactDetails",
        "type": "ContactDetails | null",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "legalEntityId",
        "baseName": "legalEntityId",
        "type": "string",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "migratedAccountHolderCode",
        "baseName": "migratedAccountHolderCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "primaryBalanceAccount",
        "baseName": "primaryBalanceAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AccountHolder.StatusEnum",
        "format": ""
    },
    {
        "name": "timeZone",
        "baseName": "timeZone",
        "type": "string",
        "format": ""
    },
    {
        "name": "verificationDeadlines",
        "baseName": "verificationDeadlines",
        "type": "Array<VerificationDeadline>",
        "format": ""
    }
];
(function (AccountHolder) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Suspended"] = "suspended";
    })(StatusEnum = AccountHolder.StatusEnum || (AccountHolder.StatusEnum = {}));
})(AccountHolder = exports.AccountHolder || (exports.AccountHolder = {}));
