// @adyen/api-library/lib/src/typings/configurationWebhooks/balanceAccountNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceAccountNotificationRequest = void 0;
class BalanceAccountNotificationRequest {
    static getAttributeTypeMap() {
        return BalanceAccountNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BalanceAccountNotificationRequest = BalanceAccountNotificationRequest;
BalanceAccountNotificationRequest.discriminator = undefined;
BalanceAccountNotificationRequest.mapping = undefined;
BalanceAccountNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "BalanceAccountNotificationData",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BalanceAccountNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (BalanceAccountNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformBalanceAccountUpdated"] = "balancePlatform.balanceAccount.updated";
        TypeEnum["BalancePlatformBalanceAccountCreated"] = "balancePlatform.balanceAccount.created";
    })(TypeEnum = BalanceAccountNotificationRequest.TypeEnum || (BalanceAccountNotificationRequest.TypeEnum = {}));
})(BalanceAccountNotificationRequest = exports.BalanceAccountNotificationRequest || (exports.BalanceAccountNotificationRequest = {}));
