// @adyen/api-library/lib/src/typings/configurationWebhooks/cardOrderNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardOrderNotificationRequest = void 0;
class CardOrderNotificationRequest {
    static getAttributeTypeMap() {
        return CardOrderNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CardOrderNotificationRequest = CardOrderNotificationRequest;
CardOrderNotificationRequest.discriminator = undefined;
CardOrderNotificationRequest.mapping = undefined;
CardOrderNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "CardOrderItem",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CardOrderNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (CardOrderNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformCardorderCreated"] = "balancePlatform.cardorder.created";
        TypeEnum["BalancePlatformCardorderUpdated"] = "balancePlatform.cardorder.updated";
    })(TypeEnum = CardOrderNotificationRequest.TypeEnum || (CardOrderNotificationRequest.TypeEnum = {}));
})(CardOrderNotificationRequest = exports.CardOrderNotificationRequest || (exports.CardOrderNotificationRequest = {}));
