// @adyen/api-library/lib/src/typings/configurationWebhooks/networkTokenNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkTokenNotificationRequest = void 0;
class NetworkTokenNotificationRequest {
    static getAttributeTypeMap() {
        return NetworkTokenNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.NetworkTokenNotificationRequest = NetworkTokenNotificationRequest;
NetworkTokenNotificationRequest.discriminator = undefined;
NetworkTokenNotificationRequest.mapping = undefined;
NetworkTokenNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "NetworkTokenNotificationDataV2",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "NetworkTokenNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (NetworkTokenNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformNetworkTokenCreated"] = "balancePlatform.networkToken.created";
        TypeEnum["BalancePlatformNetworkTokenUpdated"] = "balancePlatform.networkToken.updated";
    })(TypeEnum = NetworkTokenNotificationRequest.TypeEnum || (NetworkTokenNotificationRequest.TypeEnum = {}));
})(NetworkTokenNotificationRequest = exports.NetworkTokenNotificationRequest || (exports.NetworkTokenNotificationRequest = {}));
