// @adyen/api-library/lib/src/typings/configurationWebhooks/paymentNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentNotificationRequest = void 0;
class PaymentNotificationRequest {
    static getAttributeTypeMap() {
        return PaymentNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentNotificationRequest = PaymentNotificationRequest;
PaymentNotificationRequest.discriminator = undefined;
PaymentNotificationRequest.mapping = undefined;
PaymentNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "PaymentInstrumentNotificationData",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (PaymentNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformPaymentInstrumentCreated"] = "balancePlatform.paymentInstrument.created";
        TypeEnum["BalancePlatformPaymentInstrumentUpdated"] = "balancePlatform.paymentInstrument.updated";
    })(TypeEnum = PaymentNotificationRequest.TypeEnum || (PaymentNotificationRequest.TypeEnum = {}));
})(PaymentNotificationRequest = exports.PaymentNotificationRequest || (exports.PaymentNotificationRequest = {}));
