// @adyen/api-library/lib/src/typings/configurationWebhooks/phoneNumber.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumber = void 0;
class PhoneNumber {
    static getAttributeTypeMap() {
        return PhoneNumber.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PhoneNumber = PhoneNumber;
PhoneNumber.discriminator = undefined;
PhoneNumber.mapping = undefined;
PhoneNumber.attributeTypeMap = [
    {
        "name": "phoneCountryCode",
        "baseName": "phoneCountryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "phoneType",
        "baseName": "phoneType",
        "type": "PhoneNumber.PhoneTypeEnum",
        "format": ""
    }
];
(function (PhoneNumber) {
    let PhoneTypeEnum;
    (function (PhoneTypeEnum) {
        PhoneTypeEnum["Fax"] = "Fax";
        PhoneTypeEnum["Landline"] = "Landline";
        PhoneTypeEnum["Mobile"] = "Mobile";
        PhoneTypeEnum["Sip"] = "SIP";
    })(PhoneTypeEnum = PhoneNumber.PhoneTypeEnum || (PhoneNumber.PhoneTypeEnum = {}));
})(PhoneNumber = exports.PhoneNumber || (exports.PhoneNumber = {}));
