// @adyen/api-library/lib/src/typings/configurationWebhooks/phone.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Phone = void 0;
class Phone {
    static getAttributeTypeMap() {
        return Phone.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Phone = Phone;
Phone.discriminator = undefined;
Phone.mapping = undefined;
Phone.attributeTypeMap = [
    {
        "name": "number",
        "baseName": "number",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Phone.TypeEnum",
        "format": ""
    }
];
(function (Phone) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Landline"] = "landline";
        TypeEnum["Mobile"] = "mobile";
    })(TypeEnum = Phone.TypeEnum || (Phone.TypeEnum = {}));
})(Phone = exports.Phone || (exports.Phone = {}));
