// @adyen/api-library/lib/src/typings/configurationWebhooks/sweepConfigurationNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepConfigurationNotificationRequest = void 0;
class SweepConfigurationNotificationRequest {
    static getAttributeTypeMap() {
        return SweepConfigurationNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SweepConfigurationNotificationRequest = SweepConfigurationNotificationRequest;
SweepConfigurationNotificationRequest.discriminator = undefined;
SweepConfigurationNotificationRequest.mapping = undefined;
SweepConfigurationNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "SweepConfigurationNotificationData",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepConfigurationNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (SweepConfigurationNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformBalanceAccountSweepCreated"] = "balancePlatform.balanceAccountSweep.created";
        TypeEnum["BalancePlatformBalanceAccountSweepUpdated"] = "balancePlatform.balanceAccountSweep.updated";
        TypeEnum["BalancePlatformBalanceAccountSweepDeleted"] = "balancePlatform.balanceAccountSweep.deleted";
    })(TypeEnum = SweepConfigurationNotificationRequest.TypeEnum || (SweepConfigurationNotificationRequest.TypeEnum = {}));
})(SweepConfigurationNotificationRequest = exports.SweepConfigurationNotificationRequest || (exports.SweepConfigurationNotificationRequest = {}));
