// @adyen/api-library/lib/src/typings/configurationWebhooks/validationFacts.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationFacts = void 0;
class ValidationFacts {
    static getAttributeTypeMap() {
        return ValidationFacts.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ValidationFacts = ValidationFacts;
ValidationFacts.discriminator = undefined;
ValidationFacts.mapping = undefined;
ValidationFacts.attributeTypeMap = [
    {
        "name": "reasons",
        "baseName": "reasons",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "result",
        "baseName": "result",
        "type": "ValidationFacts.ResultEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string",
        "format": ""
    }
];
(function (ValidationFacts) {
    let ResultEnum;
    (function (ResultEnum) {
        ResultEnum["Invalid"] = "invalid";
        ResultEnum["NotApplicable"] = "notApplicable";
        ResultEnum["NotValidated"] = "notValidated";
        ResultEnum["Valid"] = "valid";
    })(ResultEnum = ValidationFacts.ResultEnum || (ValidationFacts.ResultEnum = {}));
})(ValidationFacts = exports.ValidationFacts || (exports.ValidationFacts = {}));
