// @adyen/api-library/lib/src/typings/configurationWebhooks/wallet.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Wallet = void 0;
class Wallet {
    static getAttributeTypeMap() {
        return Wallet.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Wallet = Wallet;
Wallet.discriminator = undefined;
Wallet.mapping = undefined;
Wallet.attributeTypeMap = [
    {
        "name": "accountScore",
        "baseName": "accountScore",
        "type": "string",
        "format": ""
    },
    {
        "name": "device",
        "baseName": "device",
        "type": "Device | null",
        "format": ""
    },
    {
        "name": "deviceScore",
        "baseName": "deviceScore",
        "type": "string",
        "format": ""
    },
    {
        "name": "provisioningMethod",
        "baseName": "provisioningMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "recommendationReasons",
        "baseName": "recommendationReasons",
        "type": "Wallet.RecommendationReasonsEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string",
        "format": ""
    }
];
(function (Wallet) {
    let RecommendationReasonsEnum;
    (function (RecommendationReasonsEnum) {
        RecommendationReasonsEnum["AccountCardTooNew"] = "accountCardTooNew";
        RecommendationReasonsEnum["AccountHighRisk"] = "accountHighRisk";
        RecommendationReasonsEnum["AccountRecentlyChanged"] = "accountRecentlyChanged";
        RecommendationReasonsEnum["AccountTooNew"] = "accountTooNew";
        RecommendationReasonsEnum["AccountTooNewSinceLaunch"] = "accountTooNewSinceLaunch";
        RecommendationReasonsEnum["CardholderPanAssociatedToAccountWithinThresholdDays"] = "cardholderPanAssociatedToAccountWithinThresholdDays";
        RecommendationReasonsEnum["ChangesMadeToAccountDataWithinThresholdDays"] = "changesMadeToAccountDataWithinThresholdDays";
        RecommendationReasonsEnum["DeviceProvisioningLocationOutsideOfCardholdersWalletAccountHomeCountry"] = "deviceProvisioningLocationOutsideOfCardholdersWalletAccountHomeCountry";
        RecommendationReasonsEnum["DeviceRecentlyLost"] = "deviceRecentlyLost";
        RecommendationReasonsEnum["EncryptedPaymentInstrumentDataIsBeingPushedByTheIssuerToTheSameDeviceThatIssuerApplicationAuthenticatedButWithSuccessfulUpfrontAuthentication"] = "encryptedPaymentInstrumentDataIsBeingPushedByTheIssuerToTheSameDeviceThatIssuerApplicationAuthenticatedButWithSuccessfulUpfrontAuthentication";
        RecommendationReasonsEnum["EncryptedPaymentInstrumentDataIsBeingPushedByTheIssuerToTheSameDeviceThatIssuerApplicationAuthenticatedButWithoutAnyUpfrontAuthentication"] = "encryptedPaymentInstrumentDataIsBeingPushedByTheIssuerToTheSameDeviceThatIssuerApplicationAuthenticatedButWithoutAnyUpfrontAuthentication";
        RecommendationReasonsEnum["EncryptedPaymentInstrumentDataIsPushedToADifferentDeviceThanTheOneThatIssuerApplicationAuthenticated"] = "encryptedPaymentInstrumentDataIsPushedToADifferentDeviceThanTheOneThatIssuerApplicationAuthenticated";
        RecommendationReasonsEnum["EncryptedPaymentInstrumentDataIsPushedToADifferentUserThanTheCardHolder"] = "encryptedPaymentInstrumentDataIsPushedToADifferentUserThanTheCardHolder";
        RecommendationReasonsEnum["HasSuspendedTokens"] = "hasSuspendedTokens";
        RecommendationReasonsEnum["InactiveAccount"] = "inactiveAccount";
        RecommendationReasonsEnum["IssuerDeferredIdvDecision"] = "issuerDeferredIDVDecision";
        RecommendationReasonsEnum["IssuerEncryptedPaymentInstrumentDataExpired"] = "issuerEncryptedPaymentInstrumentDataExpired";
        RecommendationReasonsEnum["LowAccountScore"] = "lowAccountScore";
        RecommendationReasonsEnum["LowDeviceScore"] = "lowDeviceScore";
        RecommendationReasonsEnum["LowPhoneNumberScore"] = "lowPhoneNumberScore";
        RecommendationReasonsEnum["NumberOfActiveTokensGreaterThanThreshold"] = "numberOfActiveTokensGreaterThanThreshold";
        RecommendationReasonsEnum["NumberOfActiveTokensOnAllDevicesIsGreaterThanThreshold"] = "numberOfActiveTokensOnAllDevicesIsGreaterThanThreshold";
        RecommendationReasonsEnum["NumberOfDaysSinceDeviceWasLastReportedLostIsLessThanThresholdDays"] = "numberOfDaysSinceDeviceWasLastReportedLostIsLessThanThresholdDays";
        RecommendationReasonsEnum["NumberOfDevicesWithSameUseridWithTokenIsGreaterThanThreshold"] = "numberOfDevicesWithSameUseridWithTokenIsGreaterThanThreshold";
        RecommendationReasonsEnum["NumberOfTransactionsInLast12MonthsLessThanThresholdNumber"] = "numberOfTransactionsInLast12MonthsLessThanThresholdNumber";
        RecommendationReasonsEnum["OutSideHomeTerritory"] = "outSideHomeTerritory";
        RecommendationReasonsEnum["SuspendedCardsInTheWalletAccountIsGreaterThanThreshold"] = "suspendedCardsInTheWALLETAccountIsGreaterThanThreshold";
        RecommendationReasonsEnum["SuspiciousActivity"] = "suspiciousActivity";
        RecommendationReasonsEnum["TheNumberOfProvisioningAttemptsAcrossAllCardsOnThisDeviceInTheLast24HoursExceedsTheThreshold"] = "theNumberOfProvisioningAttemptsAcrossAllCardsOnThisDeviceInTheLast24HoursExceedsTheThreshold";
        RecommendationReasonsEnum["TheWalletAccountIntoWhichTheCardIsBeingProvisionedContainDistinctNamesGreaterThanThreshold"] = "theWALLETAccountIntoWhichTheCardIsBeingProvisionedContainDistinctNamesGreaterThanThreshold";
        RecommendationReasonsEnum["ThisAccountHasNotHadActivityWithinThresholdPeriod"] = "thisAccountHasNotHadActivityWithinThresholdPeriod";
        RecommendationReasonsEnum["TooManyDifferentCardholders"] = "tooManyDifferentCardholders";
        RecommendationReasonsEnum["TooManyRecentAttempts"] = "tooManyRecentAttempts";
        RecommendationReasonsEnum["TooManyRecentTokens"] = "tooManyRecentTokens";
        RecommendationReasonsEnum["UnableToAssess"] = "unableToAssess";
        RecommendationReasonsEnum["Unknown"] = "unknown";
        RecommendationReasonsEnum["UserAccountWasCreatedWithinThresholdDays"] = "userAccountWasCreatedWithinThresholdDays";
        RecommendationReasonsEnum["UserDeviceReceivingEncryptedPaymentInstrumentDataIsDifferentThanTheOneThatIsProvisioningTheToken"] = "userDeviceReceivingEncryptedPaymentInstrumentDataIsDifferentThanTheOneThatIsProvisioningTheToken";
        RecommendationReasonsEnum["UsersAccountOnDeviceLessThanThresholdDays"] = "usersAccountOnDeviceLessThanThresholdDays";
        RecommendationReasonsEnum["WalletAccountCreatedWithinThresholdDays"] = "walletAccountCreatedWithinThresholdDays";
        RecommendationReasonsEnum["WalletAccountHolderNameOnFileDoesNotMatchCardholderEnteredName"] = "walletAccountHolderNameOnFileDoesNotMatchCardholderEnteredName";
    })(RecommendationReasonsEnum = Wallet.RecommendationReasonsEnum || (Wallet.RecommendationReasonsEnum = {}));
})(Wallet = exports.Wallet || (exports.Wallet = {}));
