// @adyen/api-library/lib/src/typings/tokenizationWebhooks/tokenizationUpdatedDetailsNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenizationUpdatedDetailsNotificationRequest = void 0;
class TokenizationUpdatedDetailsNotificationRequest {
    static getAttributeTypeMap() {
        return TokenizationUpdatedDetailsNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TokenizationUpdatedDetailsNotificationRequest = TokenizationUpdatedDetailsNotificationRequest;
TokenizationUpdatedDetailsNotificationRequest.discriminator = undefined;
TokenizationUpdatedDetailsNotificationRequest.mapping = undefined;
TokenizationUpdatedDetailsNotificationRequest.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "RecurringTokenStoreOperation",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "TokenizationUpdatedDetailsNotificationRequest.EnvironmentEnum",
        "format": ""
    },
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TokenizationUpdatedDetailsNotificationRequest.TypeEnum",
        "format": ""
    },
    {
        "name": "version",
        "baseName": "version",
        "type": "string",
        "format": ""
    }
];
(function (TokenizationUpdatedDetailsNotificationRequest) {
    let EnvironmentEnum;
    (function (EnvironmentEnum) {
        EnvironmentEnum["Test"] = "test";
        EnvironmentEnum["Live"] = "live";
    })(EnvironmentEnum = TokenizationUpdatedDetailsNotificationRequest.EnvironmentEnum || (TokenizationUpdatedDetailsNotificationRequest.EnvironmentEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["RecurringTokenUpdated"] = "recurring.token.updated";
    })(TypeEnum = TokenizationUpdatedDetailsNotificationRequest.TypeEnum || (TokenizationUpdatedDetailsNotificationRequest.TypeEnum = {}));
})(TokenizationUpdatedDetailsNotificationRequest = exports.TokenizationUpdatedDetailsNotificationRequest || (exports.TokenizationUpdatedDetailsNotificationRequest = {}));
