// @adyen/api-library/lib/src/typings/tokenizationWebhooks/tokenizationDisabledDetailsNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenizationDisabledDetailsNotificationRequest = void 0;
class TokenizationDisabledDetailsNotificationRequest {
    static getAttributeTypeMap() {
        return TokenizationDisabledDetailsNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TokenizationDisabledDetailsNotificationRequest = TokenizationDisabledDetailsNotificationRequest;
TokenizationDisabledDetailsNotificationRequest.discriminator = undefined;
TokenizationDisabledDetailsNotificationRequest.mapping = undefined;
TokenizationDisabledDetailsNotificationRequest.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "RecurringToken",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "TokenizationDisabledDetailsNotificationRequest.EnvironmentEnum",
        "format": ""
    },
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TokenizationDisabledDetailsNotificationRequest.TypeEnum",
        "format": ""
    },
    {
        "name": "version",
        "baseName": "version",
        "type": "string",
        "format": ""
    }
];
(function (TokenizationDisabledDetailsNotificationRequest) {
    let EnvironmentEnum;
    (function (EnvironmentEnum) {
        EnvironmentEnum["Test"] = "test";
        EnvironmentEnum["Live"] = "live";
    })(EnvironmentEnum = TokenizationDisabledDetailsNotificationRequest.EnvironmentEnum || (TokenizationDisabledDetailsNotificationRequest.EnvironmentEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["RecurringTokenDisabled"] = "recurring.token.disabled";
    })(TypeEnum = TokenizationDisabledDetailsNotificationRequest.TypeEnum || (TokenizationDisabledDetailsNotificationRequest.TypeEnum = {}));
})(TokenizationDisabledDetailsNotificationRequest = exports.TokenizationDisabledDetailsNotificationRequest || (exports.TokenizationDisabledDetailsNotificationRequest = {}));
