// @adyen/api-library/lib/src/typings/tokenizationWebhooks/tokenizationWebhooksHandler.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenizationWebhooksHandler = void 0;
const __1 = require("..");
/**
 * Handler for processing TokenizationWebhooks.
 *
 * This class provides functionality to deserialize the payload of TokenizationWebhooks events.
 */
class TokenizationWebhooksHandler {
    constructor(jsonPayload) {
        this.payload = JSON.parse(jsonPayload);
    }
    /**
     * This method checks the type of the webhook payload and returns the appropriate deserialized object.
     *
     * @returns A deserialized object of type GenericWebhook.
     * @throws Error if the type is not recognized.
     */
    getGenericWebhook() {
        const type = this.payload["type"];
        if (Object.values(__1.tokenizationWebhooks.TokenizationAlreadyExistingDetailsNotificationRequest.TypeEnum).includes(type)) {
            return this.getTokenizationAlreadyExistingDetailsNotificationRequest();
        }
        if (Object.values(__1.tokenizationWebhooks.TokenizationCreatedDetailsNotificationRequest.TypeEnum).includes(type)) {
            return this.getTokenizationCreatedDetailsNotificationRequest();
        }
        if (Object.values(__1.tokenizationWebhooks.TokenizationDisabledDetailsNotificationRequest.TypeEnum).includes(type)) {
            return this.getTokenizationDisabledDetailsNotificationRequest();
        }
        if (Object.values(__1.tokenizationWebhooks.TokenizationUpdatedDetailsNotificationRequest.TypeEnum).includes(type)) {
            return this.getTokenizationUpdatedDetailsNotificationRequest();
        }
        throw new Error("Could not parse the json payload: " + this.payload);
    }
    /**
     * Deserialize the webhook payload into a TokenizationAlreadyExistingDetailsNotificationRequest
     *
     * @returns Deserialized TokenizationAlreadyExistingDetailsNotificationRequest object.
     */
    getTokenizationAlreadyExistingDetailsNotificationRequest() {
        return __1.tokenizationWebhooks.ObjectSerializer.deserialize(this.payload, "TokenizationAlreadyExistingDetailsNotificationRequest");
    }
    /**
     * Deserialize the webhook payload into a TokenizationCreatedDetailsNotificationRequest
     *
     * @returns Deserialized TokenizationCreatedDetailsNotificationRequest object.
     */
    getTokenizationCreatedDetailsNotificationRequest() {
        return __1.tokenizationWebhooks.ObjectSerializer.deserialize(this.payload, "TokenizationCreatedDetailsNotificationRequest");
    }
    /**
     * Deserialize the webhook payload into a TokenizationDisabledDetailsNotificationRequest
     *
     * @returns Deserialized TokenizationDisabledDetailsNotificationRequest object.
     */
    getTokenizationDisabledDetailsNotificationRequest() {
        return __1.tokenizationWebhooks.ObjectSerializer.deserialize(this.payload, "TokenizationDisabledDetailsNotificationRequest");
    }
    /**
     * Deserialize the webhook payload into a TokenizationUpdatedDetailsNotificationRequest
     *
     * @returns Deserialized TokenizationUpdatedDetailsNotificationRequest object.
     */
    getTokenizationUpdatedDetailsNotificationRequest() {
        return __1.tokenizationWebhooks.ObjectSerializer.deserialize(this.payload, "TokenizationUpdatedDetailsNotificationRequest");
    }
}
exports.TokenizationWebhooksHandler = TokenizationWebhooksHandler;
