// @adyen/api-library/lib/src/typings/tokenizationWebhooks/tokenizationAlreadyExistingDetailsNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenizationAlreadyExistingDetailsNotificationRequest = void 0;
class TokenizationAlreadyExistingDetailsNotificationRequest {
    static getAttributeTypeMap() {
        return TokenizationAlreadyExistingDetailsNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TokenizationAlreadyExistingDetailsNotificationRequest = TokenizationAlreadyExistingDetailsNotificationRequest;
TokenizationAlreadyExistingDetailsNotificationRequest.discriminator = undefined;
TokenizationAlreadyExistingDetailsNotificationRequest.mapping = undefined;
TokenizationAlreadyExistingDetailsNotificationRequest.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "RecurringTokenStoreOperation",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "TokenizationAlreadyExistingDetailsNotificationRequest.EnvironmentEnum",
        "format": ""
    },
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TokenizationAlreadyExistingDetailsNotificationRequest.TypeEnum",
        "format": ""
    },
    {
        "name": "version",
        "baseName": "version",
        "type": "string",
        "format": ""
    }
];
(function (TokenizationAlreadyExistingDetailsNotificationRequest) {
    let EnvironmentEnum;
    (function (EnvironmentEnum) {
        EnvironmentEnum["Test"] = "test";
        EnvironmentEnum["Live"] = "live";
    })(EnvironmentEnum = TokenizationAlreadyExistingDetailsNotificationRequest.EnvironmentEnum || (TokenizationAlreadyExistingDetailsNotificationRequest.EnvironmentEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["RecurringTokenAlreadyExisting"] = "recurring.token.alreadyExisting";
    })(TypeEnum = TokenizationAlreadyExistingDetailsNotificationRequest.TypeEnum || (TokenizationAlreadyExistingDetailsNotificationRequest.TypeEnum = {}));
})(TokenizationAlreadyExistingDetailsNotificationRequest = exports.TokenizationAlreadyExistingDetailsNotificationRequest || (exports.TokenizationAlreadyExistingDetailsNotificationRequest = {}));
