// @adyen/api-library/lib/src/typings/tokenizationWebhooks/tokenizationCreatedDetailsNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenizationCreatedDetailsNotificationRequest = void 0;
class TokenizationCreatedDetailsNotificationRequest {
    static getAttributeTypeMap() {
        return TokenizationCreatedDetailsNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TokenizationCreatedDetailsNotificationRequest = TokenizationCreatedDetailsNotificationRequest;
TokenizationCreatedDetailsNotificationRequest.discriminator = undefined;
TokenizationCreatedDetailsNotificationRequest.mapping = undefined;
TokenizationCreatedDetailsNotificationRequest.attributeTypeMap = [
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "data",
        "baseName": "data",
        "type": "RecurringTokenStoreOperation",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "TokenizationCreatedDetailsNotificationRequest.EnvironmentEnum",
        "format": ""
    },
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TokenizationCreatedDetailsNotificationRequest.TypeEnum",
        "format": ""
    },
    {
        "name": "version",
        "baseName": "version",
        "type": "string",
        "format": ""
    }
];
(function (TokenizationCreatedDetailsNotificationRequest) {
    let EnvironmentEnum;
    (function (EnvironmentEnum) {
        EnvironmentEnum["Test"] = "test";
        EnvironmentEnum["Live"] = "live";
    })(EnvironmentEnum = TokenizationCreatedDetailsNotificationRequest.EnvironmentEnum || (TokenizationCreatedDetailsNotificationRequest.EnvironmentEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["RecurringTokenCreated"] = "recurring.token.created";
    })(TypeEnum = TokenizationCreatedDetailsNotificationRequest.TypeEnum || (TokenizationCreatedDetailsNotificationRequest.TypeEnum = {}));
})(TokenizationCreatedDetailsNotificationRequest = exports.TokenizationCreatedDetailsNotificationRequest || (exports.TokenizationCreatedDetailsNotificationRequest = {}));
