// @adyen/api-library/lib/src/typings/checkout/afterpayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AfterpayDetails = void 0;
class AfterpayDetails {
    static getAttributeTypeMap() {
        return AfterpayDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AfterpayDetails = AfterpayDetails;
AfterpayDetails.discriminator = undefined;
AfterpayDetails.mapping = undefined;
AfterpayDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AfterpayDetails.TypeEnum",
        "format": ""
    }
];
(function (AfterpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AfterpayDefault"] = "afterpay_default";
        TypeEnum["Afterpaytouch"] = "afterpaytouch";
        TypeEnum["AfterpayB2b"] = "afterpay_b2b";
        TypeEnum["Clearpay"] = "clearpay";
    })(TypeEnum = AfterpayDetails.TypeEnum || (AfterpayDetails.TypeEnum = {}));
})(AfterpayDetails = exports.AfterpayDetails || (exports.AfterpayDetails = {}));
