// @adyen/api-library/lib/src/typings/checkout/applePayDonations.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplePayDonations = void 0;
class ApplePayDonations {
    static getAttributeTypeMap() {
        return ApplePayDonations.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ApplePayDonations = ApplePayDonations;
ApplePayDonations.discriminator = undefined;
ApplePayDonations.mapping = undefined;
ApplePayDonations.attributeTypeMap = [
    {
        "name": "applePayToken",
        "baseName": "applePayToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "ApplePayDonations.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "ApplePayDonations.TypeEnum",
        "format": ""
    }
];
(function (ApplePayDonations) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = ApplePayDonations.FundingSourceEnum || (ApplePayDonations.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Applepay"] = "applepay";
    })(TypeEnum = ApplePayDonations.TypeEnum || (ApplePayDonations.TypeEnum = {}));
})(ApplePayDonations = exports.ApplePayDonations || (exports.ApplePayDonations = {}));
