// @adyen/api-library/lib/src/typings/checkout/authenticationData.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationData = void 0;
class AuthenticationData {
    static getAttributeTypeMap() {
        return AuthenticationData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AuthenticationData = AuthenticationData;
AuthenticationData.discriminator = undefined;
AuthenticationData.mapping = undefined;
AuthenticationData.attributeTypeMap = [
    {
        "name": "attemptAuthentication",
        "baseName": "attemptAuthentication",
        "type": "AuthenticationData.AttemptAuthenticationEnum",
        "format": ""
    },
    {
        "name": "authenticationOnly",
        "baseName": "authenticationOnly",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "threeDSRequestData",
        "baseName": "threeDSRequestData",
        "type": "ThreeDSRequestData | null",
        "format": ""
    }
];
(function (AuthenticationData) {
    let AttemptAuthenticationEnum;
    (function (AttemptAuthenticationEnum) {
        AttemptAuthenticationEnum["Always"] = "always";
        AttemptAuthenticationEnum["Never"] = "never";
    })(AttemptAuthenticationEnum = AuthenticationData.AttemptAuthenticationEnum || (AuthenticationData.AttemptAuthenticationEnum = {}));
})(AuthenticationData = exports.AuthenticationData || (exports.AuthenticationData = {}));
