// @adyen/api-library/lib/src/typings/checkout/cardDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardDetails = void 0;
class CardDetails {
    static getAttributeTypeMap() {
        return CardDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CardDetails = CardDetails;
CardDetails.discriminator = undefined;
CardDetails.mapping = undefined;
CardDetails.attributeTypeMap = [
    {
        "name": "brand",
        "baseName": "brand",
        "type": "string",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "cupsecureplus_smscode",
        "baseName": "cupsecureplus.smscode",
        "type": "string",
        "format": ""
    },
    {
        "name": "cvc",
        "baseName": "cvc",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedCard",
        "baseName": "encryptedCard",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedCardNumber",
        "baseName": "encryptedCardNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedExpiryMonth",
        "baseName": "encryptedExpiryMonth",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedExpiryYear",
        "baseName": "encryptedExpiryYear",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedSecurityCode",
        "baseName": "encryptedSecurityCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryMonth",
        "baseName": "expiryMonth",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryYear",
        "baseName": "expiryYear",
        "type": "string",
        "format": ""
    },
    {
        "name": "fastlaneData",
        "baseName": "fastlaneData",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "CardDetails.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "holderName",
        "baseName": "holderName",
        "type": "string",
        "format": ""
    },
    {
        "name": "networkPaymentReference",
        "baseName": "networkPaymentReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "number",
        "baseName": "number",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperNotificationReference",
        "baseName": "shopperNotificationReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "srcCorrelationId",
        "baseName": "srcCorrelationId",
        "type": "string",
        "format": ""
    },
    {
        "name": "srcDigitalCardId",
        "baseName": "srcDigitalCardId",
        "type": "string",
        "format": ""
    },
    {
        "name": "srcScheme",
        "baseName": "srcScheme",
        "type": "string",
        "format": ""
    },
    {
        "name": "srcTokenReference",
        "baseName": "srcTokenReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2SdkVersion",
        "baseName": "threeDS2SdkVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CardDetails.TypeEnum",
        "format": ""
    }
];
(function (CardDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = CardDetails.FundingSourceEnum || (CardDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Bcmc"] = "bcmc";
        TypeEnum["Scheme"] = "scheme";
        TypeEnum["NetworkToken"] = "networkToken";
        TypeEnum["Giftcard"] = "giftcard";
        TypeEnum["Card"] = "card";
        TypeEnum["Clicktopay"] = "clicktopay";
    })(TypeEnum = CardDetails.TypeEnum || (CardDetails.TypeEnum = {}));
})(CardDetails = exports.CardDetails || (exports.CardDetails = {}));
