// @adyen/api-library/lib/src/typings/checkout/checkoutBankAccount.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutBankAccount = void 0;
class CheckoutBankAccount {
    static getAttributeTypeMap() {
        return CheckoutBankAccount.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CheckoutBankAccount = CheckoutBankAccount;
CheckoutBankAccount.discriminator = undefined;
CheckoutBankAccount.mapping = undefined;
CheckoutBankAccount.attributeTypeMap = [
    {
        "name": "accountType",
        "baseName": "accountType",
        "type": "CheckoutBankAccount.AccountTypeEnum",
        "format": ""
    },
    {
        "name": "bankAccountNumber",
        "baseName": "bankAccountNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "bankCity",
        "baseName": "bankCity",
        "type": "string",
        "format": ""
    },
    {
        "name": "bankLocationId",
        "baseName": "bankLocationId",
        "type": "string",
        "format": ""
    },
    {
        "name": "bankName",
        "baseName": "bankName",
        "type": "string",
        "format": ""
    },
    {
        "name": "bic",
        "baseName": "bic",
        "type": "string",
        "format": ""
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "iban",
        "baseName": "iban",
        "type": "string",
        "format": ""
    },
    {
        "name": "ownerName",
        "baseName": "ownerName",
        "type": "string",
        "format": ""
    },
    {
        "name": "taxId",
        "baseName": "taxId",
        "type": "string",
        "format": ""
    }
];
(function (CheckoutBankAccount) {
    let AccountTypeEnum;
    (function (AccountTypeEnum) {
        AccountTypeEnum["Balance"] = "balance";
        AccountTypeEnum["Checking"] = "checking";
        AccountTypeEnum["Deposit"] = "deposit";
        AccountTypeEnum["General"] = "general";
        AccountTypeEnum["Other"] = "other";
        AccountTypeEnum["Payment"] = "payment";
        AccountTypeEnum["Savings"] = "savings";
    })(AccountTypeEnum = CheckoutBankAccount.AccountTypeEnum || (CheckoutBankAccount.AccountTypeEnum = {}));
})(CheckoutBankAccount = exports.CheckoutBankAccount || (exports.CheckoutBankAccount = {}));
