// @adyen/api-library/lib/src/typings/checkout/checkoutSessionInstallmentOption.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSessionInstallmentOption = void 0;
class CheckoutSessionInstallmentOption {
    static getAttributeTypeMap() {
        return CheckoutSessionInstallmentOption.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CheckoutSessionInstallmentOption = CheckoutSessionInstallmentOption;
CheckoutSessionInstallmentOption.discriminator = undefined;
CheckoutSessionInstallmentOption.mapping = undefined;
CheckoutSessionInstallmentOption.attributeTypeMap = [
    {
        "name": "plans",
        "baseName": "plans",
        "type": "CheckoutSessionInstallmentOption.PlansEnum",
        "format": ""
    },
    {
        "name": "preselectedValue",
        "baseName": "preselectedValue",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "values",
        "baseName": "values",
        "type": "Array<number>",
        "format": "int32"
    }
];
(function (CheckoutSessionInstallmentOption) {
    let PlansEnum;
    (function (PlansEnum) {
        PlansEnum["Bonus"] = "bonus";
        PlansEnum["BuynowPaylater"] = "buynow_paylater";
        PlansEnum["InteresRefundPrctg"] = "interes_refund_prctg";
        PlansEnum["InterestBonus"] = "interest_bonus";
        PlansEnum["NointeresRefundPrctg"] = "nointeres_refund_prctg";
        PlansEnum["NointerestBonus"] = "nointerest_bonus";
        PlansEnum["RefundPrctg"] = "refund_prctg";
        PlansEnum["Regular"] = "regular";
        PlansEnum["Revolving"] = "revolving";
        PlansEnum["WithInterest"] = "with_interest";
    })(PlansEnum = CheckoutSessionInstallmentOption.PlansEnum || (CheckoutSessionInstallmentOption.PlansEnum = {}));
})(CheckoutSessionInstallmentOption = exports.CheckoutSessionInstallmentOption || (exports.CheckoutSessionInstallmentOption = {}));
