// @adyen/api-library/lib/src/typings/checkout/checkoutSDKAction.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSDKAction = void 0;
class CheckoutSDKAction {
    static getAttributeTypeMap() {
        return CheckoutSDKAction.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CheckoutSDKAction = CheckoutSDKAction;
CheckoutSDKAction.discriminator = undefined;
CheckoutSDKAction.mapping = undefined;
CheckoutSDKAction.attributeTypeMap = [
    {
        "name": "paymentData",
        "baseName": "paymentData",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethodType",
        "baseName": "paymentMethodType",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkData",
        "baseName": "sdkData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CheckoutSDKAction.TypeEnum",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    }
];
(function (CheckoutSDKAction) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Sdk"] = "sdk";
        TypeEnum["WechatpaySdk"] = "wechatpaySDK";
    })(TypeEnum = CheckoutSDKAction.TypeEnum || (CheckoutSDKAction.TypeEnum = {}));
})(CheckoutSDKAction = exports.CheckoutSDKAction || (exports.CheckoutSDKAction = {}));
