// @adyen/api-library/lib/src/typings/checkout/dokuDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DokuDetails = void 0;
class DokuDetails {
    static getAttributeTypeMap() {
        return DokuDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DokuDetails = DokuDetails;
DokuDetails.discriminator = undefined;
DokuDetails.mapping = undefined;
DokuDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string",
        "format": ""
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DokuDetails.TypeEnum",
        "format": ""
    }
];
(function (DokuDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["DokuMandiriVa"] = "doku_mandiri_va";
        TypeEnum["DokuCimbVa"] = "doku_cimb_va";
        TypeEnum["DokuDanamonVa"] = "doku_danamon_va";
        TypeEnum["DokuBniVa"] = "doku_bni_va";
        TypeEnum["DokuPermataLiteAtm"] = "doku_permata_lite_atm";
        TypeEnum["DokuBriVa"] = "doku_bri_va";
        TypeEnum["DokuBcaVa"] = "doku_bca_va";
        TypeEnum["DokuAlfamart"] = "doku_alfamart";
        TypeEnum["DokuIndomaret"] = "doku_indomaret";
        TypeEnum["DokuWallet"] = "doku_wallet";
        TypeEnum["DokuOvo"] = "doku_ovo";
    })(TypeEnum = DokuDetails.TypeEnum || (DokuDetails.TypeEnum = {}));
})(DokuDetails = exports.DokuDetails || (exports.DokuDetails = {}));
