// @adyen/api-library/lib/src/typings/checkout/donationPaymentRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DonationPaymentRequest = void 0;
class DonationPaymentRequest {
    static getAttributeTypeMap() {
        return DonationPaymentRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DonationPaymentRequest = DonationPaymentRequest;
DonationPaymentRequest.discriminator = undefined;
DonationPaymentRequest.mapping = undefined;
DonationPaymentRequest.attributeTypeMap = [
    {
        "name": "accountInfo",
        "baseName": "accountInfo",
        "type": "AccountInfo | null",
        "format": ""
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo | null",
        "format": ""
    },
    {
        "name": "authenticationData",
        "baseName": "authenticationData",
        "type": "AuthenticationData | null",
        "format": ""
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "BillingAddress | null",
        "format": ""
    },
    {
        "name": "browserInfo",
        "baseName": "browserInfo",
        "type": "BrowserInfo | null",
        "format": ""
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "DonationPaymentRequest.ChannelEnum",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "conversionId",
        "baseName": "conversionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deliverAt",
        "baseName": "deliverAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "DeliveryAddress | null",
        "format": ""
    },
    {
        "name": "deviceFingerprint",
        "baseName": "deviceFingerprint",
        "type": "string",
        "format": ""
    },
    {
        "name": "donationAccount",
        "baseName": "donationAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "donationCampaignId",
        "baseName": "donationCampaignId",
        "type": "string",
        "format": ""
    },
    {
        "name": "donationOriginalPspReference",
        "baseName": "donationOriginalPspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "donationToken",
        "baseName": "donationToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantRiskIndicator",
        "baseName": "merchantRiskIndicator",
        "type": "MerchantRiskIndicator | null",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "mpiData",
        "baseName": "mpiData",
        "type": "ThreeDSecureData | null",
        "format": ""
    },
    {
        "name": "origin",
        "baseName": "origin",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "DonationPaymentRequestPaymentMethod",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "DonationPaymentRequest.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "redirectFromIssuerMethod",
        "baseName": "redirectFromIssuerMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "redirectToIssuerMethod",
        "baseName": "redirectToIssuerMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "returnUrl",
        "baseName": "returnUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "sessionValidity",
        "baseName": "sessionValidity",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "DonationPaymentRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2RequestData",
        "baseName": "threeDS2RequestData",
        "type": "ThreeDS2RequestFields | null",
        "format": ""
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean",
        "format": ""
    }
];
(function (DonationPaymentRequest) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum["IOs"] = "iOS";
        ChannelEnum["Android"] = "Android";
        ChannelEnum["Web"] = "Web";
    })(ChannelEnum = DonationPaymentRequest.ChannelEnum || (DonationPaymentRequest.ChannelEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = DonationPaymentRequest.RecurringProcessingModelEnum || (DonationPaymentRequest.RecurringProcessingModelEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = DonationPaymentRequest.ShopperInteractionEnum || (DonationPaymentRequest.ShopperInteractionEnum = {}));
})(DonationPaymentRequest = exports.DonationPaymentRequest || (exports.DonationPaymentRequest = {}));
