// @adyen/api-library/lib/src/typings/checkout/dragonpayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DragonpayDetails = void 0;
class DragonpayDetails {
    static getAttributeTypeMap() {
        return DragonpayDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DragonpayDetails = DragonpayDetails;
DragonpayDetails.discriminator = undefined;
DragonpayDetails.mapping = undefined;
DragonpayDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DragonpayDetails.TypeEnum",
        "format": ""
    }
];
(function (DragonpayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["DragonpayEbanking"] = "dragonpay_ebanking";
        TypeEnum["DragonpayOtcBanking"] = "dragonpay_otc_banking";
        TypeEnum["DragonpayOtcNonBanking"] = "dragonpay_otc_non_banking";
        TypeEnum["DragonpayOtcPhilippines"] = "dragonpay_otc_philippines";
    })(TypeEnum = DragonpayDetails.TypeEnum || (DragonpayDetails.TypeEnum = {}));
})(DragonpayDetails = exports.DragonpayDetails || (exports.DragonpayDetails = {}));
