// @adyen/api-library/lib/src/typings/checkout/econtextVoucherDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EcontextVoucherDetails = void 0;
class EcontextVoucherDetails {
    static getAttributeTypeMap() {
        return EcontextVoucherDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.EcontextVoucherDetails = EcontextVoucherDetails;
EcontextVoucherDetails.discriminator = undefined;
EcontextVoucherDetails.mapping = undefined;
EcontextVoucherDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "firstName",
        "baseName": "firstName",
        "type": "string",
        "format": ""
    },
    {
        "name": "lastName",
        "baseName": "lastName",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "EcontextVoucherDetails.TypeEnum",
        "format": ""
    }
];
(function (EcontextVoucherDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["EcontextSevenEleven"] = "econtext_seven_eleven";
        TypeEnum["EcontextOnline"] = "econtext_online";
        TypeEnum["Econtext"] = "econtext";
        TypeEnum["EcontextStores"] = "econtext_stores";
        TypeEnum["EcontextAtm"] = "econtext_atm";
    })(TypeEnum = EcontextVoucherDetails.TypeEnum || (EcontextVoucherDetails.TypeEnum = {}));
})(EcontextVoucherDetails = exports.EcontextVoucherDetails || (exports.EcontextVoucherDetails = {}));
