// @adyen/api-library/lib/src/typings/checkout/fundRecipient.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FundRecipient = void 0;
class FundRecipient {
    static getAttributeTypeMap() {
        return FundRecipient.attributeTypeMap;
    }
    constructor() {
    }
}
exports.FundRecipient = FundRecipient;
FundRecipient.discriminator = undefined;
FundRecipient.mapping = undefined;
FundRecipient.attributeTypeMap = [
    {
        "name": "IBAN",
        "baseName": "IBAN",
        "type": "string",
        "format": ""
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "CardDetails | null",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "subMerchant",
        "baseName": "subMerchant",
        "type": "SubMerchant | null",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "walletIdentifier",
        "baseName": "walletIdentifier",
        "type": "string",
        "format": ""
    },
    {
        "name": "walletOwnerTaxId",
        "baseName": "walletOwnerTaxId",
        "type": "string",
        "format": ""
    },
    {
        "name": "walletPurpose",
        "baseName": "walletPurpose",
        "type": "FundRecipient.WalletPurposeEnum",
        "format": ""
    }
];
(function (FundRecipient) {
    let WalletPurposeEnum;
    (function (WalletPurposeEnum) {
        WalletPurposeEnum["IdentifiedBoleto"] = "identifiedBoleto";
        WalletPurposeEnum["TransferDifferentWallet"] = "transferDifferentWallet";
        WalletPurposeEnum["TransferOwnWallet"] = "transferOwnWallet";
        WalletPurposeEnum["TransferSameWallet"] = "transferSameWallet";
        WalletPurposeEnum["UnidentifiedBoleto"] = "unidentifiedBoleto";
    })(WalletPurposeEnum = FundRecipient.WalletPurposeEnum || (FundRecipient.WalletPurposeEnum = {}));
})(FundRecipient = exports.FundRecipient || (exports.FundRecipient = {}));
