// @adyen/api-library/lib/src/typings/checkout/googlePayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GooglePayDetails = void 0;
class GooglePayDetails {
    static getAttributeTypeMap() {
        return GooglePayDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.GooglePayDetails = GooglePayDetails;
GooglePayDetails.discriminator = undefined;
GooglePayDetails.mapping = undefined;
GooglePayDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "GooglePayDetails.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "googlePayCardNetwork",
        "baseName": "googlePayCardNetwork",
        "type": "string",
        "format": ""
    },
    {
        "name": "googlePayToken",
        "baseName": "googlePayToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2SdkVersion",
        "baseName": "threeDS2SdkVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GooglePayDetails.TypeEnum",
        "format": ""
    }
];
(function (GooglePayDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = GooglePayDetails.FundingSourceEnum || (GooglePayDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Googlepay"] = "googlepay";
    })(TypeEnum = GooglePayDetails.TypeEnum || (GooglePayDetails.TypeEnum = {}));
})(GooglePayDetails = exports.GooglePayDetails || (exports.GooglePayDetails = {}));
