// @adyen/api-library/lib/src/typings/checkout/googlePayDonations.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GooglePayDonations = void 0;
class GooglePayDonations {
    static getAttributeTypeMap() {
        return GooglePayDonations.attributeTypeMap;
    }
    constructor() {
    }
}
exports.GooglePayDonations = GooglePayDonations;
GooglePayDonations.discriminator = undefined;
GooglePayDonations.mapping = undefined;
GooglePayDonations.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "GooglePayDonations.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "googlePayCardNetwork",
        "baseName": "googlePayCardNetwork",
        "type": "string",
        "format": ""
    },
    {
        "name": "googlePayToken",
        "baseName": "googlePayToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2SdkVersion",
        "baseName": "threeDS2SdkVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "GooglePayDonations.TypeEnum",
        "format": ""
    }
];
(function (GooglePayDonations) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = GooglePayDonations.FundingSourceEnum || (GooglePayDonations.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Googlepay"] = "googlepay";
    })(TypeEnum = GooglePayDonations.TypeEnum || (GooglePayDonations.TypeEnum = {}));
})(GooglePayDonations = exports.GooglePayDonations || (exports.GooglePayDonations = {}));
