// @adyen/api-library/lib/src/typings/checkout/klarnaDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KlarnaDetails = void 0;
class KlarnaDetails {
    static getAttributeTypeMap() {
        return KlarnaDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.KlarnaDetails = KlarnaDetails;
KlarnaDetails.discriminator = undefined;
KlarnaDetails.mapping = undefined;
KlarnaDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "subtype",
        "baseName": "subtype",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "KlarnaDetails.TypeEnum",
        "format": ""
    }
];
(function (KlarnaDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Klarna"] = "klarna";
        TypeEnum["Klarnapayments"] = "klarnapayments";
        TypeEnum["KlarnapaymentsAccount"] = "klarnapayments_account";
        TypeEnum["KlarnapaymentsB2b"] = "klarnapayments_b2b";
        TypeEnum["KlarnaPaynow"] = "klarna_paynow";
        TypeEnum["KlarnaAccount"] = "klarna_account";
        TypeEnum["KlarnaB2b"] = "klarna_b2b";
    })(TypeEnum = KlarnaDetails.TypeEnum || (KlarnaDetails.TypeEnum = {}));
})(KlarnaDetails = exports.KlarnaDetails || (exports.KlarnaDetails = {}));
