// @adyen/api-library/lib/src/typings/checkout/molPayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MolPayDetails = void 0;
class MolPayDetails {
    static getAttributeTypeMap() {
        return MolPayDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.MolPayDetails = MolPayDetails;
MolPayDetails.discriminator = undefined;
MolPayDetails.mapping = undefined;
MolPayDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuer",
        "baseName": "issuer",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "MolPayDetails.TypeEnum",
        "format": ""
    }
];
(function (MolPayDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["MolpayEbankingFpxMy"] = "molpay_ebanking_fpx_MY";
        TypeEnum["MolpayEbankingTh"] = "molpay_ebanking_TH";
    })(TypeEnum = MolPayDetails.TypeEnum || (MolPayDetails.TypeEnum = {}));
})(MolPayDetails = exports.MolPayDetails || (exports.MolPayDetails = {}));
