// @adyen/api-library/lib/src/typings/checkout/paymentDetailsResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentDetailsResponse = void 0;
class PaymentDetailsResponse {
    static getAttributeTypeMap() {
        return PaymentDetailsResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentDetailsResponse = PaymentDetailsResponse;
PaymentDetailsResponse.discriminator = undefined;
PaymentDetailsResponse.mapping = undefined;
PaymentDetailsResponse.attributeTypeMap = [
    {
        "name": "action",
        "baseName": "action",
        "type": "CheckoutThreeDS2Action | null",
        "format": ""
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "donationToken",
        "baseName": "donationToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "fraudResult",
        "baseName": "fraudResult",
        "type": "FraudResult | null",
        "format": ""
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "order",
        "baseName": "order",
        "type": "CheckoutOrderResponse | null",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "ResponsePaymentMethod | null",
        "format": ""
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "refusalReason",
        "baseName": "refusalReason",
        "type": "string",
        "format": ""
    },
    {
        "name": "refusalReasonCode",
        "baseName": "refusalReasonCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "resultCode",
        "baseName": "resultCode",
        "type": "PaymentDetailsResponse.ResultCodeEnum",
        "format": ""
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2ResponseData",
        "baseName": "threeDS2ResponseData",
        "type": "ThreeDS2ResponseData | null",
        "format": ""
    },
    {
        "name": "threeDS2Result",
        "baseName": "threeDS2Result",
        "type": "ThreeDS2Result | null",
        "format": ""
    },
    {
        "name": "threeDSPaymentData",
        "baseName": "threeDSPaymentData",
        "type": "string",
        "format": ""
    }
];
(function (PaymentDetailsResponse) {
    let ResultCodeEnum;
    (function (ResultCodeEnum) {
        ResultCodeEnum["AuthenticationFinished"] = "AuthenticationFinished";
        ResultCodeEnum["AuthenticationNotRequired"] = "AuthenticationNotRequired";
        ResultCodeEnum["Authorised"] = "Authorised";
        ResultCodeEnum["Cancelled"] = "Cancelled";
        ResultCodeEnum["ChallengeShopper"] = "ChallengeShopper";
        ResultCodeEnum["Error"] = "Error";
        ResultCodeEnum["IdentifyShopper"] = "IdentifyShopper";
        ResultCodeEnum["PartiallyAuthorised"] = "PartiallyAuthorised";
        ResultCodeEnum["Pending"] = "Pending";
        ResultCodeEnum["PresentToShopper"] = "PresentToShopper";
        ResultCodeEnum["Received"] = "Received";
        ResultCodeEnum["RedirectShopper"] = "RedirectShopper";
        ResultCodeEnum["Refused"] = "Refused";
        ResultCodeEnum["Success"] = "Success";
    })(ResultCodeEnum = PaymentDetailsResponse.ResultCodeEnum || (PaymentDetailsResponse.ResultCodeEnum = {}));
})(PaymentDetailsResponse = exports.PaymentDetailsResponse || (exports.PaymentDetailsResponse = {}));
