// @adyen/api-library/lib/src/typings/checkout/paymentLinkResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentLinkResponse = void 0;
class PaymentLinkResponse {
    static getAttributeTypeMap() {
        return PaymentLinkResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentLinkResponse = PaymentLinkResponse;
PaymentLinkResponse.discriminator = undefined;
PaymentLinkResponse.mapping = undefined;
PaymentLinkResponse.attributeTypeMap = [
    {
        "name": "allowedPaymentMethods",
        "baseName": "allowedPaymentMethods",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo | null",
        "format": ""
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "blockedPaymentMethods",
        "baseName": "blockedPaymentMethods",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string",
        "format": "date"
    },
    {
        "name": "deliverAt",
        "baseName": "deliverAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiresAt",
        "baseName": "expiresAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "fundOrigin",
        "baseName": "fundOrigin",
        "type": "FundOrigin | null",
        "format": ""
    },
    {
        "name": "fundRecipient",
        "baseName": "fundRecipient",
        "type": "FundRecipient | null",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "installmentOptions",
        "baseName": "installmentOptions",
        "type": "{ [key: string]: InstallmentOption; }",
        "format": ""
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>",
        "format": ""
    },
    {
        "name": "manualCapture",
        "baseName": "manualCapture",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic | null",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "PaymentLinkResponse.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "requiredShopperFields",
        "baseName": "requiredShopperFields",
        "type": "PaymentLinkResponse.RequiredShopperFieldsEnum",
        "format": ""
    },
    {
        "name": "returnUrl",
        "baseName": "returnUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "reusable",
        "baseName": "reusable",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "riskData",
        "baseName": "riskData",
        "type": "RiskData | null",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string",
        "format": ""
    },
    {
        "name": "showRemovePaymentMethodButton",
        "baseName": "showRemovePaymentMethodButton",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "splitCardFundingSources",
        "baseName": "splitCardFundingSources",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentLinkResponse.StatusEnum",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    },
    {
        "name": "storePaymentMethodMode",
        "baseName": "storePaymentMethodMode",
        "type": "PaymentLinkResponse.StorePaymentMethodModeEnum",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "themeId",
        "baseName": "themeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2RequestData",
        "baseName": "threeDS2RequestData",
        "type": "CheckoutSessionThreeDS2RequestData | null",
        "format": ""
    },
    {
        "name": "updatedAt",
        "baseName": "updatedAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    }
];
(function (PaymentLinkResponse) {
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = PaymentLinkResponse.RecurringProcessingModelEnum || (PaymentLinkResponse.RecurringProcessingModelEnum = {}));
    let RequiredShopperFieldsEnum;
    (function (RequiredShopperFieldsEnum) {
        RequiredShopperFieldsEnum["BillingAddress"] = "billingAddress";
        RequiredShopperFieldsEnum["DeliveryAddress"] = "deliveryAddress";
        RequiredShopperFieldsEnum["ShopperEmail"] = "shopperEmail";
        RequiredShopperFieldsEnum["ShopperName"] = "shopperName";
        RequiredShopperFieldsEnum["TelephoneNumber"] = "telephoneNumber";
    })(RequiredShopperFieldsEnum = PaymentLinkResponse.RequiredShopperFieldsEnum || (PaymentLinkResponse.RequiredShopperFieldsEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Completed"] = "completed";
        StatusEnum["Expired"] = "expired";
        StatusEnum["Paid"] = "paid";
        StatusEnum["PaymentPending"] = "paymentPending";
    })(StatusEnum = PaymentLinkResponse.StatusEnum || (PaymentLinkResponse.StatusEnum = {}));
    let StorePaymentMethodModeEnum;
    (function (StorePaymentMethodModeEnum) {
        StorePaymentMethodModeEnum["AskForConsent"] = "askForConsent";
        StorePaymentMethodModeEnum["Disabled"] = "disabled";
        StorePaymentMethodModeEnum["Enabled"] = "enabled";
    })(StorePaymentMethodModeEnum = PaymentLinkResponse.StorePaymentMethodModeEnum || (PaymentLinkResponse.StorePaymentMethodModeEnum = {}));
})(PaymentLinkResponse = exports.PaymentLinkResponse || (exports.PaymentLinkResponse = {}));
