// @adyen/api-library/lib/src/typings/checkout/paymentRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequest = void 0;
class PaymentRequest {
    static getAttributeTypeMap() {
        return PaymentRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentRequest = PaymentRequest;
PaymentRequest.discriminator = undefined;
PaymentRequest.mapping = undefined;
PaymentRequest.attributeTypeMap = [
    {
        "name": "accountInfo",
        "baseName": "accountInfo",
        "type": "AccountInfo | null",
        "format": ""
    },
    {
        "name": "additionalAmount",
        "baseName": "additionalAmount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo | null",
        "format": ""
    },
    {
        "name": "authenticationData",
        "baseName": "authenticationData",
        "type": "AuthenticationData | null",
        "format": ""
    },
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "CheckoutBankAccount | null",
        "format": ""
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "BillingAddress | null",
        "format": ""
    },
    {
        "name": "browserInfo",
        "baseName": "browserInfo",
        "type": "BrowserInfo | null",
        "format": ""
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "channel",
        "baseName": "channel",
        "type": "PaymentRequest.ChannelEnum",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "company",
        "baseName": "company",
        "type": "Company | null",
        "format": ""
    },
    {
        "name": "conversionId",
        "baseName": "conversionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "countryCode",
        "baseName": "countryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "dccQuote",
        "baseName": "dccQuote",
        "type": "ForexQuote | null",
        "format": ""
    },
    {
        "name": "deliverAt",
        "baseName": "deliverAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "DeliveryAddress | null",
        "format": ""
    },
    {
        "name": "deliveryDate",
        "baseName": "deliveryDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deviceFingerprint",
        "baseName": "deviceFingerprint",
        "type": "string",
        "format": ""
    },
    {
        "name": "enableOneClick",
        "baseName": "enableOneClick",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "enablePayOut",
        "baseName": "enablePayOut",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "enableRecurring",
        "baseName": "enableRecurring",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "enhancedSchemeData",
        "baseName": "enhancedSchemeData",
        "type": "EnhancedSchemeData | null",
        "format": ""
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "PaymentRequest.EntityTypeEnum",
        "format": ""
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "fundOrigin",
        "baseName": "fundOrigin",
        "type": "FundOrigin | null",
        "format": ""
    },
    {
        "name": "fundRecipient",
        "baseName": "fundRecipient",
        "type": "FundRecipient | null",
        "format": ""
    },
    {
        "name": "industryUsage",
        "baseName": "industryUsage",
        "type": "PaymentRequest.IndustryUsageEnum",
        "format": ""
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "Installments | null",
        "format": ""
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>",
        "format": ""
    },
    {
        "name": "localizedShopperStatement",
        "baseName": "localizedShopperStatement",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "mandate",
        "baseName": "mandate",
        "type": "Mandate | null",
        "format": ""
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantRiskIndicator",
        "baseName": "merchantRiskIndicator",
        "type": "MerchantRiskIndicator | null",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "mpiData",
        "baseName": "mpiData",
        "type": "ThreeDSecureData | null",
        "format": ""
    },
    {
        "name": "order",
        "baseName": "order",
        "type": "EncryptedOrderData | null",
        "format": ""
    },
    {
        "name": "orderReference",
        "baseName": "orderReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "origin",
        "baseName": "origin",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "PaymentRequestPaymentMethod",
        "format": ""
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic | null",
        "format": ""
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "PaymentRequest.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "redirectFromIssuerMethod",
        "baseName": "redirectFromIssuerMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "redirectToIssuerMethod",
        "baseName": "redirectToIssuerMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "returnUrl",
        "baseName": "returnUrl",
        "type": "string",
        "format": ""
    },
    {
        "name": "riskData",
        "baseName": "riskData",
        "type": "RiskData | null",
        "format": ""
    },
    {
        "name": "sessionValidity",
        "baseName": "sessionValidity",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperConversionId",
        "baseName": "shopperConversionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PaymentRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string",
        "format": ""
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    },
    {
        "name": "storePaymentMethod",
        "baseName": "storePaymentMethod",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "subMerchants",
        "baseName": "subMerchants",
        "type": "Array<SubMerchantInfo>",
        "format": ""
    },
    {
        "name": "surcharge",
        "baseName": "surcharge",
        "type": "Surcharge | null",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2RequestData",
        "baseName": "threeDS2RequestData",
        "type": "ThreeDS2RequestFields | null",
        "format": ""
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "trustedShopper",
        "baseName": "trustedShopper",
        "type": "boolean",
        "format": ""
    }
];
(function (PaymentRequest) {
    let ChannelEnum;
    (function (ChannelEnum) {
        ChannelEnum["IOs"] = "iOS";
        ChannelEnum["Android"] = "Android";
        ChannelEnum["Web"] = "Web";
    })(ChannelEnum = PaymentRequest.ChannelEnum || (PaymentRequest.ChannelEnum = {}));
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["CompanyName"] = "CompanyName";
    })(EntityTypeEnum = PaymentRequest.EntityTypeEnum || (PaymentRequest.EntityTypeEnum = {}));
    let IndustryUsageEnum;
    (function (IndustryUsageEnum) {
        IndustryUsageEnum["DelayedCharge"] = "delayedCharge";
        IndustryUsageEnum["Installment"] = "installment";
        IndustryUsageEnum["NoShow"] = "noShow";
    })(IndustryUsageEnum = PaymentRequest.IndustryUsageEnum || (PaymentRequest.IndustryUsageEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = PaymentRequest.RecurringProcessingModelEnum || (PaymentRequest.RecurringProcessingModelEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = PaymentRequest.ShopperInteractionEnum || (PaymentRequest.ShopperInteractionEnum = {}));
})(PaymentRequest = exports.PaymentRequest || (exports.PaymentRequest = {}));
