// @adyen/api-library/lib/src/typings/checkout/paymentRefundResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRefundResponse = void 0;
class PaymentRefundResponse {
    static getAttributeTypeMap() {
        return PaymentRefundResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentRefundResponse = PaymentRefundResponse;
PaymentRefundResponse.discriminator = undefined;
PaymentRefundResponse.mapping = undefined;
PaymentRefundResponse.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "capturePspReference",
        "baseName": "capturePspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantRefundReason",
        "baseName": "merchantRefundReason",
        "type": "PaymentRefundResponse.MerchantRefundReasonEnum | null",
        "format": ""
    },
    {
        "name": "paymentPspReference",
        "baseName": "paymentPspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "pspReference",
        "baseName": "pspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PaymentRefundResponse.StatusEnum",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    }
];
(function (PaymentRefundResponse) {
    let MerchantRefundReasonEnum;
    (function (MerchantRefundReasonEnum) {
        MerchantRefundReasonEnum["Fraud"] = "FRAUD";
        MerchantRefundReasonEnum["CustomerRequest"] = "CUSTOMER REQUEST";
        MerchantRefundReasonEnum["Return"] = "RETURN";
        MerchantRefundReasonEnum["Duplicate"] = "DUPLICATE";
        MerchantRefundReasonEnum["Other"] = "OTHER";
    })(MerchantRefundReasonEnum = PaymentRefundResponse.MerchantRefundReasonEnum || (PaymentRefundResponse.MerchantRefundReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Received"] = "received";
    })(StatusEnum = PaymentRefundResponse.StatusEnum || (PaymentRefundResponse.StatusEnum = {}));
})(PaymentRefundResponse = exports.PaymentRefundResponse || (exports.PaymentRefundResponse = {}));
