// @adyen/api-library/lib/src/typings/checkout/paymentRefundRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRefundRequest = void 0;
class PaymentRefundRequest {
    static getAttributeTypeMap() {
        return PaymentRefundRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentRefundRequest = PaymentRefundRequest;
PaymentRefundRequest.discriminator = undefined;
PaymentRefundRequest.mapping = undefined;
PaymentRefundRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo | null",
        "format": ""
    },
    {
        "name": "capturePspReference",
        "baseName": "capturePspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "enhancedSchemeData",
        "baseName": "enhancedSchemeData",
        "type": "EnhancedSchemeData | null",
        "format": ""
    },
    {
        "name": "lineItems",
        "baseName": "lineItems",
        "type": "Array<LineItem>",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantRefundReason",
        "baseName": "merchantRefundReason",
        "type": "PaymentRefundRequest.MerchantRefundReasonEnum | null",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    }
];
(function (PaymentRefundRequest) {
    let MerchantRefundReasonEnum;
    (function (MerchantRefundReasonEnum) {
        MerchantRefundReasonEnum["Fraud"] = "FRAUD";
        MerchantRefundReasonEnum["CustomerRequest"] = "CUSTOMER REQUEST";
        MerchantRefundReasonEnum["Return"] = "RETURN";
        MerchantRefundReasonEnum["Duplicate"] = "DUPLICATE";
        MerchantRefundReasonEnum["Other"] = "OTHER";
    })(MerchantRefundReasonEnum = PaymentRefundRequest.MerchantRefundReasonEnum || (PaymentRefundRequest.MerchantRefundReasonEnum = {}));
})(PaymentRefundRequest = exports.PaymentRefundRequest || (exports.PaymentRefundRequest = {}));
