// @adyen/api-library/lib/src/typings/checkout/paymentMethod.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethod = void 0;
class PaymentMethod {
    static getAttributeTypeMap() {
        return PaymentMethod.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentMethod = PaymentMethod;
PaymentMethod.discriminator = undefined;
PaymentMethod.mapping = undefined;
PaymentMethod.attributeTypeMap = [
    {
        "name": "apps",
        "baseName": "apps",
        "type": "Array<PaymentMethodUPIApps>",
        "format": ""
    },
    {
        "name": "brand",
        "baseName": "brand",
        "type": "string",
        "format": ""
    },
    {
        "name": "brands",
        "baseName": "brands",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "configuration",
        "baseName": "configuration",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "PaymentMethod.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "group",
        "baseName": "group",
        "type": "PaymentMethodGroup | null",
        "format": ""
    },
    {
        "name": "inputDetails",
        "baseName": "inputDetails",
        "type": "Array<InputDetail>",
        "format": ""
    },
    {
        "name": "issuers",
        "baseName": "issuers",
        "type": "Array<PaymentMethodIssuer>",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "promoted",
        "baseName": "promoted",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string",
        "format": ""
    }
];
(function (PaymentMethod) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = PaymentMethod.FundingSourceEnum || (PaymentMethod.FundingSourceEnum = {}));
})(PaymentMethod = exports.PaymentMethod || (exports.PaymentMethod = {}));
