// @adyen/api-library/lib/src/typings/checkout/pixRecurring.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PixRecurring = void 0;
class PixRecurring {
    static getAttributeTypeMap() {
        return PixRecurring.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PixRecurring = PixRecurring;
PixRecurring.discriminator = undefined;
PixRecurring.mapping = undefined;
PixRecurring.attributeTypeMap = [
    {
        "name": "billingDate",
        "baseName": "billingDate",
        "type": "string",
        "format": ""
    },
    {
        "name": "businessDayOnly",
        "baseName": "businessDayOnly",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "endsAt",
        "baseName": "endsAt",
        "type": "string",
        "format": ""
    },
    {
        "name": "frequency",
        "baseName": "frequency",
        "type": "PixRecurring.FrequencyEnum",
        "format": ""
    },
    {
        "name": "minAmount",
        "baseName": "minAmount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "originalPspReference",
        "baseName": "originalPspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringAmount",
        "baseName": "recurringAmount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "recurringStatement",
        "baseName": "recurringStatement",
        "type": "string",
        "format": ""
    },
    {
        "name": "retryPolicy",
        "baseName": "retryPolicy",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "startsAt",
        "baseName": "startsAt",
        "type": "string",
        "format": ""
    }
];
(function (PixRecurring) {
    let FrequencyEnum;
    (function (FrequencyEnum) {
        FrequencyEnum["Weekly"] = "weekly";
        FrequencyEnum["Monthly"] = "monthly";
        FrequencyEnum["Quarterly"] = "quarterly";
        FrequencyEnum["HalfYearly"] = "half-yearly";
        FrequencyEnum["Yearly"] = "yearly";
    })(FrequencyEnum = PixRecurring.FrequencyEnum || (PixRecurring.FrequencyEnum = {}));
})(PixRecurring = exports.PixRecurring || (exports.PixRecurring = {}));
