// @adyen/api-library/lib/src/typings/checkout/platformChargebackLogic.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlatformChargebackLogic = void 0;
class PlatformChargebackLogic {
    static getAttributeTypeMap() {
        return PlatformChargebackLogic.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PlatformChargebackLogic = PlatformChargebackLogic;
PlatformChargebackLogic.discriminator = undefined;
PlatformChargebackLogic.mapping = undefined;
PlatformChargebackLogic.attributeTypeMap = [
    {
        "name": "behavior",
        "baseName": "behavior",
        "type": "PlatformChargebackLogic.BehaviorEnum",
        "format": ""
    },
    {
        "name": "costAllocationAccount",
        "baseName": "costAllocationAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "targetAccount",
        "baseName": "targetAccount",
        "type": "string",
        "format": ""
    }
];
(function (PlatformChargebackLogic) {
    let BehaviorEnum;
    (function (BehaviorEnum) {
        BehaviorEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
        BehaviorEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        BehaviorEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(BehaviorEnum = PlatformChargebackLogic.BehaviorEnum || (PlatformChargebackLogic.BehaviorEnum = {}));
})(PlatformChargebackLogic = exports.PlatformChargebackLogic || (exports.PlatformChargebackLogic = {}));
