// @adyen/api-library/lib/src/typings/checkout/responseAdditionalDataCard.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseAdditionalDataCard = void 0;
class ResponseAdditionalDataCard {
    static getAttributeTypeMap() {
        return ResponseAdditionalDataCard.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ResponseAdditionalDataCard = ResponseAdditionalDataCard;
ResponseAdditionalDataCard.discriminator = undefined;
ResponseAdditionalDataCard.mapping = undefined;
ResponseAdditionalDataCard.attributeTypeMap = [
    {
        "name": "cardBin",
        "baseName": "cardBin",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardHolderName",
        "baseName": "cardHolderName",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardIssuingBank",
        "baseName": "cardIssuingBank",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardIssuingCountry",
        "baseName": "cardIssuingCountry",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardIssuingCurrency",
        "baseName": "cardIssuingCurrency",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardPaymentMethod",
        "baseName": "cardPaymentMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "cardProductId",
        "baseName": "cardProductId",
        "type": "ResponseAdditionalDataCard.CardProductIdEnum",
        "format": ""
    },
    {
        "name": "cardSummary",
        "baseName": "cardSummary",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuerBin",
        "baseName": "issuerBin",
        "type": "string",
        "format": ""
    }
];
(function (ResponseAdditionalDataCard) {
    let CardProductIdEnum;
    (function (CardProductIdEnum) {
        CardProductIdEnum["A"] = "A";
        CardProductIdEnum["B"] = "B";
        CardProductIdEnum["C"] = "C";
        CardProductIdEnum["D"] = "D";
        CardProductIdEnum["F"] = "F";
        CardProductIdEnum["Mcc"] = "MCC";
        CardProductIdEnum["Mce"] = "MCE";
        CardProductIdEnum["Mcf"] = "MCF";
        CardProductIdEnum["Mcg"] = "MCG";
        CardProductIdEnum["Mch"] = "MCH";
        CardProductIdEnum["Mci"] = "MCI";
    })(CardProductIdEnum = ResponseAdditionalDataCard.CardProductIdEnum || (ResponseAdditionalDataCard.CardProductIdEnum = {}));
})(ResponseAdditionalDataCard = exports.ResponseAdditionalDataCard || (exports.ResponseAdditionalDataCard = {}));
