// @adyen/api-library/lib/src/typings/checkout/responseAdditionalDataCommon.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResponseAdditionalDataCommon = void 0;
class ResponseAdditionalDataCommon {
    static getAttributeTypeMap() {
        return ResponseAdditionalDataCommon.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ResponseAdditionalDataCommon = ResponseAdditionalDataCommon;
ResponseAdditionalDataCommon.discriminator = undefined;
ResponseAdditionalDataCommon.mapping = undefined;
ResponseAdditionalDataCommon.attributeTypeMap = [
    {
        "name": "acquirerAccountCode",
        "baseName": "acquirerAccountCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "acquirerCode",
        "baseName": "acquirerCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "acquirerReference",
        "baseName": "acquirerReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "alias",
        "baseName": "alias",
        "type": "string",
        "format": ""
    },
    {
        "name": "aliasType",
        "baseName": "aliasType",
        "type": "string",
        "format": ""
    },
    {
        "name": "authCode",
        "baseName": "authCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "authorisationMid",
        "baseName": "authorisationMid",
        "type": "string",
        "format": ""
    },
    {
        "name": "authorisedAmountCurrency",
        "baseName": "authorisedAmountCurrency",
        "type": "string",
        "format": ""
    },
    {
        "name": "authorisedAmountValue",
        "baseName": "authorisedAmountValue",
        "type": "string",
        "format": ""
    },
    {
        "name": "avsResult",
        "baseName": "avsResult",
        "type": "string",
        "format": ""
    },
    {
        "name": "avsResultRaw",
        "baseName": "avsResultRaw",
        "type": "string",
        "format": ""
    },
    {
        "name": "bic",
        "baseName": "bic",
        "type": "string",
        "format": ""
    },
    {
        "name": "coBrandedWith",
        "baseName": "coBrandedWith",
        "type": "string",
        "format": ""
    },
    {
        "name": "cvcResult",
        "baseName": "cvcResult",
        "type": "string",
        "format": ""
    },
    {
        "name": "cvcResultRaw",
        "baseName": "cvcResultRaw",
        "type": "string",
        "format": ""
    },
    {
        "name": "dsTransID",
        "baseName": "dsTransID",
        "type": "string",
        "format": ""
    },
    {
        "name": "eci",
        "baseName": "eci",
        "type": "string",
        "format": ""
    },
    {
        "name": "expiryDate",
        "baseName": "expiryDate",
        "type": "string",
        "format": ""
    },
    {
        "name": "extraCostsCurrency",
        "baseName": "extraCostsCurrency",
        "type": "string",
        "format": ""
    },
    {
        "name": "extraCostsValue",
        "baseName": "extraCostsValue",
        "type": "string",
        "format": ""
    },
    {
        "name": "fraudCheck__itemNr__FraudCheckname",
        "baseName": "fraudCheck-[itemNr]-[FraudCheckname]",
        "type": "string",
        "format": ""
    },
    {
        "name": "fraudManualReview",
        "baseName": "fraudManualReview",
        "type": "string",
        "format": ""
    },
    {
        "name": "fraudResultType",
        "baseName": "fraudResultType",
        "type": "ResponseAdditionalDataCommon.FraudResultTypeEnum",
        "format": ""
    },
    {
        "name": "fraudRiskLevel",
        "baseName": "fraudRiskLevel",
        "type": "ResponseAdditionalDataCommon.FraudRiskLevelEnum",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundsAvailability",
        "baseName": "fundsAvailability",
        "type": "string",
        "format": ""
    },
    {
        "name": "inferredRefusalReason",
        "baseName": "inferredRefusalReason",
        "type": "string",
        "format": ""
    },
    {
        "name": "isCardCommercial",
        "baseName": "isCardCommercial",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuerCountry",
        "baseName": "issuerCountry",
        "type": "string",
        "format": ""
    },
    {
        "name": "liabilityShift",
        "baseName": "liabilityShift",
        "type": "string",
        "format": ""
    },
    {
        "name": "mcBankNetReferenceNumber",
        "baseName": "mcBankNetReferenceNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantAdviceCode",
        "baseName": "merchantAdviceCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantReference",
        "baseName": "merchantReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "networkTxReference",
        "baseName": "networkTxReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "ownerName",
        "baseName": "ownerName",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentAccountReference",
        "baseName": "paymentAccountReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethodVariant",
        "baseName": "paymentMethodVariant",
        "type": "string",
        "format": ""
    },
    {
        "name": "payoutEligible",
        "baseName": "payoutEligible",
        "type": "string",
        "format": ""
    },
    {
        "name": "realtimeAccountUpdaterStatus",
        "baseName": "realtimeAccountUpdaterStatus",
        "type": "string",
        "format": ""
    },
    {
        "name": "receiptFreeText",
        "baseName": "receiptFreeText",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurring_contractTypes",
        "baseName": "recurring.contractTypes",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurring_firstPspReference",
        "baseName": "recurring.firstPspReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurring_recurringDetailReference",
        "baseName": "recurring.recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurring_shopperReference",
        "baseName": "recurring.shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "ResponseAdditionalDataCommon.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "referred",
        "baseName": "referred",
        "type": "string",
        "format": ""
    },
    {
        "name": "refusalReasonRaw",
        "baseName": "refusalReasonRaw",
        "type": "string",
        "format": ""
    },
    {
        "name": "requestAmount",
        "baseName": "requestAmount",
        "type": "string",
        "format": ""
    },
    {
        "name": "requestCurrencyCode",
        "baseName": "requestCurrencyCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "terminalId",
        "baseName": "terminalId",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDAuthenticated",
        "baseName": "threeDAuthenticated",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDAuthenticatedResponse",
        "baseName": "threeDAuthenticatedResponse",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDOffered",
        "baseName": "threeDOffered",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDOfferedResponse",
        "baseName": "threeDOfferedResponse",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSVersion",
        "baseName": "threeDSVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "tokenization_shopperReference",
        "baseName": "tokenization.shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "tokenization_store_operationType",
        "baseName": "tokenization.store.operationType",
        "type": "ResponseAdditionalDataCommon.TokenizationStoreOperationTypeEnum",
        "format": ""
    },
    {
        "name": "tokenization_storedPaymentMethodId",
        "baseName": "tokenization.storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "visaTransactionId",
        "baseName": "visaTransactionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "xid",
        "baseName": "xid",
        "type": "string",
        "format": ""
    }
];
(function (ResponseAdditionalDataCommon) {
    let FraudResultTypeEnum;
    (function (FraudResultTypeEnum) {
        FraudResultTypeEnum["Green"] = "GREEN";
        FraudResultTypeEnum["Fraud"] = "FRAUD";
    })(FraudResultTypeEnum = ResponseAdditionalDataCommon.FraudResultTypeEnum || (ResponseAdditionalDataCommon.FraudResultTypeEnum = {}));
    let FraudRiskLevelEnum;
    (function (FraudRiskLevelEnum) {
        FraudRiskLevelEnum["VeryLow"] = "veryLow";
        FraudRiskLevelEnum["Low"] = "low";
        FraudRiskLevelEnum["Medium"] = "medium";
        FraudRiskLevelEnum["High"] = "high";
        FraudRiskLevelEnum["VeryHigh"] = "veryHigh";
    })(FraudRiskLevelEnum = ResponseAdditionalDataCommon.FraudRiskLevelEnum || (ResponseAdditionalDataCommon.FraudRiskLevelEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = ResponseAdditionalDataCommon.RecurringProcessingModelEnum || (ResponseAdditionalDataCommon.RecurringProcessingModelEnum = {}));
    let TokenizationStoreOperationTypeEnum;
    (function (TokenizationStoreOperationTypeEnum) {
        TokenizationStoreOperationTypeEnum["Created"] = "created";
        TokenizationStoreOperationTypeEnum["Updated"] = "updated";
        TokenizationStoreOperationTypeEnum["AlreadyExisting"] = "alreadyExisting";
    })(TokenizationStoreOperationTypeEnum = ResponseAdditionalDataCommon.TokenizationStoreOperationTypeEnum || (ResponseAdditionalDataCommon.TokenizationStoreOperationTypeEnum = {}));
})(ResponseAdditionalDataCommon = exports.ResponseAdditionalDataCommon || (exports.ResponseAdditionalDataCommon = {}));
