// @adyen/api-library/lib/src/typings/checkout/rivertyDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RivertyDetails = void 0;
class RivertyDetails {
    static getAttributeTypeMap() {
        return RivertyDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.RivertyDetails = RivertyDetails;
RivertyDetails.discriminator = undefined;
RivertyDetails.mapping = undefined;
RivertyDetails.attributeTypeMap = [
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "string",
        "format": ""
    },
    {
        "name": "deviceFingerprint",
        "baseName": "deviceFingerprint",
        "type": "string",
        "format": ""
    },
    {
        "name": "iban",
        "baseName": "iban",
        "type": "string",
        "format": ""
    },
    {
        "name": "personalDetails",
        "baseName": "personalDetails",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "subtype",
        "baseName": "subtype",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "RivertyDetails.TypeEnum",
        "format": ""
    }
];
(function (RivertyDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Riverty"] = "riverty";
        TypeEnum["RivertyAccount"] = "riverty_account";
        TypeEnum["RivertyInstallments"] = "riverty_installments";
        TypeEnum["SepadirectdebitRiverty"] = "sepadirectdebit_riverty";
    })(TypeEnum = RivertyDetails.TypeEnum || (RivertyDetails.TypeEnum = {}));
})(RivertyDetails = exports.RivertyDetails || (exports.RivertyDetails = {}));
