// @adyen/api-library/lib/src/typings/checkout/samsungPayDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SamsungPayDetails = void 0;
class SamsungPayDetails {
    static getAttributeTypeMap() {
        return SamsungPayDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SamsungPayDetails = SamsungPayDetails;
SamsungPayDetails.discriminator = undefined;
SamsungPayDetails.mapping = undefined;
SamsungPayDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "SamsungPayDetails.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "samsungPayToken",
        "baseName": "samsungPayToken",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SamsungPayDetails.TypeEnum",
        "format": ""
    }
];
(function (SamsungPayDetails) {
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = SamsungPayDetails.FundingSourceEnum || (SamsungPayDetails.FundingSourceEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Samsungpay"] = "samsungpay";
    })(TypeEnum = SamsungPayDetails.TypeEnum || (SamsungPayDetails.TypeEnum = {}));
})(SamsungPayDetails = exports.SamsungPayDetails || (exports.SamsungPayDetails = {}));
