// @adyen/api-library/lib/src/typings/checkout/sepaDirectDebitDetails.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SepaDirectDebitDetails = void 0;
class SepaDirectDebitDetails {
    static getAttributeTypeMap() {
        return SepaDirectDebitDetails.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SepaDirectDebitDetails = SepaDirectDebitDetails;
SepaDirectDebitDetails.discriminator = undefined;
SepaDirectDebitDetails.mapping = undefined;
SepaDirectDebitDetails.attributeTypeMap = [
    {
        "name": "checkoutAttemptId",
        "baseName": "checkoutAttemptId",
        "type": "string",
        "format": ""
    },
    {
        "name": "iban",
        "baseName": "iban",
        "type": "string",
        "format": ""
    },
    {
        "name": "ownerName",
        "baseName": "ownerName",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "storedPaymentMethodId",
        "baseName": "storedPaymentMethodId",
        "type": "string",
        "format": ""
    },
    {
        "name": "transferInstrumentId",
        "baseName": "transferInstrumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SepaDirectDebitDetails.TypeEnum",
        "format": ""
    }
];
(function (SepaDirectDebitDetails) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Sepadirectdebit"] = "sepadirectdebit";
        TypeEnum["SepadirectdebitAmazonpay"] = "sepadirectdebit_amazonpay";
    })(TypeEnum = SepaDirectDebitDetails.TypeEnum || (SepaDirectDebitDetails.TypeEnum = {}));
})(SepaDirectDebitDetails = exports.SepaDirectDebitDetails || (exports.SepaDirectDebitDetails = {}));
