// @adyen/api-library/lib/src/typings/checkout/storedPaymentMethodRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredPaymentMethodRequest = void 0;
class StoredPaymentMethodRequest {
    static getAttributeTypeMap() {
        return StoredPaymentMethodRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.StoredPaymentMethodRequest = StoredPaymentMethodRequest;
StoredPaymentMethodRequest.discriminator = undefined;
StoredPaymentMethodRequest.mapping = undefined;
StoredPaymentMethodRequest.attributeTypeMap = [
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "PaymentMethodToStore",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "StoredPaymentMethodRequest.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    }
];
(function (StoredPaymentMethodRequest) {
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = StoredPaymentMethodRequest.RecurringProcessingModelEnum || (StoredPaymentMethodRequest.RecurringProcessingModelEnum = {}));
})(StoredPaymentMethodRequest = exports.StoredPaymentMethodRequest || (exports.StoredPaymentMethodRequest = {}));
