// @adyen/api-library/lib/src/typings/checkout/threeDSecureData.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDSecureData = void 0;
class ThreeDSecureData {
    static getAttributeTypeMap() {
        return ThreeDSecureData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ThreeDSecureData = ThreeDSecureData;
ThreeDSecureData.discriminator = undefined;
ThreeDSecureData.mapping = undefined;
ThreeDSecureData.attributeTypeMap = [
    {
        "name": "authenticationResponse",
        "baseName": "authenticationResponse",
        "type": "ThreeDSecureData.AuthenticationResponseEnum",
        "format": ""
    },
    {
        "name": "cavv",
        "baseName": "cavv",
        "type": "string",
        "format": "byte"
    },
    {
        "name": "cavvAlgorithm",
        "baseName": "cavvAlgorithm",
        "type": "string",
        "format": ""
    },
    {
        "name": "challengeCancel",
        "baseName": "challengeCancel",
        "type": "ThreeDSecureData.ChallengeCancelEnum",
        "format": ""
    },
    {
        "name": "directoryResponse",
        "baseName": "directoryResponse",
        "type": "ThreeDSecureData.DirectoryResponseEnum",
        "format": ""
    },
    {
        "name": "dsTransID",
        "baseName": "dsTransID",
        "type": "string",
        "format": ""
    },
    {
        "name": "eci",
        "baseName": "eci",
        "type": "string",
        "format": ""
    },
    {
        "name": "riskScore",
        "baseName": "riskScore",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSVersion",
        "baseName": "threeDSVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "tokenAuthenticationVerificationValue",
        "baseName": "tokenAuthenticationVerificationValue",
        "type": "string",
        "format": "byte"
    },
    {
        "name": "transStatusReason",
        "baseName": "transStatusReason",
        "type": "string",
        "format": ""
    },
    {
        "name": "xid",
        "baseName": "xid",
        "type": "string",
        "format": "byte"
    }
];
(function (ThreeDSecureData) {
    let AuthenticationResponseEnum;
    (function (AuthenticationResponseEnum) {
        AuthenticationResponseEnum["Y"] = "Y";
        AuthenticationResponseEnum["N"] = "N";
        AuthenticationResponseEnum["U"] = "U";
        AuthenticationResponseEnum["A"] = "A";
    })(AuthenticationResponseEnum = ThreeDSecureData.AuthenticationResponseEnum || (ThreeDSecureData.AuthenticationResponseEnum = {}));
    let ChallengeCancelEnum;
    (function (ChallengeCancelEnum) {
        ChallengeCancelEnum["_01"] = "01";
        ChallengeCancelEnum["_02"] = "02";
        ChallengeCancelEnum["_03"] = "03";
        ChallengeCancelEnum["_04"] = "04";
        ChallengeCancelEnum["_05"] = "05";
        ChallengeCancelEnum["_06"] = "06";
        ChallengeCancelEnum["_07"] = "07";
    })(ChallengeCancelEnum = ThreeDSecureData.ChallengeCancelEnum || (ThreeDSecureData.ChallengeCancelEnum = {}));
    let DirectoryResponseEnum;
    (function (DirectoryResponseEnum) {
        DirectoryResponseEnum["A"] = "A";
        DirectoryResponseEnum["C"] = "C";
        DirectoryResponseEnum["D"] = "D";
        DirectoryResponseEnum["I"] = "I";
        DirectoryResponseEnum["N"] = "N";
        DirectoryResponseEnum["R"] = "R";
        DirectoryResponseEnum["U"] = "U";
        DirectoryResponseEnum["Y"] = "Y";
    })(DirectoryResponseEnum = ThreeDSecureData.DirectoryResponseEnum || (ThreeDSecureData.DirectoryResponseEnum = {}));
})(ThreeDSecureData = exports.ThreeDSecureData || (exports.ThreeDSecureData = {}));
