// @adyen/api-library/lib/src/typings/checkout/threeDS2RequestData.js
"use strict";
/*
 * The version of the OpenAPI document: v71
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThreeDS2RequestData = void 0;
class ThreeDS2RequestData {
    static getAttributeTypeMap() {
        return ThreeDS2RequestData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.ThreeDS2RequestData = ThreeDS2RequestData;
ThreeDS2RequestData.discriminator = undefined;
ThreeDS2RequestData.mapping = undefined;
ThreeDS2RequestData.attributeTypeMap = [
    {
        "name": "acctInfo",
        "baseName": "acctInfo",
        "type": "AcctInfo | null",
        "format": ""
    },
    {
        "name": "acctType",
        "baseName": "acctType",
        "type": "ThreeDS2RequestData.AcctTypeEnum",
        "format": ""
    },
    {
        "name": "acquirerBIN",
        "baseName": "acquirerBIN",
        "type": "string",
        "format": ""
    },
    {
        "name": "acquirerMerchantID",
        "baseName": "acquirerMerchantID",
        "type": "string",
        "format": ""
    },
    {
        "name": "addrMatch",
        "baseName": "addrMatch",
        "type": "ThreeDS2RequestData.AddrMatchEnum",
        "format": ""
    },
    {
        "name": "authenticationOnly",
        "baseName": "authenticationOnly",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "challengeIndicator",
        "baseName": "challengeIndicator",
        "type": "ThreeDS2RequestData.ChallengeIndicatorEnum",
        "format": ""
    },
    {
        "name": "deviceChannel",
        "baseName": "deviceChannel",
        "type": "string",
        "format": ""
    },
    {
        "name": "deviceRenderOptions",
        "baseName": "deviceRenderOptions",
        "type": "DeviceRenderOptions | null",
        "format": ""
    },
    {
        "name": "homePhone",
        "baseName": "homePhone",
        "type": "Phone | null",
        "format": ""
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantName",
        "baseName": "merchantName",
        "type": "string",
        "format": ""
    },
    {
        "name": "messageVersion",
        "baseName": "messageVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "mobilePhone",
        "baseName": "mobilePhone",
        "type": "Phone | null",
        "format": ""
    },
    {
        "name": "notificationURL",
        "baseName": "notificationURL",
        "type": "string",
        "format": ""
    },
    {
        "name": "payTokenInd",
        "baseName": "payTokenInd",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "paymentAuthenticationUseCase",
        "baseName": "paymentAuthenticationUseCase",
        "type": "string",
        "format": ""
    },
    {
        "name": "purchaseInstalData",
        "baseName": "purchaseInstalData",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkAppID",
        "baseName": "sdkAppID",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkEncData",
        "baseName": "sdkEncData",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkEphemPubKey",
        "baseName": "sdkEphemPubKey",
        "type": "SDKEphemPubKey | null",
        "format": ""
    },
    {
        "name": "sdkMaxTimeout",
        "baseName": "sdkMaxTimeout",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "sdkReferenceNumber",
        "baseName": "sdkReferenceNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkTransID",
        "baseName": "sdkTransID",
        "type": "string",
        "format": ""
    },
    {
        "name": "sdkVersion",
        "baseName": "sdkVersion",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSCompInd",
        "baseName": "threeDSCompInd",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSRequestorAuthenticationInd",
        "baseName": "threeDSRequestorAuthenticationInd",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSRequestorAuthenticationInfo",
        "baseName": "threeDSRequestorAuthenticationInfo",
        "type": "ThreeDSRequestorAuthenticationInfo | null",
        "format": ""
    },
    {
        "name": "threeDSRequestorChallengeInd",
        "baseName": "threeDSRequestorChallengeInd",
        "type": "ThreeDS2RequestData.ThreeDSRequestorChallengeIndEnum",
        "format": ""
    },
    {
        "name": "threeDSRequestorID",
        "baseName": "threeDSRequestorID",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSRequestorName",
        "baseName": "threeDSRequestorName",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDSRequestorPriorAuthenticationInfo",
        "baseName": "threeDSRequestorPriorAuthenticationInfo",
        "type": "ThreeDSRequestorPriorAuthenticationInfo | null",
        "format": ""
    },
    {
        "name": "threeDSRequestorURL",
        "baseName": "threeDSRequestorURL",
        "type": "string",
        "format": ""
    },
    {
        "name": "transType",
        "baseName": "transType",
        "type": "ThreeDS2RequestData.TransTypeEnum",
        "format": ""
    },
    {
        "name": "transactionType",
        "baseName": "transactionType",
        "type": "ThreeDS2RequestData.TransactionTypeEnum",
        "format": ""
    },
    {
        "name": "whiteListStatus",
        "baseName": "whiteListStatus",
        "type": "string",
        "format": ""
    },
    {
        "name": "workPhone",
        "baseName": "workPhone",
        "type": "Phone | null",
        "format": ""
    }
];
(function (ThreeDS2RequestData) {
    let AcctTypeEnum;
    (function (AcctTypeEnum) {
        AcctTypeEnum["_01"] = "01";
        AcctTypeEnum["_02"] = "02";
        AcctTypeEnum["_03"] = "03";
    })(AcctTypeEnum = ThreeDS2RequestData.AcctTypeEnum || (ThreeDS2RequestData.AcctTypeEnum = {}));
    let AddrMatchEnum;
    (function (AddrMatchEnum) {
        AddrMatchEnum["Y"] = "Y";
        AddrMatchEnum["N"] = "N";
    })(AddrMatchEnum = ThreeDS2RequestData.AddrMatchEnum || (ThreeDS2RequestData.AddrMatchEnum = {}));
    let ChallengeIndicatorEnum;
    (function (ChallengeIndicatorEnum) {
        ChallengeIndicatorEnum["NoPreference"] = "noPreference";
        ChallengeIndicatorEnum["RequestNoChallenge"] = "requestNoChallenge";
        ChallengeIndicatorEnum["RequestChallenge"] = "requestChallenge";
        ChallengeIndicatorEnum["RequestChallengeAsMandate"] = "requestChallengeAsMandate";
    })(ChallengeIndicatorEnum = ThreeDS2RequestData.ChallengeIndicatorEnum || (ThreeDS2RequestData.ChallengeIndicatorEnum = {}));
    let ThreeDSRequestorChallengeIndEnum;
    (function (ThreeDSRequestorChallengeIndEnum) {
        ThreeDSRequestorChallengeIndEnum["_01"] = "01";
        ThreeDSRequestorChallengeIndEnum["_02"] = "02";
        ThreeDSRequestorChallengeIndEnum["_03"] = "03";
        ThreeDSRequestorChallengeIndEnum["_04"] = "04";
        ThreeDSRequestorChallengeIndEnum["_05"] = "05";
        ThreeDSRequestorChallengeIndEnum["_06"] = "06";
    })(ThreeDSRequestorChallengeIndEnum = ThreeDS2RequestData.ThreeDSRequestorChallengeIndEnum || (ThreeDS2RequestData.ThreeDSRequestorChallengeIndEnum = {}));
    let TransTypeEnum;
    (function (TransTypeEnum) {
        TransTypeEnum["_01"] = "01";
        TransTypeEnum["_03"] = "03";
        TransTypeEnum["_10"] = "10";
        TransTypeEnum["_11"] = "11";
        TransTypeEnum["_28"] = "28";
    })(TransTypeEnum = ThreeDS2RequestData.TransTypeEnum || (ThreeDS2RequestData.TransTypeEnum = {}));
    let TransactionTypeEnum;
    (function (TransactionTypeEnum) {
        TransactionTypeEnum["GoodsOrServicePurchase"] = "goodsOrServicePurchase";
        TransactionTypeEnum["CheckAcceptance"] = "checkAcceptance";
        TransactionTypeEnum["AccountFunding"] = "accountFunding";
        TransactionTypeEnum["QuasiCashTransaction"] = "quasiCashTransaction";
        TransactionTypeEnum["PrepaidActivationAndLoad"] = "prepaidActivationAndLoad";
    })(TransactionTypeEnum = ThreeDS2RequestData.TransactionTypeEnum || (ThreeDS2RequestData.TransactionTypeEnum = {}));
})(ThreeDS2RequestData = exports.ThreeDS2RequestData || (exports.ThreeDS2RequestData = {}));
