// @adyen/api-library/lib/src/typings/acsWebhooks/authenticationNotificationData.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthenticationNotificationData = void 0;
class AuthenticationNotificationData {
    static getAttributeTypeMap() {
        return AuthenticationNotificationData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AuthenticationNotificationData = AuthenticationNotificationData;
AuthenticationNotificationData.discriminator = undefined;
AuthenticationNotificationData.mapping = undefined;
AuthenticationNotificationData.attributeTypeMap = [
    {
        "name": "authentication",
        "baseName": "authentication",
        "type": "AuthenticationInfo",
        "format": ""
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "purchase",
        "baseName": "purchase",
        "type": "PurchaseInfo",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AuthenticationNotificationData.StatusEnum",
        "format": ""
    }
];
(function (AuthenticationNotificationData) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Authenticated"] = "authenticated";
        StatusEnum["Rejected"] = "rejected";
        StatusEnum["Error"] = "error";
    })(StatusEnum = AuthenticationNotificationData.StatusEnum || (AuthenticationNotificationData.StatusEnum = {}));
})(AuthenticationNotificationData = exports.AuthenticationNotificationData || (exports.AuthenticationNotificationData = {}));
