// @adyen/api-library/lib/src/typings/negativeBalanceWarningWebhooks/negativeBalanceWarningWebhooksHandler.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NegativeBalanceWarningWebhooksHandler = void 0;
const __1 = require("..");
/**
 * Handler for processing NegativeBalanceWarningWebhooks.
 *
 * This class provides functionality to deserialize the payload of NegativeBalanceWarningWebhooks events.
 */
class NegativeBalanceWarningWebhooksHandler {
    constructor(jsonPayload) {
        this.payload = JSON.parse(jsonPayload);
    }
    /**
     * This method checks the type of the webhook payload and returns the appropriate deserialized object.
     *
     * @returns A deserialized object of type GenericWebhook.
     * @throws Error if the type is not recognized.
     */
    getGenericWebhook() {
        const type = this.payload["type"];
        if (Object.values(__1.negativeBalanceWarningWebhooks.NegativeBalanceCompensationWarningNotificationRequest.TypeEnum).includes(type)) {
            return this.getNegativeBalanceCompensationWarningNotificationRequest();
        }
        throw new Error("Could not parse the json payload: " + this.payload);
    }
    /**
     * Deserialize the webhook payload into a NegativeBalanceCompensationWarningNotificationRequest
     *
     * @returns Deserialized NegativeBalanceCompensationWarningNotificationRequest object.
     */
    getNegativeBalanceCompensationWarningNotificationRequest() {
        return __1.negativeBalanceWarningWebhooks.ObjectSerializer.deserialize(this.payload, "NegativeBalanceCompensationWarningNotificationRequest");
    }
}
exports.NegativeBalanceWarningWebhooksHandler = NegativeBalanceWarningWebhooksHandler;
