// @adyen/api-library/lib/src/typings/management/accelInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccelInfo = void 0;
class AccelInfo {
    static getAttributeTypeMap() {
        return AccelInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AccelInfo = AccelInfo;
AccelInfo.discriminator = undefined;
AccelInfo.mapping = undefined;
AccelInfo.attributeTypeMap = [
    {
        "name": "processingType",
        "baseName": "processingType",
        "type": "AccelInfo.ProcessingTypeEnum",
        "format": ""
    },
    {
        "name": "transactionDescription",
        "baseName": "transactionDescription",
        "type": "TransactionDescriptionInfo | null",
        "format": ""
    }
];
(function (AccelInfo) {
    let ProcessingTypeEnum;
    (function (ProcessingTypeEnum) {
        ProcessingTypeEnum["Billpay"] = "billpay";
        ProcessingTypeEnum["Ecom"] = "ecom";
        ProcessingTypeEnum["Pos"] = "pos";
    })(ProcessingTypeEnum = AccelInfo.ProcessingTypeEnum || (AccelInfo.ProcessingTypeEnum = {}));
})(AccelInfo = exports.AccelInfo || (exports.AccelInfo = {}));
