// @adyen/api-library/lib/src/typings/management/amexInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AmexInfo = void 0;
class AmexInfo {
    static getAttributeTypeMap() {
        return AmexInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AmexInfo = AmexInfo;
AmexInfo.discriminator = undefined;
AmexInfo.mapping = undefined;
AmexInfo.attributeTypeMap = [
    {
        "name": "midNumber",
        "baseName": "midNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "reuseMidNumber",
        "baseName": "reuseMidNumber",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "serviceLevel",
        "baseName": "serviceLevel",
        "type": "AmexInfo.ServiceLevelEnum",
        "format": ""
    }
];
(function (AmexInfo) {
    let ServiceLevelEnum;
    (function (ServiceLevelEnum) {
        ServiceLevelEnum["NoContract"] = "noContract";
        ServiceLevelEnum["GatewayContract"] = "gatewayContract";
        ServiceLevelEnum["PaymentDesignatorContract"] = "paymentDesignatorContract";
    })(ServiceLevelEnum = AmexInfo.ServiceLevelEnum || (AmexInfo.ServiceLevelEnum = {}));
})(AmexInfo = exports.AmexInfo || (exports.AmexInfo = {}));
