// @adyen/api-library/lib/src/typings/management/androidApp.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AndroidApp = void 0;
class AndroidApp {
    static getAttributeTypeMap() {
        return AndroidApp.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AndroidApp = AndroidApp;
AndroidApp.discriminator = undefined;
AndroidApp.mapping = undefined;
AndroidApp.attributeTypeMap = [
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "errorCode",
        "baseName": "errorCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<AndroidAppError>",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string",
        "format": ""
    },
    {
        "name": "packageName",
        "baseName": "packageName",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "AndroidApp.StatusEnum",
        "format": ""
    },
    {
        "name": "versionCode",
        "baseName": "versionCode",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "versionName",
        "baseName": "versionName",
        "type": "string",
        "format": ""
    }
];
(function (AndroidApp) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Archived"] = "archived";
        StatusEnum["Error"] = "error";
        StatusEnum["Invalid"] = "invalid";
        StatusEnum["Processing"] = "processing";
        StatusEnum["Ready"] = "ready";
    })(StatusEnum = AndroidApp.StatusEnum || (AndroidApp.StatusEnum = {}));
})(AndroidApp = exports.AndroidApp || (exports.AndroidApp = {}));
