// @adyen/api-library/lib/src/typings/management/createCompanyWebhookRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCompanyWebhookRequest = void 0;
class CreateCompanyWebhookRequest {
    static getAttributeTypeMap() {
        return CreateCompanyWebhookRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CreateCompanyWebhookRequest = CreateCompanyWebhookRequest;
CreateCompanyWebhookRequest.discriminator = undefined;
CreateCompanyWebhookRequest.mapping = undefined;
CreateCompanyWebhookRequest.attributeTypeMap = [
    {
        "name": "acceptsExpiredCertificate",
        "baseName": "acceptsExpiredCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "acceptsSelfSignedCertificate",
        "baseName": "acceptsSelfSignedCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "acceptsUntrustedRootCertificate",
        "baseName": "acceptsUntrustedRootCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "additionalSettings",
        "baseName": "additionalSettings",
        "type": "AdditionalSettings | null",
        "format": ""
    },
    {
        "name": "communicationFormat",
        "baseName": "communicationFormat",
        "type": "CreateCompanyWebhookRequest.CommunicationFormatEnum",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptionProtocol",
        "baseName": "encryptionProtocol",
        "type": "CreateCompanyWebhookRequest.EncryptionProtocolEnum",
        "format": ""
    },
    {
        "name": "filterMerchantAccountType",
        "baseName": "filterMerchantAccountType",
        "type": "CreateCompanyWebhookRequest.FilterMerchantAccountTypeEnum",
        "format": ""
    },
    {
        "name": "filterMerchantAccounts",
        "baseName": "filterMerchantAccounts",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "networkType",
        "baseName": "networkType",
        "type": "CreateCompanyWebhookRequest.NetworkTypeEnum",
        "format": ""
    },
    {
        "name": "password",
        "baseName": "password",
        "type": "string",
        "format": ""
    },
    {
        "name": "populateSoapActionHeader",
        "baseName": "populateSoapActionHeader",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "string",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "username",
        "baseName": "username",
        "type": "string",
        "format": ""
    }
];
(function (CreateCompanyWebhookRequest) {
    let CommunicationFormatEnum;
    (function (CommunicationFormatEnum) {
        CommunicationFormatEnum["Http"] = "http";
        CommunicationFormatEnum["Json"] = "json";
        CommunicationFormatEnum["Soap"] = "soap";
    })(CommunicationFormatEnum = CreateCompanyWebhookRequest.CommunicationFormatEnum || (CreateCompanyWebhookRequest.CommunicationFormatEnum = {}));
    let EncryptionProtocolEnum;
    (function (EncryptionProtocolEnum) {
        EncryptionProtocolEnum["Http"] = "HTTP";
        EncryptionProtocolEnum["Tlsv12"] = "TLSv1.2";
        EncryptionProtocolEnum["Tlsv13"] = "TLSv1.3";
    })(EncryptionProtocolEnum = CreateCompanyWebhookRequest.EncryptionProtocolEnum || (CreateCompanyWebhookRequest.EncryptionProtocolEnum = {}));
    let FilterMerchantAccountTypeEnum;
    (function (FilterMerchantAccountTypeEnum) {
        FilterMerchantAccountTypeEnum["AllAccounts"] = "allAccounts";
        FilterMerchantAccountTypeEnum["ExcludeAccounts"] = "excludeAccounts";
        FilterMerchantAccountTypeEnum["IncludeAccounts"] = "includeAccounts";
    })(FilterMerchantAccountTypeEnum = CreateCompanyWebhookRequest.FilterMerchantAccountTypeEnum || (CreateCompanyWebhookRequest.FilterMerchantAccountTypeEnum = {}));
    let NetworkTypeEnum;
    (function (NetworkTypeEnum) {
        NetworkTypeEnum["Local"] = "local";
        NetworkTypeEnum["Public"] = "public";
    })(NetworkTypeEnum = CreateCompanyWebhookRequest.NetworkTypeEnum || (CreateCompanyWebhookRequest.NetworkTypeEnum = {}));
})(CreateCompanyWebhookRequest = exports.CreateCompanyWebhookRequest || (exports.CreateCompanyWebhookRequest = {}));
