// @adyen/api-library/lib/src/typings/management/jCBInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.JCBInfo = void 0;
class JCBInfo {
    static getAttributeTypeMap() {
        return JCBInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.JCBInfo = JCBInfo;
JCBInfo.discriminator = undefined;
JCBInfo.mapping = undefined;
JCBInfo.attributeTypeMap = [
    {
        "name": "midNumber",
        "baseName": "midNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "reuseMidNumber",
        "baseName": "reuseMidNumber",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "serviceLevel",
        "baseName": "serviceLevel",
        "type": "JCBInfo.ServiceLevelEnum",
        "format": ""
    },
    {
        "name": "transactionDescription",
        "baseName": "transactionDescription",
        "type": "TransactionDescriptionInfo | null",
        "format": ""
    }
];
(function (JCBInfo) {
    let ServiceLevelEnum;
    (function (ServiceLevelEnum) {
        ServiceLevelEnum["NoContract"] = "noContract";
        ServiceLevelEnum["GatewayContract"] = "gatewayContract";
        ServiceLevelEnum["PaymentDesignatorContract"] = "paymentDesignatorContract";
    })(ServiceLevelEnum = JCBInfo.ServiceLevelEnum || (JCBInfo.ServiceLevelEnum = {}));
})(JCBInfo = exports.JCBInfo || (exports.JCBInfo = {}));
