// @adyen/api-library/lib/src/typings/management/notification.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Notification = void 0;
class Notification {
    static getAttributeTypeMap() {
        return Notification.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Notification = Notification;
Notification.discriminator = undefined;
Notification.mapping = undefined;
Notification.attributeTypeMap = [
    {
        "name": "category",
        "baseName": "category",
        "type": "Notification.CategoryEnum",
        "format": ""
    },
    {
        "name": "details",
        "baseName": "details",
        "type": "string",
        "format": ""
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "showButton",
        "baseName": "showButton",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "title",
        "baseName": "title",
        "type": "string",
        "format": ""
    }
];
(function (Notification) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["SaleWakeUp"] = "SaleWakeUp";
        CategoryEnum["KeyPressed"] = "KeyPressed";
        CategoryEnum["Empty"] = "";
    })(CategoryEnum = Notification.CategoryEnum || (Notification.CategoryEnum = {}));
})(Notification = exports.Notification || (exports.Notification = {}));
