// @adyen/api-library/lib/src/typings/management/paymentMethodSetupInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodSetupInfo = void 0;
class PaymentMethodSetupInfo {
    static getAttributeTypeMap() {
        return PaymentMethodSetupInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentMethodSetupInfo = PaymentMethodSetupInfo;
PaymentMethodSetupInfo.discriminator = undefined;
PaymentMethodSetupInfo.mapping = undefined;
PaymentMethodSetupInfo.attributeTypeMap = [
    {
        "name": "accel",
        "baseName": "accel",
        "type": "AccelInfo | null",
        "format": ""
    },
    {
        "name": "affirm",
        "baseName": "affirm",
        "type": "AffirmInfo | null",
        "format": ""
    },
    {
        "name": "afterpayTouch",
        "baseName": "afterpayTouch",
        "type": "AfterpayTouchInfo | null",
        "format": ""
    },
    {
        "name": "alipayPlus",
        "baseName": "alipayPlus",
        "type": "AlipayPlusInfo | null",
        "format": ""
    },
    {
        "name": "amex",
        "baseName": "amex",
        "type": "AmexInfo | null",
        "format": ""
    },
    {
        "name": "applePay",
        "baseName": "applePay",
        "type": "ApplePayInfo | null",
        "format": ""
    },
    {
        "name": "bcmc",
        "baseName": "bcmc",
        "type": "BcmcInfo | null",
        "format": ""
    },
    {
        "name": "businessLineId",
        "baseName": "businessLineId",
        "type": "string",
        "format": ""
    },
    {
        "name": "cartesBancaires",
        "baseName": "cartesBancaires",
        "type": "CartesBancairesInfo | null",
        "format": ""
    },
    {
        "name": "clearpay",
        "baseName": "clearpay",
        "type": "ClearpayInfo | null",
        "format": ""
    },
    {
        "name": "countries",
        "baseName": "countries",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "cup",
        "baseName": "cup",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "currencies",
        "baseName": "currencies",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "customRoutingFlags",
        "baseName": "customRoutingFlags",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "diners",
        "baseName": "diners",
        "type": "DinersInfo | null",
        "format": ""
    },
    {
        "name": "discover",
        "baseName": "discover",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "eft_directdebit_CA",
        "baseName": "eft_directdebit_CA",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "eftpos_australia",
        "baseName": "eftpos_australia",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "girocard",
        "baseName": "girocard",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "googlePay",
        "baseName": "googlePay",
        "type": "GooglePayInfo | null",
        "format": ""
    },
    {
        "name": "ideal",
        "baseName": "ideal",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "interac_card",
        "baseName": "interac_card",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "jcb",
        "baseName": "jcb",
        "type": "JCBInfo | null",
        "format": ""
    },
    {
        "name": "klarna",
        "baseName": "klarna",
        "type": "KlarnaInfo | null",
        "format": ""
    },
    {
        "name": "maestro",
        "baseName": "maestro",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "maestro_usa",
        "baseName": "maestro_usa",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "mc",
        "baseName": "mc",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "mealVoucher_FR",
        "baseName": "mealVoucher_FR",
        "type": "MealVoucherFRInfo | null",
        "format": ""
    },
    {
        "name": "nyce",
        "baseName": "nyce",
        "type": "NyceInfo | null",
        "format": ""
    },
    {
        "name": "paybybank_plaid",
        "baseName": "paybybank_plaid",
        "type": "PayByBankPlaidInfo | null",
        "format": ""
    },
    {
        "name": "payme",
        "baseName": "payme",
        "type": "PayMeInfo | null",
        "format": ""
    },
    {
        "name": "paypal",
        "baseName": "paypal",
        "type": "PayPalInfo | null",
        "format": ""
    },
    {
        "name": "payto",
        "baseName": "payto",
        "type": "PayToInfo | null",
        "format": ""
    },
    {
        "name": "pulse",
        "baseName": "pulse",
        "type": "PulseInfo | null",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "sepadirectdebit",
        "baseName": "sepadirectdebit",
        "type": "SepaDirectDebitInfo | null",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PaymentMethodSetupInfo.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "sodexo",
        "baseName": "sodexo",
        "type": "SodexoInfo | null",
        "format": ""
    },
    {
        "name": "sofort",
        "baseName": "sofort",
        "type": "SofortInfo | null",
        "format": ""
    },
    {
        "name": "star",
        "baseName": "star",
        "type": "StarInfo | null",
        "format": ""
    },
    {
        "name": "storeIds",
        "baseName": "storeIds",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "swish",
        "baseName": "swish",
        "type": "SwishInfo | null",
        "format": ""
    },
    {
        "name": "ticket",
        "baseName": "ticket",
        "type": "TicketInfo | null",
        "format": ""
    },
    {
        "name": "twint",
        "baseName": "twint",
        "type": "TwintInfo | null",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentMethodSetupInfo.TypeEnum",
        "format": ""
    },
    {
        "name": "vipps",
        "baseName": "vipps",
        "type": "VippsInfo | null",
        "format": ""
    },
    {
        "name": "visa",
        "baseName": "visa",
        "type": "GenericPmWithTdiInfo | null",
        "format": ""
    },
    {
        "name": "wechatpay",
        "baseName": "wechatpay",
        "type": "WeChatPayInfo | null",
        "format": ""
    },
    {
        "name": "wechatpay_pos",
        "baseName": "wechatpay_pos",
        "type": "WeChatPayPosInfo | null",
        "format": ""
    }
];
(function (PaymentMethodSetupInfo) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["ECommerce"] = "eCommerce";
        ShopperInteractionEnum["Pos"] = "pos";
        ShopperInteractionEnum["Moto"] = "moto";
        ShopperInteractionEnum["ContAuth"] = "contAuth";
    })(ShopperInteractionEnum = PaymentMethodSetupInfo.ShopperInteractionEnum || (PaymentMethodSetupInfo.ShopperInteractionEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Abrapetite"] = "abrapetite";
        TypeEnum["AbrapetiteCredit"] = "abrapetite_credit";
        TypeEnum["AbrapetiteDebit"] = "abrapetite_debit";
        TypeEnum["AbrapetitePrepaid"] = "abrapetite_prepaid";
        TypeEnum["Accel"] = "accel";
        TypeEnum["Ach"] = "ach";
        TypeEnum["Affirm"] = "affirm";
        TypeEnum["Afterpaytouch"] = "afterpaytouch";
        TypeEnum["Alelo"] = "alelo";
        TypeEnum["Alipay"] = "alipay";
        TypeEnum["AlipayHk"] = "alipay_hk";
        TypeEnum["AlipayPlus"] = "alipay_plus";
        TypeEnum["AlipayWap"] = "alipay_wap";
        TypeEnum["Amex"] = "amex";
        TypeEnum["Applepay"] = "applepay";
        TypeEnum["Avancard"] = "avancard";
        TypeEnum["AvancardCredit"] = "avancard_credit";
        TypeEnum["AvancardDebit"] = "avancard_debit";
        TypeEnum["BaneseCard"] = "banese_card";
        TypeEnum["BaneseCardCredit"] = "banese_card_credit";
        TypeEnum["BaneseCardDebit"] = "banese_card_debit";
        TypeEnum["BaneseCardPrepaid"] = "banese_card_prepaid";
        TypeEnum["Bcmc"] = "bcmc";
        TypeEnum["Blik"] = "blik";
        TypeEnum["Cartebancaire"] = "cartebancaire";
        TypeEnum["Clearpay"] = "clearpay";
        TypeEnum["Clicktopay"] = "clicktopay";
        TypeEnum["Cooper"] = "cooper";
        TypeEnum["CooperCredit"] = "cooper_credit";
        TypeEnum["CooperDebit"] = "cooper_debit";
        TypeEnum["CooperFoodDebit"] = "cooper_food_debit";
        TypeEnum["CooperMealDebit"] = "cooper_meal_debit";
        TypeEnum["CooperPrepaid"] = "cooper_prepaid";
        TypeEnum["CooperPrivateCredit"] = "cooper_private_credit";
        TypeEnum["CooperRetailCredit"] = "cooper_retail_credit";
        TypeEnum["Credtodos"] = "credtodos";
        TypeEnum["CredtodosPrivateCredit"] = "credtodos_private_credit";
        TypeEnum["CredtodosPrivateDebit"] = "credtodos_private_debit";
        TypeEnum["Cup"] = "cup";
        TypeEnum["Diners"] = "diners";
        TypeEnum["DirectdebitGb"] = "directdebit_GB";
        TypeEnum["Discover"] = "discover";
        TypeEnum["EbankingFi"] = "ebanking_FI";
        TypeEnum["EftDirectdebitCa"] = "eft_directdebit_CA";
        TypeEnum["EftposAustralia"] = "eftpos_australia";
        TypeEnum["Elo"] = "elo";
        TypeEnum["Elocredit"] = "elocredit";
        TypeEnum["Elodebit"] = "elodebit";
        TypeEnum["Girocard"] = "girocard";
        TypeEnum["Googlepay"] = "googlepay";
        TypeEnum["GreenCard"] = "green_card";
        TypeEnum["GreenCardCredit"] = "green_card_credit";
        TypeEnum["GreenCardDebit"] = "green_card_debit";
        TypeEnum["GreenCardFoodPrepaid"] = "green_card_food_prepaid";
        TypeEnum["GreenCardMealPrepaid"] = "green_card_meal_prepaid";
        TypeEnum["GreenCardPrepaid"] = "green_card_prepaid";
        TypeEnum["Hiper"] = "hiper";
        TypeEnum["Hipercard"] = "hipercard";
        TypeEnum["Ideal"] = "ideal";
        TypeEnum["InteracCard"] = "interac_card";
        TypeEnum["Jcb"] = "jcb";
        TypeEnum["Klarna"] = "klarna";
        TypeEnum["KlarnaAccount"] = "klarna_account";
        TypeEnum["KlarnaPaynow"] = "klarna_paynow";
        TypeEnum["LeCard"] = "le_card";
        TypeEnum["LeCardCredit"] = "le_card_credit";
        TypeEnum["LeCardDebit"] = "le_card_debit";
        TypeEnum["Maestro"] = "maestro";
        TypeEnum["MaestroUsa"] = "maestro_usa";
        TypeEnum["Maxifrota"] = "maxifrota";
        TypeEnum["MaxifrotaPrepaid"] = "maxifrota_prepaid";
        TypeEnum["Mbway"] = "mbway";
        TypeEnum["Mc"] = "mc";
        TypeEnum["Mcdebit"] = "mcdebit";
        TypeEnum["MealVoucherFr"] = "mealVoucher_FR";
        TypeEnum["Megaleve"] = "megaleve";
        TypeEnum["MegaleveCredit"] = "megaleve_credit";
        TypeEnum["MegaleveDebit"] = "megaleve_debit";
        TypeEnum["Mobilepay"] = "mobilepay";
        TypeEnum["Multibanco"] = "multibanco";
        TypeEnum["Nutricash"] = "nutricash";
        TypeEnum["NutricashPrepaid"] = "nutricash_prepaid";
        TypeEnum["Nyce"] = "nyce";
        TypeEnum["OnlineBankingPl"] = "onlineBanking_PL";
        TypeEnum["Paybybank"] = "paybybank";
        TypeEnum["PaybybankPlaid"] = "paybybank_plaid";
        TypeEnum["Payme"] = "payme";
        TypeEnum["PaymePos"] = "payme_pos";
        TypeEnum["Paynow"] = "paynow";
        TypeEnum["PaynowPos"] = "paynow_pos";
        TypeEnum["Paypal"] = "paypal";
        TypeEnum["Payto"] = "payto";
        TypeEnum["PersonalCard"] = "personal_card";
        TypeEnum["PersonalCardCredit"] = "personal_card_credit";
        TypeEnum["PersonalCardDebit"] = "personal_card_debit";
        TypeEnum["Pulse"] = "pulse";
        TypeEnum["Romcard"] = "romcard";
        TypeEnum["RomcardCredit"] = "romcard_credit";
        TypeEnum["RomcardDebit"] = "romcard_debit";
        TypeEnum["Senff"] = "senff";
        TypeEnum["SenffCredit"] = "senff_credit";
        TypeEnum["Sepadirectdebit"] = "sepadirectdebit";
        TypeEnum["Sodexo"] = "sodexo";
        TypeEnum["Star"] = "star";
        TypeEnum["Swish"] = "swish";
        TypeEnum["Ticket"] = "ticket";
        TypeEnum["TodoGiftcard"] = "todo_giftcard";
        TypeEnum["Trustly"] = "trustly";
        TypeEnum["Twint"] = "twint";
        TypeEnum["TwintPos"] = "twint_pos";
        TypeEnum["UpBrazil"] = "up_brazil";
        TypeEnum["UpBrazilCredit"] = "up_brazil_credit";
        TypeEnum["UpBrazilDebit"] = "up_brazil_debit";
        TypeEnum["UpBrazilPrepaid"] = "up_brazil_prepaid";
        TypeEnum["ValeRefeicao"] = "vale_refeicao";
        TypeEnum["ValeRefeicaoPrepaid"] = "vale_refeicao_prepaid";
        TypeEnum["VegasCard"] = "vegas_card";
        TypeEnum["VegasCardCredit"] = "vegas_card_credit";
        TypeEnum["VegasCardDebit"] = "vegas_card_debit";
        TypeEnum["VeroCard"] = "vero_card";
        TypeEnum["VeroCardCredit"] = "vero_card_credit";
        TypeEnum["VeroCardDebit"] = "vero_card_debit";
        TypeEnum["VeroCardPrepaid"] = "vero_card_prepaid";
        TypeEnum["Vipps"] = "vipps";
        TypeEnum["Visa"] = "visa";
        TypeEnum["Visadebit"] = "visadebit";
        TypeEnum["Vpay"] = "vpay";
        TypeEnum["Wechatpay"] = "wechatpay";
        TypeEnum["WechatpayPos"] = "wechatpay_pos";
    })(TypeEnum = PaymentMethodSetupInfo.TypeEnum || (PaymentMethodSetupInfo.TypeEnum = {}));
})(PaymentMethodSetupInfo = exports.PaymentMethodSetupInfo || (exports.PaymentMethodSetupInfo = {}));
