// @adyen/api-library/lib/src/typings/management/payoutSettings.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PayoutSettings = void 0;
class PayoutSettings {
    static getAttributeTypeMap() {
        return PayoutSettings.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PayoutSettings = PayoutSettings;
PayoutSettings.discriminator = undefined;
PayoutSettings.mapping = undefined;
PayoutSettings.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "enabledFromDate",
        "baseName": "enabledFromDate",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "PayoutSettings.PriorityEnum",
        "format": ""
    },
    {
        "name": "transferInstrumentId",
        "baseName": "transferInstrumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "PayoutSettings.VerificationStatusEnum",
        "format": ""
    }
];
(function (PayoutSettings) {
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["First"] = "first";
        PriorityEnum["Normal"] = "normal";
        PriorityEnum["Urgent"] = "urgent";
    })(PriorityEnum = PayoutSettings.PriorityEnum || (PayoutSettings.PriorityEnum = {}));
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum["Invalid"] = "invalid";
        VerificationStatusEnum["Pending"] = "pending";
        VerificationStatusEnum["Rejected"] = "rejected";
        VerificationStatusEnum["Valid"] = "valid";
    })(VerificationStatusEnum = PayoutSettings.VerificationStatusEnum || (PayoutSettings.VerificationStatusEnum = {}));
})(PayoutSettings = exports.PayoutSettings || (exports.PayoutSettings = {}));
