// @adyen/api-library/lib/src/typings/management/splitConfigurationLogic.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SplitConfigurationLogic = void 0;
class SplitConfigurationLogic {
    static getAttributeTypeMap() {
        return SplitConfigurationLogic.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SplitConfigurationLogic = SplitConfigurationLogic;
SplitConfigurationLogic.discriminator = undefined;
SplitConfigurationLogic.mapping = undefined;
SplitConfigurationLogic.attributeTypeMap = [
    {
        "name": "acquiringFees",
        "baseName": "acquiringFees",
        "type": "SplitConfigurationLogic.AcquiringFeesEnum",
        "format": ""
    },
    {
        "name": "additionalCommission",
        "baseName": "additionalCommission",
        "type": "AdditionalCommission | null",
        "format": ""
    },
    {
        "name": "adyenCommission",
        "baseName": "adyenCommission",
        "type": "SplitConfigurationLogic.AdyenCommissionEnum",
        "format": ""
    },
    {
        "name": "adyenFees",
        "baseName": "adyenFees",
        "type": "SplitConfigurationLogic.AdyenFeesEnum",
        "format": ""
    },
    {
        "name": "adyenMarkup",
        "baseName": "adyenMarkup",
        "type": "SplitConfigurationLogic.AdyenMarkupEnum",
        "format": ""
    },
    {
        "name": "chargeback",
        "baseName": "chargeback",
        "type": "SplitConfigurationLogic.ChargebackEnum",
        "format": ""
    },
    {
        "name": "chargebackCostAllocation",
        "baseName": "chargebackCostAllocation",
        "type": "SplitConfigurationLogic.ChargebackCostAllocationEnum",
        "format": ""
    },
    {
        "name": "commission",
        "baseName": "commission",
        "type": "Commission",
        "format": ""
    },
    {
        "name": "interchange",
        "baseName": "interchange",
        "type": "SplitConfigurationLogic.InterchangeEnum",
        "format": ""
    },
    {
        "name": "paymentFee",
        "baseName": "paymentFee",
        "type": "SplitConfigurationLogic.PaymentFeeEnum",
        "format": ""
    },
    {
        "name": "refund",
        "baseName": "refund",
        "type": "SplitConfigurationLogic.RefundEnum",
        "format": ""
    },
    {
        "name": "refundCostAllocation",
        "baseName": "refundCostAllocation",
        "type": "SplitConfigurationLogic.RefundCostAllocationEnum",
        "format": ""
    },
    {
        "name": "remainder",
        "baseName": "remainder",
        "type": "SplitConfigurationLogic.RemainderEnum",
        "format": ""
    },
    {
        "name": "schemeFee",
        "baseName": "schemeFee",
        "type": "SplitConfigurationLogic.SchemeFeeEnum",
        "format": ""
    },
    {
        "name": "splitLogicId",
        "baseName": "splitLogicId",
        "type": "string",
        "format": ""
    },
    {
        "name": "surcharge",
        "baseName": "surcharge",
        "type": "SplitConfigurationLogic.SurchargeEnum",
        "format": ""
    },
    {
        "name": "tip",
        "baseName": "tip",
        "type": "SplitConfigurationLogic.TipEnum",
        "format": ""
    }
];
(function (SplitConfigurationLogic) {
    let AcquiringFeesEnum;
    (function (AcquiringFeesEnum) {
        AcquiringFeesEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AcquiringFeesEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AcquiringFeesEnum = SplitConfigurationLogic.AcquiringFeesEnum || (SplitConfigurationLogic.AcquiringFeesEnum = {}));
    let AdyenCommissionEnum;
    (function (AdyenCommissionEnum) {
        AdyenCommissionEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenCommissionEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenCommissionEnum = SplitConfigurationLogic.AdyenCommissionEnum || (SplitConfigurationLogic.AdyenCommissionEnum = {}));
    let AdyenFeesEnum;
    (function (AdyenFeesEnum) {
        AdyenFeesEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenFeesEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenFeesEnum = SplitConfigurationLogic.AdyenFeesEnum || (SplitConfigurationLogic.AdyenFeesEnum = {}));
    let AdyenMarkupEnum;
    (function (AdyenMarkupEnum) {
        AdyenMarkupEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        AdyenMarkupEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(AdyenMarkupEnum = SplitConfigurationLogic.AdyenMarkupEnum || (SplitConfigurationLogic.AdyenMarkupEnum = {}));
    let ChargebackEnum;
    (function (ChargebackEnum) {
        ChargebackEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
        ChargebackEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
    })(ChargebackEnum = SplitConfigurationLogic.ChargebackEnum || (SplitConfigurationLogic.ChargebackEnum = {}));
    let ChargebackCostAllocationEnum;
    (function (ChargebackCostAllocationEnum) {
        ChargebackCostAllocationEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        ChargebackCostAllocationEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(ChargebackCostAllocationEnum = SplitConfigurationLogic.ChargebackCostAllocationEnum || (SplitConfigurationLogic.ChargebackCostAllocationEnum = {}));
    let InterchangeEnum;
    (function (InterchangeEnum) {
        InterchangeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        InterchangeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(InterchangeEnum = SplitConfigurationLogic.InterchangeEnum || (SplitConfigurationLogic.InterchangeEnum = {}));
    let PaymentFeeEnum;
    (function (PaymentFeeEnum) {
        PaymentFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        PaymentFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(PaymentFeeEnum = SplitConfigurationLogic.PaymentFeeEnum || (SplitConfigurationLogic.PaymentFeeEnum = {}));
    let RefundEnum;
    (function (RefundEnum) {
        RefundEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        RefundEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
        RefundEnum["DeductAccordingToSplitRatio"] = "deductAccordingToSplitRatio";
    })(RefundEnum = SplitConfigurationLogic.RefundEnum || (SplitConfigurationLogic.RefundEnum = {}));
    let RefundCostAllocationEnum;
    (function (RefundCostAllocationEnum) {
        RefundCostAllocationEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        RefundCostAllocationEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(RefundCostAllocationEnum = SplitConfigurationLogic.RefundCostAllocationEnum || (SplitConfigurationLogic.RefundCostAllocationEnum = {}));
    let RemainderEnum;
    (function (RemainderEnum) {
        RemainderEnum["AddToLiableAccount"] = "addToLiableAccount";
        RemainderEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(RemainderEnum = SplitConfigurationLogic.RemainderEnum || (SplitConfigurationLogic.RemainderEnum = {}));
    let SchemeFeeEnum;
    (function (SchemeFeeEnum) {
        SchemeFeeEnum["DeductFromLiableAccount"] = "deductFromLiableAccount";
        SchemeFeeEnum["DeductFromOneBalanceAccount"] = "deductFromOneBalanceAccount";
    })(SchemeFeeEnum = SplitConfigurationLogic.SchemeFeeEnum || (SplitConfigurationLogic.SchemeFeeEnum = {}));
    let SurchargeEnum;
    (function (SurchargeEnum) {
        SurchargeEnum["AddToLiableAccount"] = "addToLiableAccount";
        SurchargeEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(SurchargeEnum = SplitConfigurationLogic.SurchargeEnum || (SplitConfigurationLogic.SurchargeEnum = {}));
    let TipEnum;
    (function (TipEnum) {
        TipEnum["AddToLiableAccount"] = "addToLiableAccount";
        TipEnum["AddToOneBalanceAccount"] = "addToOneBalanceAccount";
    })(TipEnum = SplitConfigurationLogic.TipEnum || (SplitConfigurationLogic.TipEnum = {}));
})(SplitConfigurationLogic = exports.SplitConfigurationLogic || (exports.SplitConfigurationLogic = {}));
