// @adyen/api-library/lib/src/typings/management/terminalAssignment.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalAssignment = void 0;
class TerminalAssignment {
    static getAttributeTypeMap() {
        return TerminalAssignment.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TerminalAssignment = TerminalAssignment;
TerminalAssignment.discriminator = undefined;
TerminalAssignment.mapping = undefined;
TerminalAssignment.attributeTypeMap = [
    {
        "name": "companyId",
        "baseName": "companyId",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantId",
        "baseName": "merchantId",
        "type": "string",
        "format": ""
    },
    {
        "name": "reassignmentTarget",
        "baseName": "reassignmentTarget",
        "type": "TerminalReassignmentTarget | null",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "TerminalAssignment.StatusEnum",
        "format": ""
    },
    {
        "name": "storeId",
        "baseName": "storeId",
        "type": "string",
        "format": ""
    }
];
(function (TerminalAssignment) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Boarded"] = "boarded";
        StatusEnum["Deployed"] = "deployed";
        StatusEnum["Inventory"] = "inventory";
        StatusEnum["ReassignmentInProgress"] = "reassignmentInProgress";
    })(StatusEnum = TerminalAssignment.StatusEnum || (TerminalAssignment.StatusEnum = {}));
})(TerminalAssignment = exports.TerminalAssignment || (exports.TerminalAssignment = {}));
