// @adyen/api-library/lib/src/typings/management/updateStoreRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateStoreRequest = void 0;
class UpdateStoreRequest {
    static getAttributeTypeMap() {
        return UpdateStoreRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.UpdateStoreRequest = UpdateStoreRequest;
UpdateStoreRequest.discriminator = undefined;
UpdateStoreRequest.mapping = undefined;
UpdateStoreRequest.attributeTypeMap = [
    {
        "name": "address",
        "baseName": "address",
        "type": "UpdatableAddress | null",
        "format": ""
    },
    {
        "name": "businessLineIds",
        "baseName": "businessLineIds",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "externalReferenceId",
        "baseName": "externalReferenceId",
        "type": "string",
        "format": ""
    },
    {
        "name": "phoneNumber",
        "baseName": "phoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "splitConfiguration",
        "baseName": "splitConfiguration",
        "type": "StoreSplitConfiguration | null",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "UpdateStoreRequest.StatusEnum",
        "format": ""
    },
    {
        "name": "subMerchantData",
        "baseName": "subMerchantData",
        "type": "SubMerchantData | null",
        "format": ""
    }
];
(function (UpdateStoreRequest) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Closed"] = "closed";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = UpdateStoreRequest.StatusEnum || (UpdateStoreRequest.StatusEnum = {}));
})(UpdateStoreRequest = exports.UpdateStoreRequest || (exports.UpdateStoreRequest = {}));
