// @adyen/api-library/lib/src/typings/management/updateMerchantWebhookRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateMerchantWebhookRequest = void 0;
class UpdateMerchantWebhookRequest {
    static getAttributeTypeMap() {
        return UpdateMerchantWebhookRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.UpdateMerchantWebhookRequest = UpdateMerchantWebhookRequest;
UpdateMerchantWebhookRequest.discriminator = undefined;
UpdateMerchantWebhookRequest.mapping = undefined;
UpdateMerchantWebhookRequest.attributeTypeMap = [
    {
        "name": "acceptsExpiredCertificate",
        "baseName": "acceptsExpiredCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "acceptsSelfSignedCertificate",
        "baseName": "acceptsSelfSignedCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "acceptsUntrustedRootCertificate",
        "baseName": "acceptsUntrustedRootCertificate",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "active",
        "baseName": "active",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "additionalSettings",
        "baseName": "additionalSettings",
        "type": "AdditionalSettings | null",
        "format": ""
    },
    {
        "name": "communicationFormat",
        "baseName": "communicationFormat",
        "type": "UpdateMerchantWebhookRequest.CommunicationFormatEnum",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptionProtocol",
        "baseName": "encryptionProtocol",
        "type": "UpdateMerchantWebhookRequest.EncryptionProtocolEnum",
        "format": ""
    },
    {
        "name": "networkType",
        "baseName": "networkType",
        "type": "UpdateMerchantWebhookRequest.NetworkTypeEnum",
        "format": ""
    },
    {
        "name": "password",
        "baseName": "password",
        "type": "string",
        "format": ""
    },
    {
        "name": "populateSoapActionHeader",
        "baseName": "populateSoapActionHeader",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "url",
        "baseName": "url",
        "type": "string",
        "format": ""
    },
    {
        "name": "username",
        "baseName": "username",
        "type": "string",
        "format": ""
    }
];
(function (UpdateMerchantWebhookRequest) {
    let CommunicationFormatEnum;
    (function (CommunicationFormatEnum) {
        CommunicationFormatEnum["Http"] = "http";
        CommunicationFormatEnum["Json"] = "json";
        CommunicationFormatEnum["Soap"] = "soap";
    })(CommunicationFormatEnum = UpdateMerchantWebhookRequest.CommunicationFormatEnum || (UpdateMerchantWebhookRequest.CommunicationFormatEnum = {}));
    let EncryptionProtocolEnum;
    (function (EncryptionProtocolEnum) {
        EncryptionProtocolEnum["Http"] = "HTTP";
        EncryptionProtocolEnum["Tlsv12"] = "TLSv1.2";
        EncryptionProtocolEnum["Tlsv13"] = "TLSv1.3";
    })(EncryptionProtocolEnum = UpdateMerchantWebhookRequest.EncryptionProtocolEnum || (UpdateMerchantWebhookRequest.EncryptionProtocolEnum = {}));
    let NetworkTypeEnum;
    (function (NetworkTypeEnum) {
        NetworkTypeEnum["Local"] = "local";
        NetworkTypeEnum["Public"] = "public";
    })(NetworkTypeEnum = UpdateMerchantWebhookRequest.NetworkTypeEnum || (UpdateMerchantWebhookRequest.NetworkTypeEnum = {}));
})(UpdateMerchantWebhookRequest = exports.UpdateMerchantWebhookRequest || (exports.UpdateMerchantWebhookRequest = {}));
