// @adyen/api-library/lib/src/typings/balancePlatform/accountHolderCapability.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountHolderCapability = void 0;
class AccountHolderCapability {
    static getAttributeTypeMap() {
        return AccountHolderCapability.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AccountHolderCapability = AccountHolderCapability;
AccountHolderCapability.discriminator = undefined;
AccountHolderCapability.mapping = undefined;
AccountHolderCapability.attributeTypeMap = [
    {
        "name": "allowed",
        "baseName": "allowed",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "allowedLevel",
        "baseName": "allowedLevel",
        "type": "AccountHolderCapability.AllowedLevelEnum",
        "format": ""
    },
    {
        "name": "allowedSettings",
        "baseName": "allowedSettings",
        "type": "CapabilitySettings | null",
        "format": ""
    },
    {
        "name": "enabled",
        "baseName": "enabled",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "problems",
        "baseName": "problems",
        "type": "Array<CapabilityProblem>",
        "format": ""
    },
    {
        "name": "requested",
        "baseName": "requested",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "requestedLevel",
        "baseName": "requestedLevel",
        "type": "AccountHolderCapability.RequestedLevelEnum",
        "format": ""
    },
    {
        "name": "requestedSettings",
        "baseName": "requestedSettings",
        "type": "CapabilitySettings | null",
        "format": ""
    },
    {
        "name": "transferInstruments",
        "baseName": "transferInstruments",
        "type": "Array<AccountSupportingEntityCapability>",
        "format": ""
    },
    {
        "name": "verificationStatus",
        "baseName": "verificationStatus",
        "type": "AccountHolderCapability.VerificationStatusEnum",
        "format": ""
    }
];
(function (AccountHolderCapability) {
    let AllowedLevelEnum;
    (function (AllowedLevelEnum) {
        AllowedLevelEnum["High"] = "high";
        AllowedLevelEnum["Low"] = "low";
        AllowedLevelEnum["Medium"] = "medium";
        AllowedLevelEnum["NotApplicable"] = "notApplicable";
    })(AllowedLevelEnum = AccountHolderCapability.AllowedLevelEnum || (AccountHolderCapability.AllowedLevelEnum = {}));
    let RequestedLevelEnum;
    (function (RequestedLevelEnum) {
        RequestedLevelEnum["High"] = "high";
        RequestedLevelEnum["Low"] = "low";
        RequestedLevelEnum["Medium"] = "medium";
        RequestedLevelEnum["NotApplicable"] = "notApplicable";
    })(RequestedLevelEnum = AccountHolderCapability.RequestedLevelEnum || (AccountHolderCapability.RequestedLevelEnum = {}));
    let VerificationStatusEnum;
    (function (VerificationStatusEnum) {
        VerificationStatusEnum["Invalid"] = "invalid";
        VerificationStatusEnum["Pending"] = "pending";
        VerificationStatusEnum["Rejected"] = "rejected";
        VerificationStatusEnum["Valid"] = "valid";
    })(VerificationStatusEnum = AccountHolderCapability.VerificationStatusEnum || (AccountHolderCapability.VerificationStatusEnum = {}));
})(AccountHolderCapability = exports.AccountHolderCapability || (exports.AccountHolderCapability = {}));
