// @adyen/api-library/lib/src/typings/balancePlatform/additionalBankIdentificationRequirement.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalBankIdentificationRequirement = void 0;
class AdditionalBankIdentificationRequirement {
    static getAttributeTypeMap() {
        return AdditionalBankIdentificationRequirement.attributeTypeMap;
    }
    constructor() {
    }
}
exports.AdditionalBankIdentificationRequirement = AdditionalBankIdentificationRequirement;
AdditionalBankIdentificationRequirement.discriminator = undefined;
AdditionalBankIdentificationRequirement.mapping = undefined;
AdditionalBankIdentificationRequirement.attributeTypeMap = [
    {
        "name": "additionalBankIdentificationType",
        "baseName": "additionalBankIdentificationType",
        "type": "AdditionalBankIdentificationRequirement.AdditionalBankIdentificationTypeEnum",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "AdditionalBankIdentificationRequirement.TypeEnum",
        "format": ""
    }
];
(function (AdditionalBankIdentificationRequirement) {
    let AdditionalBankIdentificationTypeEnum;
    (function (AdditionalBankIdentificationTypeEnum) {
        AdditionalBankIdentificationTypeEnum["AuBsbCode"] = "auBsbCode";
        AdditionalBankIdentificationTypeEnum["CaRoutingNumber"] = "caRoutingNumber";
        AdditionalBankIdentificationTypeEnum["GbSortCode"] = "gbSortCode";
        AdditionalBankIdentificationTypeEnum["UsRoutingNumber"] = "usRoutingNumber";
    })(AdditionalBankIdentificationTypeEnum = AdditionalBankIdentificationRequirement.AdditionalBankIdentificationTypeEnum || (AdditionalBankIdentificationRequirement.AdditionalBankIdentificationTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AdditionalBankIdentificationRequirement"] = "additionalBankIdentificationRequirement";
    })(TypeEnum = AdditionalBankIdentificationRequirement.TypeEnum || (AdditionalBankIdentificationRequirement.TypeEnum = {}));
})(AdditionalBankIdentificationRequirement = exports.AdditionalBankIdentificationRequirement || (exports.AdditionalBankIdentificationRequirement = {}));
