// @adyen/api-library/lib/src/typings/balancePlatform/balanceWebhookSettingInfoUpdate.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceWebhookSettingInfoUpdate = void 0;
class BalanceWebhookSettingInfoUpdate {
    static getAttributeTypeMap() {
        return BalanceWebhookSettingInfoUpdate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BalanceWebhookSettingInfoUpdate = BalanceWebhookSettingInfoUpdate;
BalanceWebhookSettingInfoUpdate.discriminator = undefined;
BalanceWebhookSettingInfoUpdate.mapping = undefined;
BalanceWebhookSettingInfoUpdate.attributeTypeMap = [
    {
        "name": "conditions",
        "baseName": "conditions",
        "type": "Array<Condition>",
        "format": ""
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceWebhookSettingInfoUpdate.StatusEnum",
        "format": ""
    },
    {
        "name": "target",
        "baseName": "target",
        "type": "TargetUpdate | null",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BalanceWebhookSettingInfoUpdate.TypeEnum",
        "format": ""
    }
];
(function (BalanceWebhookSettingInfoUpdate) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = BalanceWebhookSettingInfoUpdate.StatusEnum || (BalanceWebhookSettingInfoUpdate.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Balance"] = "balance";
    })(TypeEnum = BalanceWebhookSettingInfoUpdate.TypeEnum || (BalanceWebhookSettingInfoUpdate.TypeEnum = {}));
})(BalanceWebhookSettingInfoUpdate = exports.BalanceWebhookSettingInfoUpdate || (exports.BalanceWebhookSettingInfoUpdate = {}));
