// @adyen/api-library/lib/src/typings/balancePlatform/balanceWebhookSettingInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BalanceWebhookSettingInfo = void 0;
class BalanceWebhookSettingInfo {
    static getAttributeTypeMap() {
        return BalanceWebhookSettingInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.BalanceWebhookSettingInfo = BalanceWebhookSettingInfo;
BalanceWebhookSettingInfo.discriminator = undefined;
BalanceWebhookSettingInfo.mapping = undefined;
BalanceWebhookSettingInfo.attributeTypeMap = [
    {
        "name": "conditions",
        "baseName": "conditions",
        "type": "Array<Condition>",
        "format": ""
    },
    {
        "name": "currency",
        "baseName": "currency",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BalanceWebhookSettingInfo.StatusEnum",
        "format": ""
    },
    {
        "name": "target",
        "baseName": "target",
        "type": "Target",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "BalanceWebhookSettingInfo.TypeEnum",
        "format": ""
    }
];
(function (BalanceWebhookSettingInfo) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = BalanceWebhookSettingInfo.StatusEnum || (BalanceWebhookSettingInfo.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Balance"] = "balance";
    })(TypeEnum = BalanceWebhookSettingInfo.TypeEnum || (BalanceWebhookSettingInfo.TypeEnum = {}));
})(BalanceWebhookSettingInfo = exports.BalanceWebhookSettingInfo || (exports.BalanceWebhookSettingInfo = {}));
