// @adyen/api-library/lib/src/typings/balancePlatform/cardOrder.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardOrder = void 0;
class CardOrder {
    static getAttributeTypeMap() {
        return CardOrder.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CardOrder = CardOrder;
CardOrder.discriminator = undefined;
CardOrder.mapping = undefined;
CardOrder.attributeTypeMap = [
    {
        "name": "beginDate",
        "baseName": "beginDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "cardManufacturingProfileId",
        "baseName": "cardManufacturingProfileId",
        "type": "string",
        "format": ""
    },
    {
        "name": "closedDate",
        "baseName": "closedDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "endDate",
        "baseName": "endDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "lockDate",
        "baseName": "lockDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "serviceCenter",
        "baseName": "serviceCenter",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "CardOrder.StatusEnum",
        "format": ""
    }
];
(function (CardOrder) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Closed"] = "closed";
        StatusEnum["Open"] = "open";
    })(StatusEnum = CardOrder.StatusEnum || (CardOrder.StatusEnum = {}));
})(CardOrder = exports.CardOrder || (exports.CardOrder = {}));
