// @adyen/api-library/lib/src/typings/balancePlatform/condition.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Condition = void 0;
class Condition {
    static getAttributeTypeMap() {
        return Condition.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Condition = Condition;
Condition.discriminator = undefined;
Condition.mapping = undefined;
Condition.attributeTypeMap = [
    {
        "name": "balanceType",
        "baseName": "balanceType",
        "type": "Condition.BalanceTypeEnum",
        "format": ""
    },
    {
        "name": "conditionType",
        "baseName": "conditionType",
        "type": "Condition.ConditionTypeEnum",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "number",
        "format": "int64"
    }
];
(function (Condition) {
    let BalanceTypeEnum;
    (function (BalanceTypeEnum) {
        BalanceTypeEnum["Balance"] = "balance";
        BalanceTypeEnum["Available"] = "available";
        BalanceTypeEnum["Pending"] = "pending";
        BalanceTypeEnum["Reserved"] = "reserved";
    })(BalanceTypeEnum = Condition.BalanceTypeEnum || (Condition.BalanceTypeEnum = {}));
    let ConditionTypeEnum;
    (function (ConditionTypeEnum) {
        ConditionTypeEnum["GreaterThan"] = "greaterThan";
        ConditionTypeEnum["GreaterThanOrEqual"] = "greaterThanOrEqual";
        ConditionTypeEnum["LessThan"] = "lessThan";
        ConditionTypeEnum["LessThanOrEqual"] = "lessThanOrEqual";
    })(ConditionTypeEnum = Condition.ConditionTypeEnum || (Condition.ConditionTypeEnum = {}));
})(Condition = exports.Condition || (exports.Condition = {}));
