// @adyen/api-library/lib/src/typings/balancePlatform/counterpartyTypesRestriction.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CounterpartyTypesRestriction = void 0;
class CounterpartyTypesRestriction {
    static getAttributeTypeMap() {
        return CounterpartyTypesRestriction.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CounterpartyTypesRestriction = CounterpartyTypesRestriction;
CounterpartyTypesRestriction.discriminator = undefined;
CounterpartyTypesRestriction.mapping = undefined;
CounterpartyTypesRestriction.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "CounterpartyTypesRestriction.ValueEnum",
        "format": ""
    }
];
(function (CounterpartyTypesRestriction) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["BalanceAccount"] = "balanceAccount";
        ValueEnum["BankAccount"] = "bankAccount";
        ValueEnum["Card"] = "card";
        ValueEnum["TransferInstrument"] = "transferInstrument";
    })(ValueEnum = CounterpartyTypesRestriction.ValueEnum || (CounterpartyTypesRestriction.ValueEnum = {}));
})(CounterpartyTypesRestriction = exports.CounterpartyTypesRestriction || (exports.CounterpartyTypesRestriction = {}));
