// @adyen/api-library/lib/src/typings/balancePlatform/device.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Device = void 0;
class Device {
    static getAttributeTypeMap() {
        return Device.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Device = Device;
Device.discriminator = undefined;
Device.mapping = undefined;
Device.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Device.TypeEnum",
        "format": ""
    }
];
(function (Device) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Ios"] = "ios";
        TypeEnum["Android"] = "android";
        TypeEnum["Browser"] = "browser";
    })(TypeEnum = Device.TypeEnum || (Device.TypeEnum = {}));
})(Device = exports.Device || (exports.Device = {}));
