// @adyen/api-library/lib/src/typings/balancePlatform/paymentInstrumentRequirement.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentInstrumentRequirement = void 0;
class PaymentInstrumentRequirement {
    static getAttributeTypeMap() {
        return PaymentInstrumentRequirement.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentInstrumentRequirement = PaymentInstrumentRequirement;
PaymentInstrumentRequirement.discriminator = undefined;
PaymentInstrumentRequirement.mapping = undefined;
PaymentInstrumentRequirement.attributeTypeMap = [
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuingCountryCode",
        "baseName": "issuingCountryCode",
        "type": "string",
        "format": ""
    },
    {
        "name": "issuingCountryCodes",
        "baseName": "issuingCountryCodes",
        "type": "Array<string>",
        "format": ""
    },
    {
        "name": "onlyForCrossBalancePlatform",
        "baseName": "onlyForCrossBalancePlatform",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "paymentInstrumentType",
        "baseName": "paymentInstrumentType",
        "type": "PaymentInstrumentRequirement.PaymentInstrumentTypeEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "PaymentInstrumentRequirement.TypeEnum",
        "format": ""
    }
];
(function (PaymentInstrumentRequirement) {
    let PaymentInstrumentTypeEnum;
    (function (PaymentInstrumentTypeEnum) {
        PaymentInstrumentTypeEnum["BankAccount"] = "BankAccount";
        PaymentInstrumentTypeEnum["Card"] = "Card";
    })(PaymentInstrumentTypeEnum = PaymentInstrumentRequirement.PaymentInstrumentTypeEnum || (PaymentInstrumentRequirement.PaymentInstrumentTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["PaymentInstrumentRequirement"] = "paymentInstrumentRequirement";
    })(TypeEnum = PaymentInstrumentRequirement.TypeEnum || (PaymentInstrumentRequirement.TypeEnum = {}));
})(PaymentInstrumentRequirement = exports.PaymentInstrumentRequirement || (exports.PaymentInstrumentRequirement = {}));
