// @adyen/api-library/lib/src/typings/balancePlatform/pinChangeResponse.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PinChangeResponse = void 0;
class PinChangeResponse {
    static getAttributeTypeMap() {
        return PinChangeResponse.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PinChangeResponse = PinChangeResponse;
PinChangeResponse.discriminator = undefined;
PinChangeResponse.mapping = undefined;
PinChangeResponse.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "PinChangeResponse.StatusEnum",
        "format": ""
    }
];
(function (PinChangeResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Completed"] = "completed";
        StatusEnum["Pending"] = "pending";
        StatusEnum["Unavailable"] = "unavailable";
    })(StatusEnum = PinChangeResponse.StatusEnum || (PinChangeResponse.StatusEnum = {}));
})(PinChangeResponse = exports.PinChangeResponse || (exports.PinChangeResponse = {}));
