// @adyen/api-library/lib/src/typings/balancePlatform/sourceAccountTypesRestriction.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceAccountTypesRestriction = void 0;
class SourceAccountTypesRestriction {
    static getAttributeTypeMap() {
        return SourceAccountTypesRestriction.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SourceAccountTypesRestriction = SourceAccountTypesRestriction;
SourceAccountTypesRestriction.discriminator = undefined;
SourceAccountTypesRestriction.mapping = undefined;
SourceAccountTypesRestriction.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "SourceAccountTypesRestriction.ValueEnum",
        "format": ""
    }
];
(function (SourceAccountTypesRestriction) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["BalanceAccount"] = "balanceAccount";
        ValueEnum["BusinessAccount"] = "businessAccount";
    })(ValueEnum = SourceAccountTypesRestriction.ValueEnum || (SourceAccountTypesRestriction.ValueEnum = {}));
})(SourceAccountTypesRestriction = exports.SourceAccountTypesRestriction || (exports.SourceAccountTypesRestriction = {}));
