// @adyen/api-library/lib/src/typings/balancePlatform/targetUpdate.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TargetUpdate = void 0;
class TargetUpdate {
    static getAttributeTypeMap() {
        return TargetUpdate.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TargetUpdate = TargetUpdate;
TargetUpdate.discriminator = undefined;
TargetUpdate.mapping = undefined;
TargetUpdate.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TargetUpdate.TypeEnum",
        "format": ""
    }
];
(function (TargetUpdate) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalanceAccount"] = "balanceAccount";
        TypeEnum["AccountHolder"] = "accountHolder";
        TypeEnum["BalancePlatform"] = "balancePlatform";
    })(TypeEnum = TargetUpdate.TypeEnum || (TargetUpdate.TypeEnum = {}));
})(TargetUpdate = exports.TargetUpdate || (exports.TargetUpdate = {}));
