// @adyen/api-library/lib/src/typings/balancePlatform/target.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Target = void 0;
class Target {
    static getAttributeTypeMap() {
        return Target.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Target = Target;
Target.discriminator = undefined;
Target.mapping = undefined;
Target.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Target.TypeEnum",
        "format": ""
    }
];
(function (Target) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalanceAccount"] = "balanceAccount";
        TypeEnum["AccountHolder"] = "accountHolder";
        TypeEnum["BalancePlatform"] = "balancePlatform";
    })(TypeEnum = Target.TypeEnum || (Target.TypeEnum = {}));
})(Target = exports.Target || (exports.Target = {}));
