// @adyen/api-library/lib/src/typings/balancePlatform/uSLocalAccountIdentification.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.USLocalAccountIdentification = void 0;
class USLocalAccountIdentification {
    static getAttributeTypeMap() {
        return USLocalAccountIdentification.attributeTypeMap;
    }
    constructor() {
    }
}
exports.USLocalAccountIdentification = USLocalAccountIdentification;
USLocalAccountIdentification.discriminator = undefined;
USLocalAccountIdentification.mapping = undefined;
USLocalAccountIdentification.attributeTypeMap = [
    {
        "name": "accountNumber",
        "baseName": "accountNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "accountType",
        "baseName": "accountType",
        "type": "USLocalAccountIdentification.AccountTypeEnum",
        "format": ""
    },
    {
        "name": "routingNumber",
        "baseName": "routingNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "USLocalAccountIdentification.TypeEnum",
        "format": ""
    }
];
(function (USLocalAccountIdentification) {
    let AccountTypeEnum;
    (function (AccountTypeEnum) {
        AccountTypeEnum["Checking"] = "checking";
        AccountTypeEnum["Savings"] = "savings";
    })(AccountTypeEnum = USLocalAccountIdentification.AccountTypeEnum || (USLocalAccountIdentification.AccountTypeEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["UsLocal"] = "usLocal";
    })(TypeEnum = USLocalAccountIdentification.TypeEnum || (USLocalAccountIdentification.TypeEnum = {}));
})(USLocalAccountIdentification = exports.USLocalAccountIdentification || (exports.USLocalAccountIdentification = {}));
