// @adyen/api-library/lib/src/typings/balancePlatform/walletProviderDeviceType.js
"use strict";
/*
 * The version of the OpenAPI document: v2
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletProviderDeviceType = void 0;
class WalletProviderDeviceType {
    static getAttributeTypeMap() {
        return WalletProviderDeviceType.attributeTypeMap;
    }
    constructor() {
    }
}
exports.WalletProviderDeviceType = WalletProviderDeviceType;
WalletProviderDeviceType.discriminator = undefined;
WalletProviderDeviceType.mapping = undefined;
WalletProviderDeviceType.attributeTypeMap = [
    {
        "name": "operation",
        "baseName": "operation",
        "type": "string",
        "format": ""
    },
    {
        "name": "value",
        "baseName": "value",
        "type": "WalletProviderDeviceType.ValueEnum",
        "format": ""
    }
];
(function (WalletProviderDeviceType) {
    let ValueEnum;
    (function (ValueEnum) {
        ValueEnum["Card"] = "CARD";
        ValueEnum["MobilePhone"] = "MOBILE_PHONE";
        ValueEnum["Other"] = "OTHER";
        ValueEnum["Pc"] = "PC";
        ValueEnum["TabletOrEreader"] = "TABLET_OR_EREADER";
        ValueEnum["Unknown"] = "UNKNOWN";
        ValueEnum["WatchOrWristband"] = "WATCH_OR_WRISTBAND";
        ValueEnum["Wearable"] = "WEARABLE";
    })(ValueEnum = WalletProviderDeviceType.ValueEnum || (WalletProviderDeviceType.ValueEnum = {}));
})(WalletProviderDeviceType = exports.WalletProviderDeviceType || (exports.WalletProviderDeviceType = {}));
