// @adyen/api-library/lib/src/typings/managementWebhooks/terminalSettingsData.js
"use strict";
/*
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerminalSettingsData = void 0;
class TerminalSettingsData {
    static getAttributeTypeMap() {
        return TerminalSettingsData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TerminalSettingsData = TerminalSettingsData;
TerminalSettingsData.discriminator = undefined;
TerminalSettingsData.mapping = undefined;
TerminalSettingsData.attributeTypeMap = [
    {
        "name": "companyId",
        "baseName": "companyId",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantId",
        "baseName": "merchantId",
        "type": "string",
        "format": ""
    },
    {
        "name": "storeId",
        "baseName": "storeId",
        "type": "string",
        "format": ""
    },
    {
        "name": "terminalId",
        "baseName": "terminalId",
        "type": "string",
        "format": ""
    },
    {
        "name": "updateSource",
        "baseName": "updateSource",
        "type": "TerminalSettingsData.UpdateSourceEnum",
        "format": ""
    },
    {
        "name": "user",
        "baseName": "user",
        "type": "string",
        "format": ""
    }
];
(function (TerminalSettingsData) {
    let UpdateSourceEnum;
    (function (UpdateSourceEnum) {
        UpdateSourceEnum["CustomerArea"] = "Customer Area";
        UpdateSourceEnum["ManagementApi"] = "Management Api";
    })(UpdateSourceEnum = TerminalSettingsData.UpdateSourceEnum || (TerminalSettingsData.UpdateSourceEnum = {}));
})(TerminalSettingsData = exports.TerminalSettingsData || (exports.TerminalSettingsData = {}));
