// @adyen/api-library/lib/src/typings/payout/recurring.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Recurring = void 0;
class Recurring {
    static getAttributeTypeMap() {
        return Recurring.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Recurring = Recurring;
Recurring.discriminator = undefined;
Recurring.mapping = undefined;
Recurring.attributeTypeMap = [
    {
        "name": "contract",
        "baseName": "contract",
        "type": "Recurring.ContractEnum",
        "format": ""
    },
    {
        "name": "recurringDetailName",
        "baseName": "recurringDetailName",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurringExpiry",
        "baseName": "recurringExpiry",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "recurringFrequency",
        "baseName": "recurringFrequency",
        "type": "string",
        "format": ""
    },
    {
        "name": "tokenService",
        "baseName": "tokenService",
        "type": "Recurring.TokenServiceEnum",
        "format": ""
    }
];
(function (Recurring) {
    let ContractEnum;
    (function (ContractEnum) {
        ContractEnum["Oneclick"] = "ONECLICK";
        ContractEnum["Recurring"] = "RECURRING";
        ContractEnum["Payout"] = "PAYOUT";
    })(ContractEnum = Recurring.ContractEnum || (Recurring.ContractEnum = {}));
    let TokenServiceEnum;
    (function (TokenServiceEnum) {
        TokenServiceEnum["Visatokenservice"] = "VISATOKENSERVICE";
        TokenServiceEnum["Mctokenservice"] = "MCTOKENSERVICE";
        TokenServiceEnum["Amextokenservice"] = "AMEXTOKENSERVICE";
        TokenServiceEnum["TokenSharing"] = "TOKEN_SHARING";
    })(TokenServiceEnum = Recurring.TokenServiceEnum || (Recurring.TokenServiceEnum = {}));
})(Recurring = exports.Recurring || (exports.Recurring = {}));
