// @adyen/api-library/lib/src/typings/payout/submitRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubmitRequest = void 0;
class SubmitRequest {
    static getAttributeTypeMap() {
        return SubmitRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SubmitRequest = SubmitRequest;
SubmitRequest.discriminator = undefined;
SubmitRequest.mapping = undefined;
SubmitRequest.attributeTypeMap = [
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string",
        "format": "date"
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "SubmitRequest.EntityTypeEnum",
        "format": ""
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "nationality",
        "baseName": "nationality",
        "type": "string",
        "format": ""
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "selectedRecurringDetailReference",
        "baseName": "selectedRecurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string",
        "format": ""
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string",
        "format": ""
    }
];
(function (SubmitRequest) {
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["Company"] = "Company";
    })(EntityTypeEnum = SubmitRequest.EntityTypeEnum || (SubmitRequest.EntityTypeEnum = {}));
})(SubmitRequest = exports.SubmitRequest || (exports.SubmitRequest = {}));
