// @adyen/api-library/lib/src/typings/storedValue/storedValueStatusChangeRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueStatusChangeRequest = void 0;
class StoredValueStatusChangeRequest {
    static getAttributeTypeMap() {
        return StoredValueStatusChangeRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.StoredValueStatusChangeRequest = StoredValueStatusChangeRequest;
StoredValueStatusChangeRequest.discriminator = undefined;
StoredValueStatusChangeRequest.mapping = undefined;
StoredValueStatusChangeRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueStatusChangeRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "StoredValueStatusChangeRequest.StatusEnum",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    }
];
(function (StoredValueStatusChangeRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = StoredValueStatusChangeRequest.ShopperInteractionEnum || (StoredValueStatusChangeRequest.ShopperInteractionEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = StoredValueStatusChangeRequest.StatusEnum || (StoredValueStatusChangeRequest.StatusEnum = {}));
})(StoredValueStatusChangeRequest = exports.StoredValueStatusChangeRequest || (exports.StoredValueStatusChangeRequest = {}));
