// @adyen/api-library/lib/src/typings/storedValue/storedValueIssueRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueIssueRequest = void 0;
class StoredValueIssueRequest {
    static getAttributeTypeMap() {
        return StoredValueIssueRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.StoredValueIssueRequest = StoredValueIssueRequest;
StoredValueIssueRequest.discriminator = undefined;
StoredValueIssueRequest.mapping = undefined;
StoredValueIssueRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueIssueRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    }
];
(function (StoredValueIssueRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = StoredValueIssueRequest.ShopperInteractionEnum || (StoredValueIssueRequest.ShopperInteractionEnum = {}));
})(StoredValueIssueRequest = exports.StoredValueIssueRequest || (exports.StoredValueIssueRequest = {}));
