// @adyen/api-library/lib/src/typings/storedValue/storedValueLoadRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v46
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredValueLoadRequest = void 0;
class StoredValueLoadRequest {
    static getAttributeTypeMap() {
        return StoredValueLoadRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.StoredValueLoadRequest = StoredValueLoadRequest;
StoredValueLoadRequest.discriminator = undefined;
StoredValueLoadRequest.mapping = undefined;
StoredValueLoadRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "loadType",
        "baseName": "loadType",
        "type": "StoredValueLoadRequest.LoadTypeEnum",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentMethod",
        "baseName": "paymentMethod",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "recurringDetailReference",
        "baseName": "recurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "StoredValueLoadRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    }
];
(function (StoredValueLoadRequest) {
    let LoadTypeEnum;
    (function (LoadTypeEnum) {
        LoadTypeEnum["MerchandiseReturn"] = "merchandiseReturn";
        LoadTypeEnum["Load"] = "load";
    })(LoadTypeEnum = StoredValueLoadRequest.LoadTypeEnum || (StoredValueLoadRequest.LoadTypeEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = StoredValueLoadRequest.ShopperInteractionEnum || (StoredValueLoadRequest.ShopperInteractionEnum = {}));
})(StoredValueLoadRequest = exports.StoredValueLoadRequest || (exports.StoredValueLoadRequest = {}));
