// @adyen/api-library/lib/src/typings/payment/paymentRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentRequest = void 0;
class PaymentRequest {
    static getAttributeTypeMap() {
        return PaymentRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.PaymentRequest = PaymentRequest;
PaymentRequest.discriminator = undefined;
PaymentRequest.mapping = undefined;
PaymentRequest.attributeTypeMap = [
    {
        "name": "accountInfo",
        "baseName": "accountInfo",
        "type": "AccountInfo | null",
        "format": ""
    },
    {
        "name": "additionalAmount",
        "baseName": "additionalAmount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "additionalData",
        "baseName": "additionalData",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "applicationInfo",
        "baseName": "applicationInfo",
        "type": "ApplicationInfo | null",
        "format": ""
    },
    {
        "name": "bankAccount",
        "baseName": "bankAccount",
        "type": "BankAccount | null",
        "format": ""
    },
    {
        "name": "billingAddress",
        "baseName": "billingAddress",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "browserInfo",
        "baseName": "browserInfo",
        "type": "BrowserInfo | null",
        "format": ""
    },
    {
        "name": "captureDelayHours",
        "baseName": "captureDelayHours",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "card",
        "baseName": "card",
        "type": "Card | null",
        "format": ""
    },
    {
        "name": "dateOfBirth",
        "baseName": "dateOfBirth",
        "type": "string",
        "format": "date"
    },
    {
        "name": "dccQuote",
        "baseName": "dccQuote",
        "type": "ForexQuote | null",
        "format": ""
    },
    {
        "name": "deliveryAddress",
        "baseName": "deliveryAddress",
        "type": "Address | null",
        "format": ""
    },
    {
        "name": "deliveryDate",
        "baseName": "deliveryDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "deviceFingerprint",
        "baseName": "deviceFingerprint",
        "type": "string",
        "format": ""
    },
    {
        "name": "entityType",
        "baseName": "entityType",
        "type": "PaymentRequest.EntityTypeEnum",
        "format": ""
    },
    {
        "name": "fraudOffset",
        "baseName": "fraudOffset",
        "type": "number",
        "format": "int32"
    },
    {
        "name": "fundDestination",
        "baseName": "fundDestination",
        "type": "FundDestination | null",
        "format": ""
    },
    {
        "name": "fundSource",
        "baseName": "fundSource",
        "type": "FundSource | null",
        "format": ""
    },
    {
        "name": "fundingSource",
        "baseName": "fundingSource",
        "type": "PaymentRequest.FundingSourceEnum",
        "format": ""
    },
    {
        "name": "installments",
        "baseName": "installments",
        "type": "Installments | null",
        "format": ""
    },
    {
        "name": "localizedShopperStatement",
        "baseName": "localizedShopperStatement",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "mandate",
        "baseName": "mandate",
        "type": "Mandate | null",
        "format": ""
    },
    {
        "name": "mcc",
        "baseName": "mcc",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantOrderReference",
        "baseName": "merchantOrderReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantRiskIndicator",
        "baseName": "merchantRiskIndicator",
        "type": "MerchantRiskIndicator | null",
        "format": ""
    },
    {
        "name": "metadata",
        "baseName": "metadata",
        "type": "{ [key: string]: string; }",
        "format": ""
    },
    {
        "name": "mpiData",
        "baseName": "mpiData",
        "type": "ThreeDSecureData | null",
        "format": ""
    },
    {
        "name": "nationality",
        "baseName": "nationality",
        "type": "string",
        "format": ""
    },
    {
        "name": "orderReference",
        "baseName": "orderReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "platformChargebackLogic",
        "baseName": "platformChargebackLogic",
        "type": "PlatformChargebackLogic | null",
        "format": ""
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring | null",
        "format": ""
    },
    {
        "name": "recurringProcessingModel",
        "baseName": "recurringProcessingModel",
        "type": "PaymentRequest.RecurringProcessingModelEnum",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "secureRemoteCommerceCheckoutData",
        "baseName": "secureRemoteCommerceCheckoutData",
        "type": "SecureRemoteCommerceCheckoutData | null",
        "format": ""
    },
    {
        "name": "selectedBrand",
        "baseName": "selectedBrand",
        "type": "string",
        "format": ""
    },
    {
        "name": "selectedRecurringDetailReference",
        "baseName": "selectedRecurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "sessionId",
        "baseName": "sessionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperEmail",
        "baseName": "shopperEmail",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperIP",
        "baseName": "shopperIP",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "PaymentRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperLocale",
        "baseName": "shopperLocale",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperName",
        "baseName": "shopperName",
        "type": "Name | null",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperStatement",
        "baseName": "shopperStatement",
        "type": "string",
        "format": ""
    },
    {
        "name": "socialSecurityNumber",
        "baseName": "socialSecurityNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "splits",
        "baseName": "splits",
        "type": "Array<Split>",
        "format": ""
    },
    {
        "name": "store",
        "baseName": "store",
        "type": "string",
        "format": ""
    },
    {
        "name": "telephoneNumber",
        "baseName": "telephoneNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "threeDS2RequestData",
        "baseName": "threeDS2RequestData",
        "type": "ThreeDS2RequestData | null",
        "format": ""
    },
    {
        "name": "threeDSAuthenticationOnly",
        "baseName": "threeDSAuthenticationOnly",
        "type": "boolean",
        "format": ""
    },
    {
        "name": "totalsGroup",
        "baseName": "totalsGroup",
        "type": "string",
        "format": ""
    },
    {
        "name": "trustedShopper",
        "baseName": "trustedShopper",
        "type": "boolean",
        "format": ""
    }
];
(function (PaymentRequest) {
    let EntityTypeEnum;
    (function (EntityTypeEnum) {
        EntityTypeEnum["NaturalPerson"] = "NaturalPerson";
        EntityTypeEnum["CompanyName"] = "CompanyName";
    })(EntityTypeEnum = PaymentRequest.EntityTypeEnum || (PaymentRequest.EntityTypeEnum = {}));
    let FundingSourceEnum;
    (function (FundingSourceEnum) {
        FundingSourceEnum["Credit"] = "credit";
        FundingSourceEnum["Debit"] = "debit";
    })(FundingSourceEnum = PaymentRequest.FundingSourceEnum || (PaymentRequest.FundingSourceEnum = {}));
    let RecurringProcessingModelEnum;
    (function (RecurringProcessingModelEnum) {
        RecurringProcessingModelEnum["CardOnFile"] = "CardOnFile";
        RecurringProcessingModelEnum["Subscription"] = "Subscription";
        RecurringProcessingModelEnum["UnscheduledCardOnFile"] = "UnscheduledCardOnFile";
    })(RecurringProcessingModelEnum = PaymentRequest.RecurringProcessingModelEnum || (PaymentRequest.RecurringProcessingModelEnum = {}));
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = PaymentRequest.ShopperInteractionEnum || (PaymentRequest.ShopperInteractionEnum = {}));
})(PaymentRequest = exports.PaymentRequest || (exports.PaymentRequest = {}));
