// @adyen/api-library/lib/src/typings/payment/secureRemoteCommerceCheckoutData.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecureRemoteCommerceCheckoutData = void 0;
class SecureRemoteCommerceCheckoutData {
    static getAttributeTypeMap() {
        return SecureRemoteCommerceCheckoutData.attributeTypeMap;
    }
    constructor() {
    }
}
exports.SecureRemoteCommerceCheckoutData = SecureRemoteCommerceCheckoutData;
SecureRemoteCommerceCheckoutData.discriminator = undefined;
SecureRemoteCommerceCheckoutData.mapping = undefined;
SecureRemoteCommerceCheckoutData.attributeTypeMap = [
    {
        "name": "checkoutPayload",
        "baseName": "checkoutPayload",
        "type": "string",
        "format": ""
    },
    {
        "name": "correlationId",
        "baseName": "correlationId",
        "type": "string",
        "format": ""
    },
    {
        "name": "cvc",
        "baseName": "cvc",
        "type": "string",
        "format": ""
    },
    {
        "name": "digitalCardId",
        "baseName": "digitalCardId",
        "type": "string",
        "format": ""
    },
    {
        "name": "scheme",
        "baseName": "scheme",
        "type": "SecureRemoteCommerceCheckoutData.SchemeEnum",
        "format": ""
    },
    {
        "name": "tokenReference",
        "baseName": "tokenReference",
        "type": "string",
        "format": ""
    }
];
(function (SecureRemoteCommerceCheckoutData) {
    let SchemeEnum;
    (function (SchemeEnum) {
        SchemeEnum["Mc"] = "mc";
        SchemeEnum["Visa"] = "visa";
    })(SchemeEnum = SecureRemoteCommerceCheckoutData.SchemeEnum || (SecureRemoteCommerceCheckoutData.SchemeEnum = {}));
})(SecureRemoteCommerceCheckoutData = exports.SecureRemoteCommerceCheckoutData || (exports.SecureRemoteCommerceCheckoutData = {}));
