// @adyen/api-library/lib/src/typings/payment/split.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Split = void 0;
class Split {
    static getAttributeTypeMap() {
        return Split.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Split = Split;
Split.discriminator = undefined;
Split.mapping = undefined;
Split.attributeTypeMap = [
    {
        "name": "account",
        "baseName": "account",
        "type": "string",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "SplitAmount | null",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Split.TypeEnum",
        "format": ""
    }
];
(function (Split) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["AcquiringFees"] = "AcquiringFees";
        TypeEnum["AdyenCommission"] = "AdyenCommission";
        TypeEnum["AdyenFees"] = "AdyenFees";
        TypeEnum["AdyenMarkup"] = "AdyenMarkup";
        TypeEnum["BalanceAccount"] = "BalanceAccount";
        TypeEnum["Commission"] = "Commission";
        TypeEnum["Default"] = "Default";
        TypeEnum["Interchange"] = "Interchange";
        TypeEnum["MarketPlace"] = "MarketPlace";
        TypeEnum["PaymentFee"] = "PaymentFee";
        TypeEnum["Remainder"] = "Remainder";
        TypeEnum["SchemeFee"] = "SchemeFee";
        TypeEnum["Surcharge"] = "Surcharge";
        TypeEnum["Tip"] = "Tip";
        TypeEnum["TopUp"] = "TopUp";
        TypeEnum["Vat"] = "VAT";
    })(TypeEnum = Split.TypeEnum || (Split.TypeEnum = {}));
})(Split = exports.Split || (exports.Split = {}));
