// @adyen/api-library/lib/src/typings/transferWebhooks/transferWebhooksHandler.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferWebhooksHandler = void 0;
const __1 = require("..");
/**
 * Handler for processing TransferWebhooks.
 *
 * This class provides functionality to deserialize the payload of TransferWebhooks events.
 */
class TransferWebhooksHandler {
    constructor(jsonPayload) {
        this.payload = JSON.parse(jsonPayload);
    }
    /**
     * This method checks the type of the webhook payload and returns the appropriate deserialized object.
     *
     * @returns A deserialized object of type GenericWebhook.
     * @throws Error if the type is not recognized.
     */
    getGenericWebhook() {
        const type = this.payload["type"];
        if (Object.values(__1.transferWebhooks.TransferNotificationRequest.TypeEnum).includes(type)) {
            return this.getTransferNotificationRequest();
        }
        throw new Error("Could not parse the json payload: " + this.payload);
    }
    /**
     * Deserialize the webhook payload into a TransferNotificationRequest
     *
     * @returns Deserialized TransferNotificationRequest object.
     */
    getTransferNotificationRequest() {
        return __1.transferWebhooks.ObjectSerializer.deserialize(this.payload, "TransferNotificationRequest");
    }
}
exports.TransferWebhooksHandler = TransferWebhooksHandler;
