// @adyen/api-library/lib/src/typings/transferWebhooks/transferNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferNotificationRequest = void 0;
class TransferNotificationRequest {
    static getAttributeTypeMap() {
        return TransferNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TransferNotificationRequest = TransferNotificationRequest;
TransferNotificationRequest.discriminator = undefined;
TransferNotificationRequest.mapping = undefined;
TransferNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "TransferData",
        "format": ""
    },
    {
        "name": "environment",
        "baseName": "environment",
        "type": "string",
        "format": ""
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (TransferNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformTransferCreated"] = "balancePlatform.transfer.created";
        TypeEnum["BalancePlatformTransferUpdated"] = "balancePlatform.transfer.updated";
    })(TypeEnum = TransferNotificationRequest.TypeEnum || (TransferNotificationRequest.TypeEnum = {}));
})(TransferNotificationRequest = exports.TransferNotificationRequest || (exports.TransferNotificationRequest = {}));
