// @adyen/api-library/lib/src/typings/disputeWebhooks/disputeWebhooksHandler.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisputeWebhooksHandler = void 0;
const __1 = require("..");
/**
 * Handler for processing DisputeWebhooks.
 *
 * This class provides functionality to deserialize the payload of DisputeWebhooks events.
 */
class DisputeWebhooksHandler {
    constructor(jsonPayload) {
        this.payload = JSON.parse(jsonPayload);
    }
    /**
     * This method checks the type of the webhook payload and returns the appropriate deserialized object.
     *
     * @returns A deserialized object of type GenericWebhook.
     * @throws Error if the type is not recognized.
     */
    getGenericWebhook() {
        const type = this.payload["type"];
        if (Object.values(__1.disputeWebhooks.DisputeNotificationRequest.TypeEnum).includes(type)) {
            return this.getDisputeNotificationRequest();
        }
        throw new Error("Could not parse the json payload: " + this.payload);
    }
    /**
     * Deserialize the webhook payload into a DisputeNotificationRequest
     *
     * @returns Deserialized DisputeNotificationRequest object.
     */
    getDisputeNotificationRequest() {
        return __1.disputeWebhooks.ObjectSerializer.deserialize(this.payload, "DisputeNotificationRequest");
    }
}
exports.DisputeWebhooksHandler = DisputeWebhooksHandler;
