// @adyen/api-library/lib/src/typings/disputeWebhooks/disputeNotificationRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisputeNotificationRequest = void 0;
class DisputeNotificationRequest {
    static getAttributeTypeMap() {
        return DisputeNotificationRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DisputeNotificationRequest = DisputeNotificationRequest;
DisputeNotificationRequest.discriminator = undefined;
DisputeNotificationRequest.mapping = undefined;
DisputeNotificationRequest.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "DisputeEventNotification",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DisputeNotificationRequest.TypeEnum",
        "format": ""
    }
];
(function (DisputeNotificationRequest) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BalancePlatformDisputeCreated"] = "balancePlatform.dispute.created";
        TypeEnum["BalancePlatformDisputeUpdated"] = "balancePlatform.dispute.updated";
    })(TypeEnum = DisputeNotificationRequest.TypeEnum || (DisputeNotificationRequest.TypeEnum = {}));
})(DisputeNotificationRequest = exports.DisputeNotificationRequest || (exports.DisputeNotificationRequest = {}));
