// @adyen/api-library/lib/src/typings/disputeWebhooks/disputeEventNotification.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisputeEventNotification = void 0;
class DisputeEventNotification {
    static getAttributeTypeMap() {
        return DisputeEventNotification.attributeTypeMap;
    }
    constructor() {
    }
}
exports.DisputeEventNotification = DisputeEventNotification;
DisputeEventNotification.discriminator = undefined;
DisputeEventNotification.mapping = undefined;
DisputeEventNotification.attributeTypeMap = [
    {
        "name": "arn",
        "baseName": "arn",
        "type": "string",
        "format": ""
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string",
        "format": ""
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "disputedAmount",
        "baseName": "disputedAmount",
        "type": "Amount | null",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "string",
        "format": ""
    },
    {
        "name": "statusDetail",
        "baseName": "statusDetail",
        "type": "string",
        "format": ""
    },
    {
        "name": "transactionId",
        "baseName": "transactionId",
        "type": "string",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "DisputeEventNotification.TypeEnum",
        "format": ""
    }
];
(function (DisputeEventNotification) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Fraud"] = "fraud";
        TypeEnum["NotDelivered"] = "notDelivered";
        TypeEnum["Duplicate"] = "duplicate";
    })(TypeEnum = DisputeEventNotification.TypeEnum || (DisputeEventNotification.TypeEnum = {}));
})(DisputeEventNotification = exports.DisputeEventNotification || (exports.DisputeEventNotification = {}));
