// @adyen/api-library/lib/src/typings/binLookup/costEstimateRequest.js
"use strict";
/*
 * The version of the OpenAPI document: v54
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CostEstimateRequest = void 0;
class CostEstimateRequest {
    static getAttributeTypeMap() {
        return CostEstimateRequest.attributeTypeMap;
    }
    constructor() {
    }
}
exports.CostEstimateRequest = CostEstimateRequest;
CostEstimateRequest.discriminator = undefined;
CostEstimateRequest.mapping = undefined;
CostEstimateRequest.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "assumptions",
        "baseName": "assumptions",
        "type": "CostEstimateAssumptions | null",
        "format": ""
    },
    {
        "name": "cardNumber",
        "baseName": "cardNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "encryptedCardNumber",
        "baseName": "encryptedCardNumber",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string",
        "format": ""
    },
    {
        "name": "merchantDetails",
        "baseName": "merchantDetails",
        "type": "MerchantDetails | null",
        "format": ""
    },
    {
        "name": "recurring",
        "baseName": "recurring",
        "type": "Recurring | null",
        "format": ""
    },
    {
        "name": "selectedRecurringDetailReference",
        "baseName": "selectedRecurringDetailReference",
        "type": "string",
        "format": ""
    },
    {
        "name": "shopperInteraction",
        "baseName": "shopperInteraction",
        "type": "CostEstimateRequest.ShopperInteractionEnum",
        "format": ""
    },
    {
        "name": "shopperReference",
        "baseName": "shopperReference",
        "type": "string",
        "format": ""
    }
];
(function (CostEstimateRequest) {
    let ShopperInteractionEnum;
    (function (ShopperInteractionEnum) {
        ShopperInteractionEnum["Ecommerce"] = "Ecommerce";
        ShopperInteractionEnum["ContAuth"] = "ContAuth";
        ShopperInteractionEnum["Moto"] = "Moto";
        ShopperInteractionEnum["Pos"] = "POS";
    })(ShopperInteractionEnum = CostEstimateRequest.ShopperInteractionEnum || (CostEstimateRequest.ShopperInteractionEnum = {}));
})(CostEstimateRequest = exports.CostEstimateRequest || (exports.CostEstimateRequest = {}));
