// @adyen/api-library/lib/src/services/recurring/recurringApi.js
"use strict";
/*
 * The version of the OpenAPI document: v68
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecurringApi = void 0;
const getJsonResponse_1 = __importDefault(require("../../helpers/getJsonResponse"));
const service_1 = __importDefault(require("../../service"));
const resource_1 = __importDefault(require("../resource"));
const objectSerializer_1 = require("../../typings/recurring/objectSerializer");
/**
 * API handler for RecurringApi
 */
class RecurringApi extends service_1.default {
    constructor(client) {
        super(client);
        this.API_BASEPATH = "https://pal-test.adyen.com/pal/servlet/Recurring/v68";
        this.baseUrl = this.createBaseUrl(this.API_BASEPATH);
    }
    /**
    * @summary Create new permits linked to a recurring contract.
    * @param createPermitRequest {@link CreatePermitRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link CreatePermitResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async createPermit(createPermitRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/createPermit`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(createPermitRequest, "CreatePermitRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "CreatePermitResult");
    }
    /**
    * @summary Disable stored payment details
    * @param disableRequest {@link DisableRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link DisableResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async disable(disableRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/disable`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(disableRequest, "DisableRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "DisableResult");
    }
    /**
    * @summary Disable an existing permit.
    * @param disablePermitRequest {@link DisablePermitRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link DisablePermitResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async disablePermit(disablePermitRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/disablePermit`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(disablePermitRequest, "DisablePermitRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "DisablePermitResult");
    }
    /**
    * @summary Get stored payment details
    * @param recurringDetailsRequest {@link RecurringDetailsRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link RecurringDetailsResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async listRecurringDetails(recurringDetailsRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/listRecurringDetails`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(recurringDetailsRequest, "RecurringDetailsRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "RecurringDetailsResult");
    }
    /**
    * @summary Ask issuer to notify the shopper
    * @param notifyShopperRequest {@link NotifyShopperRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link NotifyShopperResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async notifyShopper(notifyShopperRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/notifyShopper`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(notifyShopperRequest, "NotifyShopperRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "NotifyShopperResult");
    }
    /**
    * @summary Schedule running the Account Updater
    * @param scheduleAccountUpdaterRequest {@link ScheduleAccountUpdaterRequest }
    * @param requestOptions {@link IRequest.Options }
    * @return {@link ScheduleAccountUpdaterResult }
    *
    * @deprecated since Adyen Recurring API v68
    */
    async scheduleAccountUpdater(scheduleAccountUpdaterRequest, requestOptions) {
        const endpoint = `${this.baseUrl}/scheduleAccountUpdater`;
        const resource = new resource_1.default(this, endpoint);
        const request = objectSerializer_1.ObjectSerializer.serialize(scheduleAccountUpdaterRequest, "ScheduleAccountUpdaterRequest");
        const response = await (0, getJsonResponse_1.default)(resource, request, { ...requestOptions, method: "POST" });
        return objectSerializer_1.ObjectSerializer.deserialize(response, "ScheduleAccountUpdaterResult");
    }
}
exports.RecurringApi = RecurringApi;
