// @adyen/api-library/lib/src/typings/transfers/transferInfo.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferInfo = void 0;
class TransferInfo {
    static getAttributeTypeMap() {
        return TransferInfo.attributeTypeMap;
    }
    constructor() {
    }
}
exports.TransferInfo = TransferInfo;
TransferInfo.discriminator = undefined;
TransferInfo.mapping = undefined;
TransferInfo.attributeTypeMap = [
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string",
        "format": ""
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "TransferInfo.CategoryEnum",
        "format": ""
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyInfoV3",
        "format": ""
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "executionDate",
        "baseName": "executionDate",
        "type": "ExecutionDate | null",
        "format": ""
    },
    {
        "name": "paymentInstrumentId",
        "baseName": "paymentInstrumentId",
        "type": "string",
        "format": ""
    },
    {
        "name": "priorities",
        "baseName": "priorities",
        "type": "TransferInfo.PrioritiesEnum",
        "format": ""
    },
    {
        "name": "priority",
        "baseName": "priority",
        "type": "TransferInfo.PriorityEnum",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string",
        "format": ""
    },
    {
        "name": "review",
        "baseName": "review",
        "type": "TransferRequestReview | null",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "TransferInfo.TypeEnum",
        "format": ""
    },
    {
        "name": "ultimateParty",
        "baseName": "ultimateParty",
        "type": "UltimatePartyIdentification | null",
        "format": ""
    }
];
(function (TransferInfo) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Card"] = "card";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["IssuedCard"] = "issuedCard";
        CategoryEnum["PlatformPayment"] = "platformPayment";
        CategoryEnum["TopUp"] = "topUp";
    })(CategoryEnum = TransferInfo.CategoryEnum || (TransferInfo.CategoryEnum = {}));
    let PrioritiesEnum;
    (function (PrioritiesEnum) {
        PrioritiesEnum["CrossBorder"] = "crossBorder";
        PrioritiesEnum["Fast"] = "fast";
        PrioritiesEnum["Instant"] = "instant";
        PrioritiesEnum["Internal"] = "internal";
        PrioritiesEnum["Regular"] = "regular";
        PrioritiesEnum["Wire"] = "wire";
    })(PrioritiesEnum = TransferInfo.PrioritiesEnum || (TransferInfo.PrioritiesEnum = {}));
    let PriorityEnum;
    (function (PriorityEnum) {
        PriorityEnum["CrossBorder"] = "crossBorder";
        PriorityEnum["Fast"] = "fast";
        PriorityEnum["Instant"] = "instant";
        PriorityEnum["Internal"] = "internal";
        PriorityEnum["Regular"] = "regular";
        PriorityEnum["Wire"] = "wire";
    })(PriorityEnum = TransferInfo.PriorityEnum || (TransferInfo.PriorityEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["BankTransfer"] = "bankTransfer";
        TypeEnum["InternalTransfer"] = "internalTransfer";
        TypeEnum["InternalDirectDebit"] = "internalDirectDebit";
    })(TypeEnum = TransferInfo.TypeEnum || (TransferInfo.TypeEnum = {}));
})(TransferInfo = exports.TransferInfo || (exports.TransferInfo = {}));
