// @adyen/api-library/lib/src/typings/transfers/transfer.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transfer = void 0;
class Transfer {
    static getAttributeTypeMap() {
        return Transfer.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Transfer = Transfer;
Transfer.discriminator = undefined;
Transfer.mapping = undefined;
Transfer.attributeTypeMap = [
    {
        "name": "accountHolder",
        "baseName": "accountHolder",
        "type": "ResourceReference | null",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "balanceAccount",
        "baseName": "balanceAccount",
        "type": "ResourceReference | null",
        "format": ""
    },
    {
        "name": "category",
        "baseName": "category",
        "type": "Transfer.CategoryEnum",
        "format": ""
    },
    {
        "name": "categoryData",
        "baseName": "categoryData",
        "type": "TransferCategoryData | null",
        "format": ""
    },
    {
        "name": "counterparty",
        "baseName": "counterparty",
        "type": "CounterpartyV3",
        "format": ""
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "directDebitInformation",
        "baseName": "directDebitInformation",
        "type": "DirectDebitInformation | null",
        "format": ""
    },
    {
        "name": "direction",
        "baseName": "direction",
        "type": "Transfer.DirectionEnum",
        "format": ""
    },
    {
        "name": "executionDate",
        "baseName": "executionDate",
        "type": "ExecutionDate | null",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentInstrument",
        "baseName": "paymentInstrument",
        "type": "PaymentInstrument | null",
        "format": ""
    },
    {
        "name": "reason",
        "baseName": "reason",
        "type": "Transfer.ReasonEnum",
        "format": ""
    },
    {
        "name": "reference",
        "baseName": "reference",
        "type": "string",
        "format": ""
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string",
        "format": ""
    },
    {
        "name": "review",
        "baseName": "review",
        "type": "TransferReview | null",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Transfer.StatusEnum",
        "format": ""
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "Transfer.TypeEnum",
        "format": ""
    }
];
(function (Transfer) {
    let CategoryEnum;
    (function (CategoryEnum) {
        CategoryEnum["Bank"] = "bank";
        CategoryEnum["Card"] = "card";
        CategoryEnum["Internal"] = "internal";
        CategoryEnum["IssuedCard"] = "issuedCard";
        CategoryEnum["PlatformPayment"] = "platformPayment";
        CategoryEnum["TopUp"] = "topUp";
    })(CategoryEnum = Transfer.CategoryEnum || (Transfer.CategoryEnum = {}));
    let DirectionEnum;
    (function (DirectionEnum) {
        DirectionEnum["Incoming"] = "incoming";
        DirectionEnum["Outgoing"] = "outgoing";
    })(DirectionEnum = Transfer.DirectionEnum || (Transfer.DirectionEnum = {}));
    let ReasonEnum;
    (function (ReasonEnum) {
        ReasonEnum["AccountHierarchyNotActive"] = "accountHierarchyNotActive";
        ReasonEnum["AmountLimitExceeded"] = "amountLimitExceeded";
        ReasonEnum["ApprovalExpired"] = "approvalExpired";
        ReasonEnum["Approved"] = "approved";
        ReasonEnum["BalanceAccountTemporarilyBlockedByTransactionRule"] = "balanceAccountTemporarilyBlockedByTransactionRule";
        ReasonEnum["CounterpartyAccountBlocked"] = "counterpartyAccountBlocked";
        ReasonEnum["CounterpartyAccountClosed"] = "counterpartyAccountClosed";
        ReasonEnum["CounterpartyAccountNotFound"] = "counterpartyAccountNotFound";
        ReasonEnum["CounterpartyAddressRequired"] = "counterpartyAddressRequired";
        ReasonEnum["CounterpartyBankTimedOut"] = "counterpartyBankTimedOut";
        ReasonEnum["CounterpartyBankUnavailable"] = "counterpartyBankUnavailable";
        ReasonEnum["Declined"] = "declined";
        ReasonEnum["DeclinedByTransactionRule"] = "declinedByTransactionRule";
        ReasonEnum["DirectDebitNotSupported"] = "directDebitNotSupported";
        ReasonEnum["Error"] = "error";
        ReasonEnum["NotEnoughBalance"] = "notEnoughBalance";
        ReasonEnum["Pending"] = "pending";
        ReasonEnum["PendingApproval"] = "pendingApproval";
        ReasonEnum["PendingExecution"] = "pendingExecution";
        ReasonEnum["RefusedByCounterpartyBank"] = "refusedByCounterpartyBank";
        ReasonEnum["RefusedByCustomer"] = "refusedByCustomer";
        ReasonEnum["RouteNotFound"] = "routeNotFound";
        ReasonEnum["ScaFailed"] = "scaFailed";
        ReasonEnum["TransferInstrumentDoesNotExist"] = "transferInstrumentDoesNotExist";
        ReasonEnum["Unknown"] = "unknown";
    })(ReasonEnum = Transfer.ReasonEnum || (Transfer.ReasonEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["ApprovalPending"] = "approvalPending";
        StatusEnum["AtmWithdrawal"] = "atmWithdrawal";
        StatusEnum["AtmWithdrawalReversalPending"] = "atmWithdrawalReversalPending";
        StatusEnum["AtmWithdrawalReversed"] = "atmWithdrawalReversed";
        StatusEnum["AuthAdjustmentAuthorised"] = "authAdjustmentAuthorised";
        StatusEnum["AuthAdjustmentError"] = "authAdjustmentError";
        StatusEnum["AuthAdjustmentRefused"] = "authAdjustmentRefused";
        StatusEnum["Authorised"] = "authorised";
        StatusEnum["BankTransfer"] = "bankTransfer";
        StatusEnum["BankTransferPending"] = "bankTransferPending";
        StatusEnum["Booked"] = "booked";
        StatusEnum["BookingPending"] = "bookingPending";
        StatusEnum["Cancelled"] = "cancelled";
        StatusEnum["CapturePending"] = "capturePending";
        StatusEnum["CaptureReversalPending"] = "captureReversalPending";
        StatusEnum["CaptureReversed"] = "captureReversed";
        StatusEnum["Captured"] = "captured";
        StatusEnum["CapturedExternally"] = "capturedExternally";
        StatusEnum["Chargeback"] = "chargeback";
        StatusEnum["ChargebackExternally"] = "chargebackExternally";
        StatusEnum["ChargebackPending"] = "chargebackPending";
        StatusEnum["ChargebackReversalPending"] = "chargebackReversalPending";
        StatusEnum["ChargebackReversed"] = "chargebackReversed";
        StatusEnum["Credited"] = "credited";
        StatusEnum["DepositCorrection"] = "depositCorrection";
        StatusEnum["DepositCorrectionPending"] = "depositCorrectionPending";
        StatusEnum["Dispute"] = "dispute";
        StatusEnum["DisputeClosed"] = "disputeClosed";
        StatusEnum["DisputeExpired"] = "disputeExpired";
        StatusEnum["DisputeNeedsReview"] = "disputeNeedsReview";
        StatusEnum["Error"] = "error";
        StatusEnum["Expired"] = "expired";
        StatusEnum["Failed"] = "failed";
        StatusEnum["Fee"] = "fee";
        StatusEnum["FeePending"] = "feePending";
        StatusEnum["InternalTransfer"] = "internalTransfer";
        StatusEnum["InternalTransferPending"] = "internalTransferPending";
        StatusEnum["InvoiceDeduction"] = "invoiceDeduction";
        StatusEnum["InvoiceDeductionPending"] = "invoiceDeductionPending";
        StatusEnum["ManualCorrectionPending"] = "manualCorrectionPending";
        StatusEnum["ManuallyCorrected"] = "manuallyCorrected";
        StatusEnum["MatchedStatement"] = "matchedStatement";
        StatusEnum["MatchedStatementPending"] = "matchedStatementPending";
        StatusEnum["MerchantPayin"] = "merchantPayin";
        StatusEnum["MerchantPayinPending"] = "merchantPayinPending";
        StatusEnum["MerchantPayinReversed"] = "merchantPayinReversed";
        StatusEnum["MerchantPayinReversedPending"] = "merchantPayinReversedPending";
        StatusEnum["MiscCost"] = "miscCost";
        StatusEnum["MiscCostPending"] = "miscCostPending";
        StatusEnum["PaymentCost"] = "paymentCost";
        StatusEnum["PaymentCostPending"] = "paymentCostPending";
        StatusEnum["PendingApproval"] = "pendingApproval";
        StatusEnum["PendingExecution"] = "pendingExecution";
        StatusEnum["Received"] = "received";
        StatusEnum["RefundPending"] = "refundPending";
        StatusEnum["RefundReversalPending"] = "refundReversalPending";
        StatusEnum["RefundReversed"] = "refundReversed";
        StatusEnum["Refunded"] = "refunded";
        StatusEnum["RefundedExternally"] = "refundedExternally";
        StatusEnum["Refused"] = "refused";
        StatusEnum["Rejected"] = "rejected";
        StatusEnum["ReserveAdjustment"] = "reserveAdjustment";
        StatusEnum["ReserveAdjustmentPending"] = "reserveAdjustmentPending";
        StatusEnum["Returned"] = "returned";
        StatusEnum["SecondChargeback"] = "secondChargeback";
        StatusEnum["SecondChargebackPending"] = "secondChargebackPending";
        StatusEnum["Undefined"] = "undefined";
    })(StatusEnum = Transfer.StatusEnum || (Transfer.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Payment"] = "payment";
        TypeEnum["Capture"] = "capture";
        TypeEnum["CaptureReversal"] = "captureReversal";
        TypeEnum["Refund"] = "refund";
        TypeEnum["RefundReversal"] = "refundReversal";
        TypeEnum["Chargeback"] = "chargeback";
        TypeEnum["ChargebackCorrection"] = "chargebackCorrection";
        TypeEnum["ChargebackReversal"] = "chargebackReversal";
        TypeEnum["ChargebackReversalCorrection"] = "chargebackReversalCorrection";
        TypeEnum["SecondChargeback"] = "secondChargeback";
        TypeEnum["SecondChargebackCorrection"] = "secondChargebackCorrection";
        TypeEnum["AtmWithdrawal"] = "atmWithdrawal";
        TypeEnum["AtmWithdrawalReversal"] = "atmWithdrawalReversal";
        TypeEnum["InternalTransfer"] = "internalTransfer";
        TypeEnum["InternalDirectDebit"] = "internalDirectDebit";
        TypeEnum["ManualCorrection"] = "manualCorrection";
        TypeEnum["InvoiceDeduction"] = "invoiceDeduction";
        TypeEnum["DepositCorrection"] = "depositCorrection";
        TypeEnum["ReserveAdjustment"] = "reserveAdjustment";
        TypeEnum["BankTransfer"] = "bankTransfer";
        TypeEnum["BankDirectDebit"] = "bankDirectDebit";
        TypeEnum["CardTransfer"] = "cardTransfer";
        TypeEnum["MiscCost"] = "miscCost";
        TypeEnum["PaymentCost"] = "paymentCost";
        TypeEnum["Fee"] = "fee";
        TypeEnum["Leftover"] = "leftover";
        TypeEnum["Grant"] = "grant";
        TypeEnum["CapitalFundsCollection"] = "capitalFundsCollection";
        TypeEnum["CashOutInstruction"] = "cashOutInstruction";
        TypeEnum["CashoutFee"] = "cashoutFee";
        TypeEnum["CashoutRepayment"] = "cashoutRepayment";
        TypeEnum["CashoutFunding"] = "cashoutFunding";
        TypeEnum["Repayment"] = "repayment";
        TypeEnum["Installment"] = "installment";
        TypeEnum["InstallmentReversal"] = "installmentReversal";
        TypeEnum["BalanceAdjustment"] = "balanceAdjustment";
        TypeEnum["BalanceRollover"] = "balanceRollover";
        TypeEnum["BalanceMigration"] = "balanceMigration";
    })(TypeEnum = Transfer.TypeEnum || (Transfer.TypeEnum = {}));
})(Transfer = exports.Transfer || (exports.Transfer = {}));
