// @adyen/api-library/lib/src/typings/transfers/transaction.js
"use strict";
/*
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transaction = void 0;
class Transaction {
    static getAttributeTypeMap() {
        return Transaction.attributeTypeMap;
    }
    constructor() {
    }
}
exports.Transaction = Transaction;
Transaction.discriminator = undefined;
Transaction.mapping = undefined;
Transaction.attributeTypeMap = [
    {
        "name": "accountHolder",
        "baseName": "accountHolder",
        "type": "ResourceReference",
        "format": ""
    },
    {
        "name": "amount",
        "baseName": "amount",
        "type": "Amount",
        "format": ""
    },
    {
        "name": "balanceAccount",
        "baseName": "balanceAccount",
        "type": "ResourceReference",
        "format": ""
    },
    {
        "name": "balancePlatform",
        "baseName": "balancePlatform",
        "type": "string",
        "format": ""
    },
    {
        "name": "bookingDate",
        "baseName": "bookingDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "creationDate",
        "baseName": "creationDate",
        "type": "Date",
        "format": "date-time"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string",
        "format": ""
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string",
        "format": ""
    },
    {
        "name": "paymentInstrument",
        "baseName": "paymentInstrument",
        "type": "PaymentInstrument | null",
        "format": ""
    },
    {
        "name": "referenceForBeneficiary",
        "baseName": "referenceForBeneficiary",
        "type": "string",
        "format": ""
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "Transaction.StatusEnum",
        "format": ""
    },
    {
        "name": "transfer",
        "baseName": "transfer",
        "type": "TransferView | null",
        "format": ""
    },
    {
        "name": "valueDate",
        "baseName": "valueDate",
        "type": "Date",
        "format": "date-time"
    }
];
(function (Transaction) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Booked"] = "booked";
        StatusEnum["Pending"] = "pending";
    })(StatusEnum = Transaction.StatusEnum || (Transaction.StatusEnum = {}));
})(Transaction = exports.Transaction || (exports.Transaction = {}));
